<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Raphael Sonabend and Andreas Bender">
<title>14&nbsp; Machine Learning Survival Models – Machine Learning in Survival Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./forests.html" rel="next">
<link href="./classical.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="styles.css">
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./classical.html">Models</a></li><li class="breadcrumb-item"><a href="./mlmodels.html"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Machine Learning Survival Models</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Machine Learning in Survival Analysis</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/mlsa-book/MLSA/tree/main/book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./Machine-Learning-in-Survival-Analysis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Symbols and Notation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Survival Analysis and Machine Learning</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">MLSA From Start to Finish</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./machinelearning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Statistical Learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./survival.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eha.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./survtsk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Survival Task</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meas_what.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">What are Survival Measures?</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meas_rank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Discrimination Measures</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meas_calib.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Calibration Measures</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meas_rules.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Evaluating Distributions by Scoring Rules</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meas_time.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Evaluating Survival Time</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meas_choosing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Choosing Measures</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Models</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./classical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Classical Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./mlmodels.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Machine Learning Survival Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./forests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Random Forests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./svm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./boosting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Boosting Methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./neuralnetworks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Neural Networks</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./alternatives.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Alternative Methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./models_choosing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Choosing Models</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Reduction Techniques</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reductions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Reductions</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./competing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Competing Risks Pipelines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discretetime.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Discrete Time Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./poisson.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Connections to Poisson Regression and Processes</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pseudo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Connections to Regression and Imputation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Advanced Methods</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Extensions and Outlook</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./common.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Common problems in survival analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./software.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Survival Software</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./next.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">What’s next for MLSA?</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./exercises.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exercises</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#sec-surv-ml" id="toc-sec-surv-ml" class="nav-link active" data-scroll-target="#sec-surv-ml"><span class="header-section-number">14.1</span> A Survey of Machine Learning Models for Survival Analysis</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/mlmodels.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/mlmodels.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./classical.html">Models</a></li><li class="breadcrumb-item"><a href="./mlmodels.html"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Machine Learning Survival Models</span></a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">
<span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Machine Learning Survival Models</span>
</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    TODO (150-200 WORDS)
  </div>
</div>


</header><div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Major changes expected!
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This page is a work in progress and major changes will be made over time.</strong></p>
</div>
</div>
<section id="sec-surv-ml" class="level2" data-number="14.1"><h2 data-number="14.1" class="anchored" data-anchor-id="sec-surv-ml">
<span class="header-section-number">14.1</span> A Survey of Machine Learning Models for Survival Analysis</h2>
<p>These next sections provide a technical, critical survey of machine learning models proposed for survival analysis with the focus on the ‘simpler’ setup of non-competing risks. Models are separated into their different ‘classes’ (1), which exists as a natural taxonomy in machine learning. Each class review is then further separated by first discussing the simpler and more standard regression setting, before expanding into their survival framework. The focus is once again on the different predict types of the model, which enables clear exposition and discussion around how some areas have successfully dealt with the survival predictive problem, whereas others have fallen short.</p>
<p>This is not the first survey of machine learning models for survival analysis. A recent 2017 survey <span class="citation" data-cites="Wang2017">(<a href="references.html#ref-Wang2017" role="doc-biblioref">P. Wang, Li, and Reddy 2019</a>)</span> focused on covering the breadth of machine learning models for survival analysis and this survey is recommended to the reader as a strong starting point to understand which ML models are available for survival analysis. However whilst this provides a comprehensive review and a ‘big-picture’ view, there is no discussion about how successful the discussed models are in solving the survival task.</p>
<p>A comprehensive survey of neural networks was presented by Schwarzer <span class="math inline">\(\textit{et al.}\)</span> (2000) <span class="citation" data-cites="Schwarzer2000">(<a href="references.html#ref-Schwarzer2000" role="doc-biblioref">Schwarzer, Vach, and Schumacher 2010</a>)</span> in which the authors collected the many ways in which neural networks have been ‘misused’ in the context of survival analysis. This level of criticism is vital in the context of survival analysis and healthcare data as transparency and understanding are often prioritised over predictive performance. Whilst the survey in this book will try not to be as critical as the Schwarzer review, it will aim to discuss models and how well they actually solve the survival problem.</p>
<p>Historically, surveys have focused primarily on predictive performance, which is generally preferred for complex classification and regression tasks. However in the context of survival analysis, transparency is of the utmost importance and any model that does not solve the task it claims to, despite strong predictive performance, can be considered sub-optimal. The survey will also examine the accessibility of survival models. A model need not be open-source to be accessible, but it should be ‘user-friendly’ and not require expert cross-domain knowledge. For example, a neural network may require knowledge of complex model building, but if set-up correctly could be handled without medical or survival knowledge. Whereas a Gaussian Process requires knowledge of the model class, simulation, (usually) Bayesian modelling, and also survival analysis.</p>
<ol class="example" type="1">
<li>provides information about the models reviewed in this survey, including a model reference for use in the <span class="citation" data-cites="Sonabend2021b">(<a href="references.html#ref-Sonabend2021b" role="doc-biblioref">R. E. B. Sonabend 2021</a>)</span> benchmark experiment, the predict types of the model, and in which <span class="math inline">\(\textsf{R}\)</span> package it is implemented.</li>
</ol>
<div id="tbl-surv-ml-returns" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-surv-ml-returns-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;14.1: Summarising the models discussed in (<a href="#sec-surv-ml" class="quarto-xref"><span>Section 14.1</span></a>) by their model class and respective survival task.
</figcaption><div aria-describedby="tbl-surv-ml-returns-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<colgroup>
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 23%">
<col style="width: 23%">
<col style="width: 23%">
</colgroup>
<thead><tr class="header">
<th>Class<span class="math inline">\(^1\)</span>
</th>
<th>Name<span class="math inline">\(^2\)</span>
</th>
<th>Authors (Year)<span class="math inline">\(^3\)</span>
</th>
<th>Task<span class="math inline">\(^4\)</span>
</th>
<th>Implementation<span class="math inline">\(^5\)</span>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>RF</td>
<td>RRT</td>
<td>LeBlanc and Crowley (1992) <span class="citation" data-cites="LeBlanc1992">(<a href="references.html#ref-LeBlanc1992" role="doc-biblioref">LeBlanc and Crowley 1992</a>)</span>
</td>
<td>Rank</td>
<td>
<span class="math inline">\(\textbf{rpart}\)</span> <span class="citation" data-cites="pkgrpart">(<a href="references.html#ref-pkgrpart" role="doc-biblioref">Therneau and Atkinson 2019</a>)</span>
</td>
</tr>
<tr class="even">
<td>RF</td>
<td>RSDF-DEV</td>
<td>Hothorn <span class="math inline">\(\textit{et al.}\)</span> (2004) <span class="citation" data-cites="Hothorn2004">(<a href="references.html#ref-Hothorn2004" role="doc-biblioref">Hothorn et al. 2004</a>)</span>
</td>
<td>Prob.</td>
<td>
<span class="math inline">\(\textbf{ipred}\)</span> <span class="citation" data-cites="pkgipred">(<a href="references.html#ref-pkgipred" role="doc-biblioref">Peters and Hothorn 2019</a>)</span>
</td>
</tr>
<tr class="odd">
<td>RF</td>
<td>RRF</td>
<td>Ishwaran <span class="math inline">\(\textit{et al.}\)</span> (2004) <span class="citation" data-cites="Ishwaran2004">(<a href="references.html#ref-Ishwaran2004" role="doc-biblioref">H. Ishwaran et al. 2004</a>)</span>
</td>
<td>Rank</td>
<td>-</td>
</tr>
<tr class="even">
<td>RF</td>
<td>RSCIFF</td>
<td>Hothorn <span class="math inline">\(\textit{et al.}\)</span> (2006) <span class="citation" data-cites="Hothorn2005">(<a href="references.html#ref-Hothorn2005" role="doc-biblioref">Hothorn et al. 2005</a>)</span>
</td>
<td>Det., Prob.</td>
<td>
<span class="math inline">\(\textbf{party}\)</span> <span class="citation" data-cites="pkgparty">(<a href="references.html#ref-pkgparty" role="doc-biblioref">Hothorn, Hornik, and Zeileis 2006</a>)</span>, <span class="math inline">\(\textbf{partykit}\)</span> <span class="citation" data-cites="pkgpartykit">(<a href="references.html#ref-pkgpartykit" role="doc-biblioref">Hothorn and Zeileis 2015</a>)</span>
</td>
</tr>
<tr class="odd">
<td>RF</td>
<td>RSDF-STAT</td>
<td>Ishwaran <span class="math inline">\(\textit{et al.}\)</span> (2008) <span class="citation" data-cites="Ishwaran2008">(<a href="references.html#ref-Ishwaran2008" role="doc-biblioref">B. H. Ishwaran et al. 2008</a>)</span>
</td>
<td>Prob.</td>
<td>
<span class="math inline">\(\textbf{randomForestSRC}\)</span> <span class="citation" data-cites="pkgrfsrc">(<a href="references.html#ref-pkgrfsrc" role="doc-biblioref">H. Ishwaran and Kogalur 2018</a>)</span>, <span class="math inline">\(\textbf{ranger}\)</span> <span class="citation" data-cites="pkgranger">(<a href="references.html#ref-pkgranger" role="doc-biblioref">Wright and Ziegler 2017</a>)</span>
</td>
</tr>
<tr class="even">
<td>GBM</td>
<td>GBM-COX</td>
<td>Ridgeway (1999) <span class="citation" data-cites="Ridgeway1999">(<a href="references.html#ref-Ridgeway1999" role="doc-biblioref">Ridgeway 1999</a>)</span> &amp; Buhlmann (2007) <span class="citation" data-cites="Buhlmann2007">(<a href="references.html#ref-Buhlmann2007" role="doc-biblioref">Buhlmann and Hothorn 2007</a>)</span>
</td>
<td>Prob.</td>
<td>
<span class="math inline">\(\textbf{mboost}\)</span> <span class="citation" data-cites="pkgmboost">(<a href="references.html#ref-pkgmboost" role="doc-biblioref">Hothorn et al. 2020</a>)</span>, <span class="math inline">\(\textbf{xgboost}\)</span> <span class="citation" data-cites="pkgxgboost">(<a href="references.html#ref-pkgxgboost" role="doc-biblioref">Chen et al. 2020</a>)</span>, <span class="math inline">\(\textbf{gbm}\)</span> <span class="citation" data-cites="pkggbm">(<a href="references.html#ref-pkggbm" role="doc-biblioref">Greenwell et al. 2019</a>)</span>
</td>
</tr>
<tr class="odd">
<td>GBM</td>
<td>CoxBoost</td>
<td>Binder &amp; Schumacher (2008) <span class="citation" data-cites="Binder2008">(<a href="references.html#ref-Binder2008" role="doc-biblioref">Binder and Schumacher 2008</a>)</span>
</td>
<td>Prob.</td>
<td>
<span class="math inline">\(\textbf{CoxBoost}\)</span> <span class="citation" data-cites="pkgcoxboost">(<a href="references.html#ref-pkgcoxboost" role="doc-biblioref">Binder 2013</a>)</span>
</td>
</tr>
<tr class="even">
<td>GBM</td>
<td>GBM-AFT</td>
<td>Schmid &amp; Hothorn (2008) <span class="citation" data-cites="Schmid2008b">(<a href="references.html#ref-Schmid2008b" role="doc-biblioref">Schmid and Hothorn 2008</a>)</span>
</td>
<td>Det.</td>
<td>
<span class="math inline">\(\textbf{mboost}\)</span>, <span class="math inline">\(\textbf{xgboost}\)</span>
</td>
</tr>
<tr class="odd">
<td>GBM</td>
<td>GBM-BUJAR</td>
<td>Wang &amp; Wang (2010) <span class="citation" data-cites="Wang2010">(<a href="references.html#ref-Wang2010" role="doc-biblioref">Z. Wang and Wang 2010</a>)</span>
</td>
<td>Det.</td>
<td>
<span class="math inline">\(\textbf{bujar}\)</span> <span class="citation" data-cites="pkgbujar">(<a href="references.html#ref-pkgbujar" role="doc-biblioref">Z. Wang 2019</a>)</span>
</td>
</tr>
<tr class="even">
<td>GBM</td>
<td>GBM-GEH</td>
<td>Johnson &amp; Long (2011) <span class="citation" data-cites="Johnson2011">(<a href="references.html#ref-Johnson2011" role="doc-biblioref">Johnson and Long 2011</a>)</span>
</td>
<td>Det.</td>
<td><span class="math inline">\(\textbf{mboost}\)</span></td>
</tr>
<tr class="odd">
<td>GBM</td>
<td>GBM-UNO</td>
<td>Mayr &amp; Schmid (2014) <span class="citation" data-cites="Mayr2014">(<a href="references.html#ref-Mayr2014" role="doc-biblioref">Mayr and Schmid 2014</a>)</span>
</td>
<td>Rank</td>
<td><span class="math inline">\(\textbf{mboost}\)</span></td>
</tr>
<tr class="even">
<td>SVM</td>
<td>SVCR</td>
<td>Shivaswamy <span class="math inline">\(\textit{et al.}\)</span> (2007) <span class="citation" data-cites="Shivaswamy2007">(<a href="references.html#ref-Shivaswamy2007" role="doc-biblioref">Shivaswamy, Chu, and Jansche 2007</a>)</span>
</td>
<td>Det.</td>
<td>
<span class="math inline">\(\textbf{survivalsvm}\)</span> <span class="citation" data-cites="pkgsurvivalsvm">(<a href="references.html#ref-pkgsurvivalsvm" role="doc-biblioref">Fouodo et al. 2018</a>)</span>
</td>
</tr>
<tr class="odd">
<td>SVM</td>
<td>SSVM-Rank</td>
<td>Van Belle <span class="math inline">\(\textit{et al.}\)</span> (2007) <span class="citation" data-cites="VanBelle2007">(<a href="references.html#ref-VanBelle2007" role="doc-biblioref">Van Belle et al. 2007</a>)</span>
</td>
<td>Rank</td>
<td><span class="math inline">\(\textbf{survivalsvm}\)</span></td>
</tr>
<tr class="even">
<td>SVM</td>
<td>SVRc</td>
<td>Khan and Zubek (2008) <span class="citation" data-cites="Khan2008">(<a href="references.html#ref-Khan2008" role="doc-biblioref">Khan and Bayer Zubek 2008</a>)</span>
</td>
<td>Det.</td>
<td>-</td>
</tr>
<tr class="odd">
<td>SVM</td>
<td>SSVM-Hybrid</td>
<td>Van Belle (2011) <span class="citation" data-cites="VanBelle2011b">(<a href="references.html#ref-VanBelle2011b" role="doc-biblioref">Van Belle et al. 2011</a>)</span>
</td>
<td>Det.</td>
<td><span class="math inline">\(\textbf{survivalsvm}\)</span></td>
</tr>
<tr class="even">
<td>SVM</td>
<td>SSVR-MRL</td>
<td>Goli <span class="math inline">\(\textit{et al.}\)</span> (2016) <span class="citation" data-cites="Goli2016a Goli2016b">(<a href="references.html#ref-Goli2016a" role="doc-biblioref">Goli, Mahjub, Faradmal, and Soltanian 2016</a>; <a href="references.html#ref-Goli2016b" role="doc-biblioref">Goli, Mahjub, Faradmal, Mashayekhi, et al. 2016</a>)</span>
</td>
<td>Det.</td>
<td>-</td>
</tr>
<tr class="odd">
<td>ANN</td>
<td>ANN-CDP</td>
<td>Liestl <span class="math inline">\(\textit{et al.}\)</span> (1994) <span class="citation" data-cites="Liestol1994">(<a href="references.html#ref-Liestol1994" role="doc-biblioref">Liestol, Andersen, and Andersen 1994</a>)</span>
</td>
<td>Prob.</td>
<td>-</td>
</tr>
<tr class="even">
<td>ANN</td>
<td>ANN-COX</td>
<td>Faraggi and Simon (1995) <span class="citation" data-cites="Faraggi1995">(<a href="references.html#ref-Faraggi1995" role="doc-biblioref">Faraggi and Simon 1995</a>)</span>
</td>
<td>Rank</td>
<td>-</td>
</tr>
<tr class="odd">
<td>ANN</td>
<td>PLANN</td>
<td>Biganzoli <span class="math inline">\(\textit{et al.}\)</span> (1998) <span class="citation" data-cites="Biganzoli1998">(<a href="references.html#ref-Biganzoli1998" role="doc-biblioref">Biganzoli et al. 1998</a>)</span>
</td>
<td>Prob.</td>
<td>-</td>
</tr>
<tr class="even">
<td>ANN</td>
<td>COX-NNET</td>
<td>Ching <span class="math inline">\(\textit{et al.}\)</span> (2018) <span class="citation" data-cites="Ching2018a">(<a href="references.html#ref-Ching2018a" role="doc-biblioref">Ching, Zhu, and Garmire 2018</a>)</span>
</td>
<td>Prob.</td>
<td>
<span class="math inline">\(^*\)</span> <span class="citation" data-cites="pkgcoxnnet">(<a href="references.html#ref-pkgcoxnnet" role="doc-biblioref">Ching 2015</a>)</span>
</td>
</tr>
<tr class="odd">
<td>ANN</td>
<td>DeepSurv</td>
<td>Katzman <span class="math inline">\(\textit{et al.}\)</span> (2018) <span class="citation" data-cites="Katzman2018">(<a href="references.html#ref-Katzman2018" role="doc-biblioref">Katzman et al. 2018</a>)</span>
</td>
<td>Prob.</td>
<td>
<span class="math inline">\(\textbf{survivalmodels}\)</span> <span class="citation" data-cites="pkgsurvivalmodels">(<a href="references.html#ref-pkgsurvivalmodels" role="doc-biblioref">R. Sonabend 2020</a>)</span>
</td>
</tr>
<tr class="even">
<td>ANN</td>
<td>DeepHit</td>
<td>Lee <span class="math inline">\(\textit{et al.}\)</span> (2018) <span class="citation" data-cites="Lee2018a">(<a href="references.html#ref-Lee2018a" role="doc-biblioref">Lee et al. 2018</a>)</span>
</td>
<td>Prob.</td>
<td><span class="math inline">\(\textbf{survivalmodels}\)</span></td>
</tr>
<tr class="odd">
<td>ANN</td>
<td>Nnet-survival</td>
<td>Gensheimer &amp; Narasimhan (2019) <span class="citation" data-cites="Gensheimer2019">(<a href="references.html#ref-Gensheimer2019" role="doc-biblioref">Gensheimer and Narasimhan 2019</a>)</span>
</td>
<td>Prob.</td>
<td><span class="math inline">\(\textbf{survivalmodels}\)</span></td>
</tr>
<tr class="even">
<td>ANN</td>
<td>Cox-Time</td>
<td>Kvamme <span class="math inline">\(\textit{et al.}\)</span> (2019) <span class="citation" data-cites="Kvamme2019a">(<a href="references.html#ref-Kvamme2019a" role="doc-biblioref">Kvamme, Borgan, and Scheel 2019</a>)</span>
</td>
<td>Prob.</td>
<td><span class="math inline">\(\textbf{survivalmodels}\)</span></td>
</tr>
<tr class="odd">
<td>ANN</td>
<td>PC-Hazard</td>
<td>Kvamme &amp; Borgan (2019) <span class="citation" data-cites="Kvamme2019">(<a href="references.html#ref-Kvamme2019" role="doc-biblioref"><strong>Kvamme2019?</strong></a>)</span>
</td>
<td>Prob.</td>
<td><span class="math inline">\(\textbf{survivalmodels}\)</span></td>
</tr>
<tr class="even">
<td>ANN</td>
<td>RankDeepSurv</td>
<td>Jing <span class="math inline">\(\textit{et al.}\)</span> (2019) <span class="citation" data-cites="Jing2019">(<a href="references.html#ref-Jing2019" role="doc-biblioref">Jing et al. 2019</a>)</span>
</td>
<td>Det.</td>
<td>
<span class="math inline">\(\textbf{RankDeepSurv}\)</span><span class="math inline">\(^{\ast, \dagger}\)</span> <span class="citation" data-cites="pkgrankdeepsurv">(<a href="references.html#ref-pkgrankdeepsurv" role="doc-biblioref">Jing et al. 2018</a>)</span>
</td>
</tr>
<tr class="odd">
<td>ANN</td>
<td>DNNSurv</td>
<td>Zhao &amp; Fend (2020) <span class="citation" data-cites="Zhao2020">(<a href="references.html#ref-Zhao2020" role="doc-biblioref">Zhao and Feng 2020</a>)</span>
</td>
<td>Prob.</td>
<td><span class="math inline">\(\textbf{survivalmodels}\)</span></td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p><sup> * 1. Model Class. RSF – Random Survival Forest; GBM – Gradient Boosting Machine; SVM – Support Vector Machine; ANN – Artificial Neural Network. There is some abuse of notation here as some of the RSFs are actually decision trees and some GBMs do not use gradient boosting. * 2. Model identifier used in this section and <span class="citation" data-cites="Sonabend2021b">(<a href="references.html#ref-Sonabend2021b" role="doc-biblioref">R. E. B. Sonabend 2021</a>)</span>. * 3. Authors and year of publication, for RSFs this is the paper most attributed to the algorithm. * 4. Survival task type: Deterministic (Det.), Probabilistic (Prob.), Ranking (Rank). * 5. If available in <span class="math inline">\(\textsf{R}\)</span> then the package in which the model is implemented, otherwise `<span class="math inline">\(\ast\)</span>’ signifies a model is only available in Python. With the exception of DNNSurv, all ANNs in <span class="math inline">\(\textbf{survivalmodels}\)</span> are implemented from the Python package <span class="math inline">\(\textbf{pycox}\)</span> <span class="citation" data-cites="pkgpycox">(<a href="references.html#ref-pkgpycox" role="doc-biblioref">Kvamme 2018</a>)</span> with <span class="math inline">\(\textbf{reticulate}\)</span> <span class="citation" data-cites="pkgreticulate">(<a href="references.html#ref-pkgreticulate" role="doc-biblioref">Ushey, Allaire, and Tang 2020</a>)</span>. * <span class="math inline">\(\dagger\)</span> – Code available to create model but not implemented ‘off-shelf’. </sup></p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Biganzoli1998" class="csl-entry" role="listitem">
Biganzoli, Elia, Patrizia Boracchi, Luigi Mariani, and Ettore Marubini. 1998. <span>“<span class="nocase">Feed forward neural networks for the analysis of censored survival data: a partial logistic regression approach</span>.”</span> <em>Statistics in Medicine</em> 17 (10): 1169–86. <a href="https://doi.org/10.1002/(SICI)1097-0258(19980530)17:10<1169::AID-SIM796>3.0.CO;2-D">https://doi.org/10.1002/(SICI)1097-0258(19980530)17:10&lt;1169::AID-SIM796&gt;3.0.CO;2-D</a>.
</div>
<div id="ref-pkgcoxboost" class="csl-entry" role="listitem">
Binder, Harald. 2013. <span>“<span class="nocase">CoxBoost: Cox models by likelihood based boosting for a single survival endpoint or competing risks</span>.”</span> CRAN.
</div>
<div id="ref-Binder2008" class="csl-entry" role="listitem">
Binder, Harald, and Martin Schumacher. 2008. <span>“<span class="nocase">Allowing for mandatory covariates in boosting estimation of sparse high-dimensional survival models</span>.”</span> <em>BMC Bioinformatics</em> 9 (1): 14. <a href="https://doi.org/10.1186/1471-2105-9-14">https://doi.org/10.1186/1471-2105-9-14</a>.
</div>
<div id="ref-Buhlmann2007" class="csl-entry" role="listitem">
Buhlmann, Peter, and Torsten Hothorn. 2007. <span>“<span class="nocase">Boosting Algorithms: Regularization, Prediction and Model Fitting</span>.”</span> <em>Statist. Sci.</em> 22 (4): 477–505. <a href="https://doi.org/10.1214/07-STS242">https://doi.org/10.1214/07-STS242</a>.
</div>
<div id="ref-pkgxgboost" class="csl-entry" role="listitem">
Chen, Tianqi, Tong He, Michael Benesty, Vadim Khotilovich, Yuan Tang, Hyunsu Cho, Kailong Chen, et al. 2020. <span>“<span class="nocase">xgboost: Extreme Gradient Boosting</span>.”</span> CRAN. <a href="https://cran.r-project.org/package=xgboost">https://cran.r-project.org/package=xgboost</a>.
</div>
<div id="ref-pkgcoxnnet" class="csl-entry" role="listitem">
Ching, Travers. 2015. <span>“Cox-Nnet.”</span> <a href="https://github.com/lanagarmire/cox-nnet">https://github.com/lanagarmire/cox-nnet</a>.
</div>
<div id="ref-Ching2018a" class="csl-entry" role="listitem">
Ching, Travers, Xun Zhu, and Lana X Garmire. 2018. <span>“<span class="nocase">Cox-nnet: An artificial neural network method for prognosis prediction of high-throughput omics data</span>.”</span> <em>PLOS Computational Biology</em> 14 (4): e1006076. <a href="https://doi.org/10.1371/journal.pcbi.1006076">https://doi.org/10.1371/journal.pcbi.1006076</a>.
</div>
<div id="ref-Faraggi1995" class="csl-entry" role="listitem">
Faraggi, David, and Richard Simon. 1995. <span>“<span class="nocase">A neural network model for survival data</span>.”</span> <em>Statistics in Medicine</em> 14 (1): 73–82. <a href="https://doi.org/10.1002/sim.4780140108">https://doi.org/10.1002/sim.4780140108</a>.
</div>
<div id="ref-pkgsurvivalsvm" class="csl-entry" role="listitem">
Fouodo, Cesaire J K, I Konig, C Weihs, A Ziegler, and M Wright. 2018. <span>“<span class="nocase">Support vector machines for survival analysis with R</span>.”</span> <em>The R Journal</em> 10 (July): 412–23.
</div>
<div id="ref-Gensheimer2019" class="csl-entry" role="listitem">
Gensheimer, Michael F, and Balasubramanian Narasimhan. 2019. <span>“<span class="nocase">A scalable discrete-time survival model for neural networks</span>.”</span> <em>PeerJ</em> 7: e6257.
</div>
<div id="ref-Goli2016b" class="csl-entry" role="listitem">
Goli, Shahrbanoo, Hossein Mahjub, Javad Faradmal, Hoda Mashayekhi, and Ali-Reza Soltanian. 2016. <span>“<span class="nocase">Survival Prediction and Feature Selection in Patients with Breast Cancer Using Support Vector Regression</span>.”</span> Edited by Francesco Pappalardo. <em>Computational and Mathematical Methods in Medicine</em> 2016: 2157984. <a href="https://doi.org/10.1155/2016/2157984">https://doi.org/10.1155/2016/2157984</a>.
</div>
<div id="ref-Goli2016a" class="csl-entry" role="listitem">
Goli, Shahrbanoo, Hossein Mahjub, Javad Faradmal, and Ali-Reza Soltanian. 2016. <span>“<span class="nocase">Performance Evaluation of Support Vector Regression Models for Survival Analysis: A Simulation Study</span>.”</span> <em>International Journal of Advanced Computer Science and Applications</em> 7 (June). <a href="https://doi.org/10.14569/IJACSA.2016.070650">https://doi.org/10.14569/IJACSA.2016.070650</a>.
</div>
<div id="ref-pkggbm" class="csl-entry" role="listitem">
Greenwell, Brandon, Bradley Boehmke, Jay Cunningham, and. GBM Developers. 2019. <span>“<span class="nocase">gbm: Generalized Boosted Regression Models</span>.”</span> CRAN. <a href="https://cran.r-project.org/package=gbm">https://cran.r-project.org/package=gbm</a>.
</div>
<div id="ref-pkgmboost" class="csl-entry" role="listitem">
Hothorn, Torsten, Peter Buehlmann, Thomas Kneib, Matthias Schmid, and Benjamin Hofner. 2020. <span>“<span class="nocase">mboost: Model-Based Boosting</span>.”</span> CRAN. <a href="https://cran.r-project.org/package=mboost">https://cran.r-project.org/package=mboost</a>.
</div>
<div id="ref-Hothorn2005" class="csl-entry" role="listitem">
Hothorn, Torsten, Peter Bühlmann, Sandrine Dudoit, Annette Molinaro, and Mark J Van Der Laan. 2005. <span>“<span class="nocase">Survival ensembles</span>.”</span> <em>Biostatistics</em> 7 (3): 355–73. <a href="https://doi.org/10.1093/biostatistics/kxj011">https://doi.org/10.1093/biostatistics/kxj011</a>.
</div>
<div id="ref-pkgparty" class="csl-entry" role="listitem">
Hothorn, Torsten, Kurt Hornik, and Achim Zeileis. 2006. <span>“<span>Unbiased Recursive Partitioning: A Conditional Inference Framework</span>.”</span> <em>Journal of Computational and Graphical Statistics</em> 15 (3): 651—–674.
</div>
<div id="ref-Hothorn2004" class="csl-entry" role="listitem">
Hothorn, Torsten, Berthold Lausen, Axel Benner, and Martin Radespiel-Tröger. 2004. <span>“<span class="nocase">Bagging survival trees</span>.”</span> <em>Statistics in Medicine</em> 23 (1): 77–91. <a href="https://doi.org/10.1002/sim.1593">https://doi.org/10.1002/sim.1593</a>.
</div>
<div id="ref-pkgpartykit" class="csl-entry" role="listitem">
Hothorn, Torsten, and Achim Zeileis. 2015. <span>“<span class="nocase">partykit: A Modular Toolkit for Recursive Partytioning in R.</span>”</span> <em>Journal of Machine Learning Research</em> 16: 3905–9. <a href="http://jmlr.org/papers/v16/hothorn15a.html">http://jmlr.org/papers/v16/hothorn15a.html</a>.
</div>
<div id="ref-Ishwaran2008" class="csl-entry" role="listitem">
Ishwaran, By Hemant, Udaya B Kogalur, Eugene H Blackstone, and Michael S Lauer. 2008. <span>“<span class="nocase">Random survival forests</span>.”</span> <em>The Annals of Statistics</em> 2 (3): 841–60. <a href="https://doi.org/10.1214/08-AOAS169">https://doi.org/10.1214/08-AOAS169</a>.
</div>
<div id="ref-Ishwaran2004" class="csl-entry" role="listitem">
Ishwaran, Hemant, Eugene H Blackstone, Claire E Pothier, and Michael S Lauer. 2004. <span>“<span class="nocase">Relative Risk Forests for Exercise Heart Rate Recovery as a Predictor of Mortality</span>.”</span> <em>Journal of the American Statistical Association</em> 99 (467): 591–600. <a href="https://doi.org/10.1198/016214504000000638">https://doi.org/10.1198/016214504000000638</a>.
</div>
<div id="ref-pkgrfsrc" class="csl-entry" role="listitem">
Ishwaran, Hemant, and Udaya B Kogalur. 2018. <span>“<span class="nocase">randomForestSRC</span>.”</span> <a href="https://cran.r-project.org/package=randomForestSRC">https://cran.r-project.org/package=randomForestSRC</a>.
</div>
<div id="ref-pkgrankdeepsurv" class="csl-entry" role="listitem">
Jing, Bingzhong, Tao Zhang, Zixian Wang, Ying Jin, Kuiyuan Liu, Wenze Qiu, Liangru Ke, et al. 2018. <span>“<span>RankDeepSurv</span>.”</span> <a href="https://github.com/sysucc-ailab/RankDeepSurv">https://github.com/sysucc-ailab/RankDeepSurv</a>.
</div>
<div id="ref-Jing2019" class="csl-entry" role="listitem">
———, et al. 2019. <span>“<span class="nocase">A deep survival analysis method based on ranking</span>.”</span> <em>Artificial Intelligence in Medicine</em> 98: 1–9. https://doi.org/<a href="https://doi.org/10.1016/j.artmed.2019.06.001">https://doi.org/10.1016/j.artmed.2019.06.001</a>.
</div>
<div id="ref-Johnson2011" class="csl-entry" role="listitem">
Johnson, Brent A, and Qi Long. 2011. <span>“<span class="nocase">Survival ensembles by the sum of pairwise differences with application to lung cancer microarray studies</span>.”</span> <em>Ann. Appl. Stat.</em> 5 (2A): 1081–101. <a href="https://doi.org/10.1214/10-AOAS426">https://doi.org/10.1214/10-AOAS426</a>.
</div>
<div id="ref-Katzman2018" class="csl-entry" role="listitem">
Katzman, Jared L, Uri Shaham, Alexander Cloninger, Jonathan Bates, Tingting Jiang, and Yuval Kluger. 2018. <span>“<span class="nocase">DeepSurv: personalized treatment recommender system using a Cox proportional hazards deep neural network</span>.”</span> <em>BMC Medical Research Methodology</em> 18 (1): 24. <a href="https://doi.org/10.1186/s12874-018-0482-1">https://doi.org/10.1186/s12874-018-0482-1</a>.
</div>
<div id="ref-Khan2008" class="csl-entry" role="listitem">
Khan, Faisal M., and Valentina Bayer Zubek. 2008. <span>“<span class="nocase">Support vector regression for censored data (SVRc): A novel tool for survival analysis</span>.”</span> <em>Proceedings - IEEE International Conference on Data Mining, ICDM</em>, 863–68. <a href="https://doi.org/10.1109/ICDM.2008.50">https://doi.org/10.1109/ICDM.2008.50</a>.
</div>
<div id="ref-pkgpycox" class="csl-entry" role="listitem">
Kvamme, Håvard. 2018. <span>“Pycox.”</span> <a href="https://pypi.org/project/pycox/">https://pypi.org/project/pycox/</a>.
</div>
<div id="ref-Kvamme2019a" class="csl-entry" role="listitem">
Kvamme, Håvard, Ørnulf Borgan, and Ida Scheel. 2019. <span>“<span class="nocase">Time-to-event prediction with neural networks and Cox regression</span>.”</span> <em>Journal of Machine Learning Research</em> 20 (129): 1–30.
</div>
<div id="ref-LeBlanc1992" class="csl-entry" role="listitem">
LeBlanc, Michael, and John Crowley. 1992. <span>“<span class="nocase">Relative Risk Trees for Censored Survival Data</span>.”</span> <em>Biometrics</em> 48 (2): 411–25. <a href="https://doi.org/10.2307/2532300">https://doi.org/10.2307/2532300</a>.
</div>
<div id="ref-Lee2018a" class="csl-entry" role="listitem">
Lee, Changhee, William Zame, Jinsung Yoon, and Mihaela Van der Schaar. 2018. <span>“<span class="nocase">DeepHit: A Deep Learning Approach to Survival Analysis With Competing Risks</span>.”</span> <em>Proceedings of the AAAI Conference on Artificial Intelligence</em> 32 (1). <a href="https://doi.org/10.1609/aaai.v32i1.11842">https://doi.org/10.1609/aaai.v32i1.11842</a>.
</div>
<div id="ref-Liestol1994" class="csl-entry" role="listitem">
Liestol, Knut, Per Kragh Andersen, and Ulrich Andersen. 1994. <span>“<span class="nocase">Survival analysis and neural nets</span>.”</span> <em>Statistics in Medicine</em> 13 (12): 1189–1200. <a href="https://doi.org/10.1002/sim.4780131202">https://doi.org/10.1002/sim.4780131202</a>.
</div>
<div id="ref-Mayr2014" class="csl-entry" role="listitem">
Mayr, Andreas, and Matthias Schmid. 2014. <span>“<span class="nocase">Boosting the concordance index for survival data–a unified framework to derive and evaluate biomarker combinations</span>.”</span> <em>PloS One</em> 9 (1): e84483–83. <a href="https://doi.org/10.1371/journal.pone.0084483">https://doi.org/10.1371/journal.pone.0084483</a>.
</div>
<div id="ref-pkgipred" class="csl-entry" role="listitem">
Peters, Andrea, and Torsten Hothorn. 2019. <span>“<span class="nocase">ipred: Improved Predictors</span>.”</span> CRAN. <a href="https://cran.r-project.org/package=ipred">https://cran.r-project.org/package=ipred</a>.
</div>
<div id="ref-Ridgeway1999" class="csl-entry" role="listitem">
Ridgeway, Greg. 1999. <span>“<span class="nocase">The state of boosting</span>.”</span> <em>Computing Science and Statistics</em> 31: 172—–181.
</div>
<div id="ref-Schmid2008b" class="csl-entry" role="listitem">
Schmid, Matthias, and Torsten Hothorn. 2008. <span>“<span class="nocase">Flexible boosting of accelerated failure time models</span>.”</span> <em>BMC Bioinformatics</em> 9 (February): 269. <a href="https://doi.org/10.1186/1471-2105-9-269">https://doi.org/10.1186/1471-2105-9-269</a>.
</div>
<div id="ref-Schwarzer2000" class="csl-entry" role="listitem">
Schwarzer, Guido, Werner Vach, and Martin Schumacher. 2010. <span>“<span class="nocase">Estimation of prediction error for survival models</span>.”</span> <em>Statistics in Medicine</em> 29 (2): 262–74. <a href="https://doi.org/10.1002/(SICI)1097-0258(20000229)19:4<541::AID-SIM355>3.0.CO;2-V">https://doi.org/10.1002/(SICI)1097-0258(20000229)19:4&lt;541::AID-SIM355&gt;3.0.CO;2-V</a>.
</div>
<div id="ref-Shivaswamy2007" class="csl-entry" role="listitem">
Shivaswamy, Pannagadatta K., Wei Chu, and Martin Jansche. 2007. <span>“<span class="nocase">A support vector approach to censored targets</span>.”</span> In <em>Proceedings - IEEE International Conference on Data Mining, ICDM</em>, 655–60. <a href="https://doi.org/10.1109/ICDM.2007.93">https://doi.org/10.1109/ICDM.2007.93</a>.
</div>
<div id="ref-pkgsurvivalmodels" class="csl-entry" role="listitem">
Sonabend, Raphael. 2020. <span>“<span class="nocase">survivalmodels: Models for Survival Analysis</span>.”</span> CRAN. <a href="https://raphaels1.r-universe.dev/ui#package:survivalmodels">https://raphaels1.r-universe.dev/ui#package:survivalmodels</a>.
</div>
<div id="ref-Sonabend2021b" class="csl-entry" role="listitem">
Sonabend, Raphael Edward Benjamin. 2021. <span>“<span class="nocase">A Theoretical and Methodological Framework for Machine Learning in Survival Analysis: Enabling Transparent and Accessible Predictive Modelling on Right-Censored Time-to-Event Data</span>.”</span> PhD, University College London (UCL). <a href="https://discovery.ucl.ac.uk/id/eprint/10129352/">https://discovery.ucl.ac.uk/id/eprint/10129352/</a>.
</div>
<div id="ref-pkgrpart" class="csl-entry" role="listitem">
Therneau, Terry M., and Beth Atkinson. 2019. <span>“<span class="nocase">rpart: Recursive Partitioning and Regression Trees</span>.”</span> CRAN.
</div>
<div id="ref-pkgreticulate" class="csl-entry" role="listitem">
Ushey, Kevin, J J Allaire, and Yuan Tang. 2020. <span>“<span class="nocase">reticulate: Interface to ’Python’</span>.”</span> CRAN. <a href="https://cran.r-project.org/package=reticulate">https://cran.r-project.org/package=reticulate</a>.
</div>
<div id="ref-VanBelle2007" class="csl-entry" role="listitem">
Van Belle, Vanya, Kristiaan Pelckmans, Johan A. K. Suykens, and Sabine Van Huffel. 2007. <span>“<span class="nocase">Support Vector Machines for Survival Analysis</span>.”</span> In <em>In Proceedings of the Third International Conference on Computational Intelligence in Medicine and Healthcare</em>. 1.
</div>
<div id="ref-VanBelle2011b" class="csl-entry" role="listitem">
Van Belle, Vanya, Kristiaan Pelckmans, Sabine Van Huffel, and Johan A. K. Suykens. 2011. <span>“<span class="nocase">Support vector methods for survival analysis: A comparison between ranking and regression approaches</span>.”</span> <em>Artificial Intelligence in Medicine</em> 53 (2): 107–18. <a href="https://doi.org/10.1016/j.artmed.2011.06.006">https://doi.org/10.1016/j.artmed.2011.06.006</a>.
</div>
<div id="ref-Wang2017" class="csl-entry" role="listitem">
Wang, Ping, Yan Li, and Chandan K. Reddy. 2019. <span>“<span class="nocase">Machine Learning for Survival Analysis</span>.”</span> <em>ACM Computing Surveys</em> 51 (6): 1–36. <a href="https://doi.org/10.1145/3214306">https://doi.org/10.1145/3214306</a>.
</div>
<div id="ref-pkgbujar" class="csl-entry" role="listitem">
Wang, Zhu. 2019. <span>“<span class="nocase">bujar: Buckley-James Regression for Survival Data with High-Dimensional Covariates</span>.”</span> CRAN. <a href="https://cran.r-project.org/package=bujar">https://cran.r-project.org/package=bujar</a>.
</div>
<div id="ref-Wang2010" class="csl-entry" role="listitem">
Wang, Zhu, and C Y Wang. 2010. <span>“<span class="nocase">Buckley-James Boosting for Survival Analysis with High-Dimensional Biomarker Data</span>.”</span> <em>Statistical Applications in Genetics and Molecular Biology</em> 9 (1). https://doi.org/<a href="https://doi.org/10.2202/1544-6115.1550">https://doi.org/10.2202/1544-6115.1550</a>.
</div>
<div id="ref-pkgranger" class="csl-entry" role="listitem">
Wright, Marvin N., and Andreas Ziegler. 2017. <span>“<span class="nocase">ranger: A Fast Implementation of Random Forests for High Dimensional Data in C++ and R</span>.”</span> <em>Journal of Statistical Software</em> 77 (1): 1—–17.
</div>
<div id="ref-Zhao2020" class="csl-entry" role="listitem">
Zhao, Lili, and Dai Feng. 2020. <span>“<span class="nocase">Deep Neural Networks for Survival Analysis Using Pseudo Values</span>.”</span> <em>IEEE Journal of Biomedical and Health Informatics</em> 24 (11): 3308–14. <a href="https://doi.org/10.1109/JBHI.2020.2980204">https://doi.org/10.1109/JBHI.2020.2980204</a>.
</div>
</div>
</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./classical.html" class="pagination-link" aria-label="Classical Models">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Classical Models</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./forests.html" class="pagination-link" aria-label="Random Forests">
        <span class="nav-page-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Random Forests</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="an">abstract:</span><span class="co"> TODO (150-200 WORDS)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>{{&lt; include _setup.qmd &gt;}}</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu"># Machine Learning Survival Models</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>{{&lt; include _wip.qmd &gt;}}</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">## A Survey of Machine Learning Models for Survival Analysis {#sec-surv-ml}</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>These next sections provide a technical, critical survey of machine learning models proposed for survival analysis with the focus on the 'simpler' setup of non-competing risks. Models are separated into their different 'classes' (@tab-surv-ml-returns), which exists as a natural taxonomy in machine learning. Each class review is then further separated by first discussing the simpler and more standard regression setting, before expanding into their survival framework. The focus is once again on the different predict types of the model, which enables clear exposition and discussion around how some areas have successfully dealt with the survival predictive problem, whereas others have fallen short.</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>This is not the first survey of machine learning models for survival analysis. A recent 2017 survey  <span class="co">[</span><span class="ot">@Wang2017</span><span class="co">]</span> focused on covering the breadth of machine learning models for survival analysis and this survey is recommended to the reader as a strong starting point to understand which ML models are available for survival analysis. However whilst this provides a comprehensive review and a 'big-picture' view, there is no discussion about how successful the discussed models are in solving the survival task.</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>A comprehensive survey of neural networks was presented by Schwarzer $\etal$ (2000)  <span class="co">[</span><span class="ot">@Schwarzer2000</span><span class="co">]</span> in which the authors collected the many ways in which neural networks have been 'misused' in the context of survival analysis. This level of criticism is vital in the context of survival analysis and healthcare data as transparency and understanding are often prioritised over predictive performance. Whilst the survey in this book will try not to be as critical as the Schwarzer review, it will aim to discuss models and how well they actually solve the survival problem.</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>Historically, surveys have focused primarily on predictive performance, which is generally preferred for complex classification and regression tasks. However in the context of survival analysis, transparency is of the utmost importance and any model that does not solve the task it claims to, despite strong predictive performance, can be considered sub-optimal. The survey will also examine the accessibility of survival models. A model need not be open-source to be accessible, but it should be 'user-friendly' and not require expert cross-domain knowledge. For example, a neural network may require knowledge of complex model building, but if set-up correctly could be handled without medical or survival knowledge. Whereas a Gaussian Process requires knowledge of the model class, simulation, (usually) Bayesian modelling, and also survival analysis.</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>(@tab-surv-ml-returns) provides information about the models reviewed in this survey, including a model reference for use in the <span class="co">[</span><span class="ot">@Sonabend2021b</span><span class="co">]</span> benchmark experiment, the predict types of the model, and in which $\Rstats$ package it is implemented.</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>| Class$^1$ |  Name$^2$ | Authors (Year)$^3$ | Task$^4$ | Implementation$^5$ |</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>| -- | -- | --- | --- | --- |</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>| RF | RRT  | LeBlanc and Crowley (1992)  <span class="co">[</span><span class="ot">@LeBlanc1992</span><span class="co">]</span> | Rank | $\pkg{rpart}$  <span class="co">[</span><span class="ot">@pkgrpart</span><span class="co">]</span> |</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>| RF | RSDF-DEV  | Hothorn $\etal$ (2004)  <span class="co">[</span><span class="ot">@Hothorn2004</span><span class="co">]</span> | Prob. | $\pkg{ipred}$  <span class="co">[</span><span class="ot">@pkgipred</span><span class="co">]</span> |</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>| RF | RRF  | Ishwaran $\etal$ (2004)  <span class="co">[</span><span class="ot">@Ishwaran2004</span><span class="co">]</span> | Rank | - |</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>| RF | RSCIFF  | Hothorn $\etal$ (2006)  <span class="co">[</span><span class="ot">@Hothorn2005</span><span class="co">]</span> | Det., Prob. | $\pkg{party}$  <span class="co">[</span><span class="ot">@pkgparty</span><span class="co">]</span>, $\pkg{partykit}$  <span class="co">[</span><span class="ot">@pkgpartykit</span><span class="co">]</span> |</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>| RF | RSDF-STAT  | Ishwaran $\etal$ (2008)  <span class="co">[</span><span class="ot">@Ishwaran2008</span><span class="co">]</span> | Prob. | $\pkg{randomForestSRC}$  <span class="co">[</span><span class="ot">@pkgrfsrc</span><span class="co">]</span>, $\pkg{ranger}$  <span class="co">[</span><span class="ot">@pkgranger</span><span class="co">]</span> |</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>| GBM | GBM-COX  | Ridgeway (1999)  <span class="co">[</span><span class="ot">@Ridgeway1999</span><span class="co">]</span> <span class="sc">\&amp;</span> Buhlmann (2007)  <span class="co">[</span><span class="ot">@Buhlmann2007</span><span class="co">]</span> | Prob. | $\pkg{mboost}$  <span class="co">[</span><span class="ot">@pkgmboost</span><span class="co">]</span>, $\pkg{xgboost}$  <span class="co">[</span><span class="ot">@pkgxgboost</span><span class="co">]</span>, $\pkg{gbm}$  <span class="co">[</span><span class="ot">@pkggbm</span><span class="co">]</span> |</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>| GBM | CoxBoost  | Binder <span class="sc">\&amp;</span> Schumacher (2008)  <span class="co">[</span><span class="ot">@Binder2008</span><span class="co">]</span> | Prob. | $\pkg{CoxBoost}$  <span class="co">[</span><span class="ot">@pkgcoxboost</span><span class="co">]</span> |</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>| GBM | GBM-AFT  | Schmid <span class="sc">\&amp;</span> Hothorn (2008)  <span class="co">[</span><span class="ot">@Schmid2008b</span><span class="co">]</span> | Det. | $\pkg{mboost}$, $\pkg{xgboost}$ |</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>| GBM | GBM-BUJAR  | Wang <span class="sc">\&amp;</span> Wang (2010)  <span class="co">[</span><span class="ot">@Wang2010</span><span class="co">]</span> | Det. | $\pkg{bujar}$  <span class="co">[</span><span class="ot">@pkgbujar</span><span class="co">]</span> |</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>| GBM | GBM-GEH  | Johnson <span class="sc">\&amp;</span> Long (2011)  <span class="co">[</span><span class="ot">@Johnson2011</span><span class="co">]</span> | Det. | $\pkg{mboost}$ |</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>| GBM | GBM-UNO  | Mayr <span class="sc">\&amp;</span> Schmid (2014)  <span class="co">[</span><span class="ot">@Mayr2014</span><span class="co">]</span> | Rank | $\pkg{mboost}$ |</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>| SVM | SVCR  | Shivaswamy $\etal$ (2007)  <span class="co">[</span><span class="ot">@Shivaswamy2007</span><span class="co">]</span> | Det. | $\pkg{survivalsvm}$  <span class="co">[</span><span class="ot">@pkgsurvivalsvm</span><span class="co">]</span> |</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>| SVM | SSVM-Rank  | Van Belle $\etal$ (2007)  <span class="co">[</span><span class="ot">@VanBelle2007</span><span class="co">]</span> | Rank | $\pkg{survivalsvm}$ |</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>| SVM | SVRc  | Khan and Zubek (2008)  <span class="co">[</span><span class="ot">@Khan2008</span><span class="co">]</span> | Det. | - |</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>| SVM | SSVM-Hybrid  | Van Belle (2011)  <span class="co">[</span><span class="ot">@VanBelle2011b</span><span class="co">]</span> | Det. | $\pkg{survivalsvm}$ |</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>| SVM | SSVR-MRL  | Goli $\etal$ (2016)  <span class="co">[</span><span class="ot">@Goli2016a; @Goli2016b</span><span class="co">]</span> | Det. | - |</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>| ANN | ANN-CDP  | Liest\o l $\etal$ (1994)  <span class="co">[</span><span class="ot">@Liestol1994</span><span class="co">]</span> | Prob. | - |</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>| ANN | ANN-COX  | Faraggi and Simon (1995)  <span class="co">[</span><span class="ot">@Faraggi1995</span><span class="co">]</span> | Rank | -|</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>| ANN | PLANN  | Biganzoli $\etal$ (1998)  <span class="co">[</span><span class="ot">@Biganzoli1998</span><span class="co">]</span> | Prob. | - |</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>| ANN | COX-NNET  | Ching $\etal$ (2018)  <span class="co">[</span><span class="ot">@Ching2018a</span><span class="co">]</span> | Prob. | \pkg{cox-nnet}$^*$  <span class="co">[</span><span class="ot">@pkgcoxnnet</span><span class="co">]</span>|</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>| ANN | DeepSurv  | Katzman $\etal$ (2018)  <span class="co">[</span><span class="ot">@Katzman2018</span><span class="co">]</span> | Prob. | $\pkg{survivalmodels}$  <span class="co">[</span><span class="ot">@pkgsurvivalmodels</span><span class="co">]</span>|</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>| ANN | DeepHit  | Lee $\etal$ (2018)  <span class="co">[</span><span class="ot">@Lee2018a</span><span class="co">]</span> | Prob. | $\pkg{survivalmodels}$|</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>| ANN | Nnet-survival  | Gensheimer <span class="sc">\&amp;</span> Narasimhan (2019)  <span class="co">[</span><span class="ot">@Gensheimer2019</span><span class="co">]</span> | Prob. | $\pkg{survivalmodels}$|</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>| ANN | Cox-Time  | Kvamme $\etal$ (2019)  <span class="co">[</span><span class="ot">@Kvamme2019a</span><span class="co">]</span> | Prob. | $\pkg{survivalmodels}$|</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>| ANN | PC-Hazard  | Kvamme <span class="sc">\&amp;</span> Borgan (2019)  <span class="co">[</span><span class="ot">@Kvamme2019</span><span class="co">]</span> | Prob. | $\pkg{survivalmodels}$|</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>| ANN | RankDeepSurv  | Jing $\etal$ (2019)  <span class="co">[</span><span class="ot">@Jing2019</span><span class="co">]</span> | Det. | $\pkg{RankDeepSurv}$$^{\ast, \dagger}$  <span class="co">[</span><span class="ot">@pkgrankdeepsurv</span><span class="co">]</span>|</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>| ANN | DNNSurv  | Zhao <span class="sc">\&amp;</span> Fend (2020)  <span class="co">[</span><span class="ot">@Zhao2020</span><span class="co">]</span> | Prob. | $\pkg{survivalmodels}$ |</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>: Summarising the models discussed in (@sec-surv-ml) by their model class and respective survival task. {#tbl-surv-ml-returns}</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>&lt;sup&gt;</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>1. Model Class. RSF -- Random Survival Forest; GBM -- Gradient Boosting Machine; SVM -- Support Vector Machine; ANN -- Artificial Neural Network. There is some abuse of notation here as some of the RSFs are actually decision trees and some GBMs do not use gradient boosting.</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>2. Model identifier used in this section and <span class="co">[</span><span class="ot">@Sonabend2021b</span><span class="co">]</span>.</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>3. Authors and year of publication, for RSFs this is the paper most attributed to the algorithm.</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>4. Survival task type: Deterministic (Det.), Probabilistic (Prob.), Ranking (Rank).</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>5. If available in $\Rstats$ then the package in which the model is implemented, otherwise `$\ast$' signifies a model is only available in Python. With the exception of DNNSurv, all ANNs in $\pkg{survivalmodels}$ are implemented from the Python package $\pkg{pycox}$  <span class="co">[</span><span class="ot">@pkgpycox</span><span class="co">]</span> with $\pkg{reticulate}$  <span class="co">[</span><span class="ot">@pkgreticulate</span><span class="co">]</span>.</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>$\dagger$ -- Code available to create model but not implemented 'off-shelf'.</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>&lt;/sup&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>All content licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> <br> © Raphael Sonabend, Andreas Bender.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://www.mlsabook.com">Website</a> | <a href="https://github.com/mlsa-book/MLSA">GitHub</a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/mlmodels.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/mlmodels.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>