<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Raphael Sonabend and Andreas Bender">

<title>3&nbsp; Survival Analysis – Machine Learning in Survival Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./P1C5_eha.html" rel="next">
<link href="./P1C3_machinelearning.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-be0bf478834dcd5fa1d207f31fcfa0a3.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-1008a5a8f5e3c2ed0fd245401abc5d00.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P1C3_machinelearning.html">Survival Analysis and Machine Learning</a></li><li class="breadcrumb-item"><a href="./P1C4_survival.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Survival Analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Machine Learning in Survival Analysis</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/mlsa-book/MLSA/tree/main/book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./Machine-Learning-in-Survival-Analysis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C0_notation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Symbols and Notation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Survival Analysis and Machine Learning</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C3_machinelearning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C4_survival.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C5_eha.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C6_survtsk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Survival Task</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C8_rank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Discrimination</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C9_calib.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Calibration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C10_rules.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Scoring Rules</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C11_time.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Survival Time Measures</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Models</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C13_classical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Traditional Survival Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C14_forests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Random Forests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C15_svm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C16_boosting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Boosting Methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C17_neural.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Neural Networks</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Reduction Techniques</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C19_reductions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Reductions</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C20_competing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Competing Risks Pipelines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C21_discrete.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Discrete Time Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C22_poisson.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Connections to Poisson Regression and Processes</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C23_pseudo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Connections to Regression and Imputation</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C24_conclusions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">FAQs and Outlook</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C26_references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-distributions" id="toc-sec-distributions" class="nav-link active" data-scroll-target="#sec-distributions"><span class="header-section-number">3.1</span> Quantifying the Distribution of Event Times</a>
  <ul class="collapse">
  <li><a href="#sec-distributions-continuous" id="toc-sec-distributions-continuous" class="nav-link" data-scroll-target="#sec-distributions-continuous"><span class="header-section-number">3.1.1</span> Continuous Time</a></li>
  <li><a href="#sec-distributions-discrete" id="toc-sec-distributions-discrete" class="nav-link" data-scroll-target="#sec-distributions-discrete"><span class="header-section-number">3.1.2</span> Discrete Time</a></li>
  </ul></li>
  <li><a href="#sec-data-rc" id="toc-sec-data-rc" class="nav-link" data-scroll-target="#sec-data-rc"><span class="header-section-number">3.2</span> Single-event, right-censored data</a></li>
  <li><a href="#sec-types-of-censoring" id="toc-sec-types-of-censoring" class="nav-link" data-scroll-target="#sec-types-of-censoring"><span class="header-section-number">3.3</span> Types of Censoring</a>
  <ul class="collapse">
  <li><a href="#right-censoring" id="toc-right-censoring" class="nav-link" data-scroll-target="#right-censoring"><span class="header-section-number">3.3.1</span> Right-censoring</a></li>
  <li><a href="#left--and-interval-censoring" id="toc-left--and-interval-censoring" class="nav-link" data-scroll-target="#left--and-interval-censoring"><span class="header-section-number">3.3.2</span> Left- and Interval-censoring</a></li>
  </ul></li>
  <li><a href="#sec-truncation" id="toc-sec-truncation" class="nav-link" data-scroll-target="#sec-truncation"><span class="header-section-number">3.4</span> Censoring vs.&nbsp;Truncation</a>
  <ul class="collapse">
  <li><a href="#sec-left-truncation" id="toc-sec-left-truncation" class="nav-link" data-scroll-target="#sec-left-truncation"><span class="header-section-number">3.4.1</span> Left-truncation</a></li>
  <li><a href="#sec-right-truncation" id="toc-sec-right-truncation" class="nav-link" data-scroll-target="#sec-right-truncation"><span class="header-section-number">3.4.2</span> Right-truncation</a></li>
  </ul></li>
  <li><a href="#sec-surv-estimation" id="toc-sec-surv-estimation" class="nav-link" data-scroll-target="#sec-surv-estimation"><span class="header-section-number">3.5</span> Estimation</a>
  <ul class="collapse">
  <li><a href="#sec-surv-estimation-param" id="toc-sec-surv-estimation-param" class="nav-link" data-scroll-target="#sec-surv-estimation-param"><span class="header-section-number">3.5.1</span> Parametric estimation</a></li>
  <li><a href="#sec-surv-estimation-non-param" id="toc-sec-surv-estimation-non-param" class="nav-link" data-scroll-target="#sec-surv-estimation-non-param"><span class="header-section-number">3.5.2</span> Non-parametric estimation</a></li>
  </ul></li>
  <li><a href="#sec-censoring-distribution" id="toc-sec-censoring-distribution" class="nav-link" data-scroll-target="#sec-censoring-distribution"><span class="header-section-number">3.6</span> Estimation of the censoring distribution</a>
  <ul class="collapse">
  <li><a href="#sec-surv-km-censoring" id="toc-sec-surv-km-censoring" class="nav-link" data-scroll-target="#sec-surv-km-censoring"><span class="header-section-number">3.6.1</span> The Kaplan-Meier estimator for the censoring distribution</a></li>
  <li><a href="#sec-surv-ipcw" id="toc-sec-surv-ipcw" class="nav-link" data-scroll-target="#sec-surv-ipcw"><span class="header-section-number">3.6.2</span> Inverse probability of censoring weights (IPCW)</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">3.7</span> Conclusion</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P1C4_survival.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P1C4_survival.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P1C3_machinelearning.html">Survival Analysis and Machine Learning</a></li><li class="breadcrumb-item"><a href="./P1C4_survival.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Survival Analysis</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span id="sec-surv" class="quarto-section-identifier"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Survival Analysis</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    TODO (150-200 WORDS)
  </div>
</div>


</header>


<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Warning</span>Minor changes expected!
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This page is a work in progress and minor changes will be made over time.</strong></p>
</div>
</div>
<p><em>Survival Analysis</em> is concerned with data where the outcome is the time until an event takes place (a ‘time-to-event’). Because the collection of such data takes place in the temporal domain (it takes time to observe a duration), the event of interest is often unobservable, for example because it did not occur by the end of the data collection period. In survival analysis terminology this is referred to as <em>censoring</em>.</p>
<p>This chapter defines basic terminology and mathematical definitions in survival analysis, which are used throughout this book. Building upon this chapter, <a href="P1C5_eha.html" class="quarto-xref"><span>Chapter 4</span></a> introduces event-history analysis, which is a generalization to settings with multiple, potentially competing or recurrent events, including multi-state outcomes. Concluding this part of the book, <a href="P1C6_survtsk.html" class="quarto-xref"><span>Chapter 5</span></a> defines different prediction tasks in survival analysis.</p>
<p>While these definitions and concepts are not new to survival analysis, it is imperative they are understood to build successful models. Evaluation functions (Part II) can identify if one model is better suited than another to minimize a given objective function, however they cannot identify if the objective function itself was specified correctly, which depends on the assumptions about the data generating process. Evaluating models with the wrong objective function yields meaningless results. Hence, it is of utmost importance for machine learning practitioners to be able identify and specify the survival problem present in their data correctly to ensure models are correctly fit and evaluated.</p>
<section id="sec-distributions" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="sec-distributions"><span class="header-section-number">3.1</span> Quantifying the Distribution of Event Times</h2>
<p>This section introduces functions that can be used to fully characterize a probability distribution, particular focus is given to functions that are important in survival analysis.</p>
<p>Note that we can generally distinguish between events taking place in discrete time or continuous time. For example, consider the time a politician serves in parliament. If we consider the number of election cycles they stay in parliament, it would constitute discrete time, as time can only take values, <span class="math inline">\(1, 2, 3, \ldots\)</span>, that is <span class="math inline">\(Y\in \mathbb{N}_{&gt;0}\)</span>. On the other hand, the time an individual stays in hospital is usually determined as the difference between the admission date-time and discharge date-time, which would constitute a continuous time <span class="math inline">\(Y\in \mathbb{R}_{\geq 0}\)</span>.</p>
<p>In practice the differences are often blurred as time-measurement will naturally be discretized at some level and precision beyond some resolution is often not of interest (hospital length of stay might be interesting up to days or hours, but not minutes and seconds). Also discrete-time methods are often applied to continuous time data and vice versa. It is nevertheless important to make the distinction as it informs mathematical treatment and definition of the different quantities introduced below.</p>
<section id="sec-distributions-continuous" class="level3" data-number="3.1.1">
<h3 data-number="3.1.1" class="anchored" data-anchor-id="sec-distributions-continuous"><span class="header-section-number">3.1.1</span> Continuous Time</h3>
<p>For now, assume a continuous, positive, random variable <span class="math inline">\(Y\)</span> taking values in (t.v.i.) <span class="math inline">\(\mathbb{R}_{\geq 0}\)</span>. A standard representation of the distribution of <span class="math inline">\(Y\)</span> is given by the probability density function (pdf), <span class="math inline">\(f_Y: \mathbb{R}_{\geq 0}\rightarrow \mathbb{R}_{\geq 0}\)</span>, and cumulative distribution function (cdf), <span class="math inline">\(F_Y: \mathbb{R}_{\geq 0}\rightarrow [0,1]; (\tau) \mapsto P(Y \leq \tau)\)</span>.</p>
<p>In survival analysis, it is more common to describe the distribution of event times <span class="math inline">\(Y\)</span> via the <em>survival function</em> and <em>hazard function</em> (or <em>hazard rate</em>) rather than the pdf or cdf. The survival function is defined as <span id="eq-surv-prob"><span class="math display">\[
S_Y(\tau) = P(Y &gt; \tau) = \int^\infty_\tau f_Y(u) \ \mathrm{d}u,
\tag{3.1}\]</span></span></p>
<p>which is the probability that an event has not occurred by <span class="math inline">\(\tau \geq 0\)</span> and thus the complement of the cdf: <span class="math inline">\(S_Y(\tau) = 1-F_Y(\tau)\)</span>. By definition, <span class="math inline">\(S_Y(0)=1\)</span> and <span class="math inline">\(S(\tau)\rightarrow 0\)</span> for <span class="math inline">\(\tau \rightarrow \infty\)</span>.</p>
<p>The hazard function is given by <span id="eq-continuous-hazard"><span class="math display">\[
\begin{aligned}
h_Y(\tau) &amp;= \lim_{\mathrm{d}\tau\searrow 0}\frac{P(\tau  \leq Y &lt; \tau + \mathrm{d}\tau|Y \geq \tau)}{\mathrm{d}\tau} \\
          &amp;= \lim_{\mathrm{d}\tau\searrow 0}\frac{P(Y \in [\tau, \tau + \mathrm{d}\tau)|Y \geq \tau)}{\mathrm{d}\tau}\\
          &amp; = \frac{f_Y(\tau)}{S_Y(\tau)}
\end{aligned}
\tag{3.2}\]</span></span></p>
<p>where <span class="math inline">\(\mathrm{d}\tau\)</span> denotes a time-interval. The hazard rate is often interpreted as the instantaneous risk of observing an event at <span class="math inline">\(\tau\)</span>, given that the event has not been observed before <span class="math inline">\(\tau\)</span>. This is not a probability and <span class="math inline">\(h_Y\)</span> can be greater than one.</p>
<p>The cumulative hazard function (chf) can be derived from the hazard function by <span id="eq-cumu-hazard"><span class="math display">\[
H_Y(\tau) = \int^\tau_0 h_Y(u) \ \mathrm{d}u,
\tag{3.3}\]</span></span></p>
<p>and relates to the survival function via</p>
<p><span class="math display">\[
\begin{aligned}
H_Y(\tau) &amp; = \int^\tau_0 h_Y(u) \ \mathrm{d}u= \int^\tau_0 \frac{f_Y(u)}{S_Y(u)} \ \mathrm{d}u\\
          &amp; = \int^\tau_0 -\frac{S'_Y(u)}{S_Y(u)} \ \mathrm{d}u= -\log(S_Y(\tau))
\end{aligned}
\]</span></p>
<p>These last relationships are particularly important, as many methods estimate the hazard rate, which is then used to calculate the cumulative hazard and survival probability <span id="eq-surv-haz"><span class="math display">\[
S_Y(\tau) = \exp(-H_Y(\tau)) = \exp\left(-\int_0^\tau h_Y(u)\ \mathrm{d}u\right)
\tag{3.4}\]</span></span></p>
<p>Unless necessary to avoid confusion, subscripts are dropped from <span class="math inline">\(S_Y, h_Y\)</span> etc. going forward and instead these functions are referred to as <span class="math inline">\(S\)</span>, <span class="math inline">\(h\)</span> (and so on).</p>
<p>Usual regression techniques cannot be used to estimate these quantities as <span class="math inline">\(Y\)</span> is only partially observed, due to different types of censoring and truncation (see <a href="#sec-data-rc" class="quarto-xref"><span>Section 3.2</span></a>, <a href="#sec-types-of-censoring" class="quarto-xref"><span>Section 3.3</span></a>, <a href="#sec-truncation" class="quarto-xref"><span>Section 3.4</span></a>).</p>
</section>
<section id="sec-distributions-discrete" class="level3" data-number="3.1.2">
<h3 data-number="3.1.2" class="anchored" data-anchor-id="sec-distributions-discrete"><span class="header-section-number">3.1.2</span> Discrete Time</h3>
<p>Now let <span class="math inline">\(\tilde{Y}\)</span> be a discrete, positive random variable that represents discrete or discretized time, taking values in <span class="math inline">\(\mathbb{N}_{&gt;0}\)</span>; and <span class="math inline">\(\tau\in \mathbb{N}_{&gt;0}\)</span> some time point in discrete time.</p>
<p>The discrete-time hazard rate is defined as</p>
<p><span id="eq-discrete-hazard"><span class="math display">\[
h^d_{\tilde{Y}}(\tau) = P(\tilde{Y}=\tau|\tilde{Y} \geq \tau).
\tag{3.5}\]</span></span></p>
<p>Thus, in contrast to the continuous time hazard <a href="#eq-continuous-hazard" class="quarto-xref"><span>3.2</span></a>, the discrete time hazard is an actual (conditional) probability, rather than a rate and therefore might be easier to interpret.</p>
<p>The cumulative discrete time hazard is given by</p>
<p><span id="eq-discrete-cumu-hazard"><span class="math display">\[
H^d_{\tilde{Y}}(\tau) = \sum_{k=1}^{\tau}h^d_{\tilde{Y}}(k)
\tag{3.6}\]</span></span></p>
<p>We also define the complement of <a href="#eq-discrete-hazard" class="quarto-xref"><span>3.5</span></a>, the probability to survive beyond <span class="math inline">\(\tau\)</span> given survival until <span class="math inline">\(\tau\)</span>, as <span class="math display">\[
s^d_{\tilde{Y}}(\tau) := 1 - h^d_{\tilde{Y}}(\tau) = P(\tilde{Y}&gt; \tau|\tilde{Y} \geq \tau).
\]</span></p>
<p>It follows that the unconditional probability to survive beyond time point <span class="math inline">\(\tau\)</span> is given by</p>
<!--  -->
<p><span id="eq-discrete-surv-prob"><span class="math display">\[
S^d_{\tilde{Y}} = P(\tilde{Y} &gt; \tau) = \prod_{k\leq\tau} s^d_{\tilde{Y}}(\tau) = \prod_{k\leq\tau}(1-h^d_{\tilde{Y}}(\tau)),
\tag{3.7}\]</span></span> <!--  --> and the unconditional probability for an event at time <span class="math inline">\(\tau\)</span> is <!--  --> <span id="eq-discrete-prob"><span class="math display">\[
P(\tilde{Y}=\tau) = S^d_{\tilde{Y}}(\tau-1) h^d_{\tilde{Y}}(\tau).
\tag{3.8}\]</span></span> <!--  --> When applied to continuous time <span class="math inline">\(Y\)</span>, the follow-up is divided in <span class="math inline">\(J\)</span> disjunct intervals <span class="math inline">\((a_0,a_{1}],\ldots,(a_{j-1},a_j]\ldots, (a_{J-1},a_{J}],j=1,\ldots,J\)</span> such that <!--  --> <span class="math display">\[
Y \in (a_{j-1}, a_j]\Leftrightarrow \tilde{Y} = j
\]</span> <!--  --> Thus, <!--  --> <span class="math display">\[
h^d_{\tilde{Y}}(j) = P(Y \in (a_{j-1},a_j]|Y &gt; a_{j-1})
\]</span></p>
<!--  -->
<p>and <!--  --> <span class="math display">\[
S^d_{\tilde{Y}}(j) = P(Y &gt; a_{j}) = S_Y(a_{j})
\]</span></p>
<!--  -->
<p>For more details on discrete time-to-event analysis consider <span class="citation" data-cites="Tutz2016">(<a href="P5C26_references.html#ref-Tutz2016" role="doc-biblioref">Tutz and Schmid 2016</a>)</span>.</p>
</section>
</section>
<section id="sec-data-rc" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="sec-data-rc"><span class="header-section-number">3.2</span> Single-event, right-censored data</h2>
<p>The complexity of Survival Analysis compared to other fields arises from the fact that the outcome of interest is often only observed partially. In particular, the time-to-event is often unknown at the end of the observation period, as the event has not occurred yet.</p>
<p>Let,</p>
<ul>
<li><span class="math inline">\(X\)</span> taking values in <span class="math inline">\(\mathbb{R}^p\)</span> be the generative random variable representing the data <em>features</em>/<em>covariates</em>/<em>independent variables</em>.</li>
<li><span class="math inline">\(Y\)</span> taking values in <span class="math inline">\(\mathbb{R}_{\geq 0}\)</span> be the (partially unobservable) <em>true survival time</em>.</li>
<li><span class="math inline">\(C\)</span> taking values in <span class="math inline">\(\mathbb{R}_{\geq 0}\)</span> be the (partially unobservable) <em>true censoring time</em>.</li>
</ul>
<p>In the presence of censoring <span class="math inline">\(C\)</span>, it is impossible to fully observe the true outcome of interest, <span class="math inline">\(Y\)</span>. Instead, the observable variables are defined by</p>
<ul>
<li><span class="math inline">\(T := \min\{Y,C\}\)</span>, the <em>outcome time</em> (realizations are referred to as the <em>observed outcome time</em>); and</li>
<li><span class="math inline">\(\Delta := \mathbb{I}(Y = T) = \mathbb{I}(Y \leq C)\)</span>, the <em>event indicator</em> (also known as the <em>censoring</em> or <em>status</em> indicator).</li>
</ul>
<p>Together <span class="math inline">\((T,\Delta)\)</span> is referred to as the <em>survival outcome</em> or <em>survival tuple</em> and they form the dependent variables. The survival outcome provides a concise mechanism for representing the outcome time and indicating which outcome (event or censoring) took place.</p>
<p>A <em>survival dataset</em> is a <span class="math inline">\(n \times p\)</span>, real-valued matrix defined by <span class="math inline">\(\mathcal{D}= ((\mathbf{x}_1, \ t_1, \ \delta_1) \cdots (\mathbf{x}_n,t_n,\delta_n))^\top\)</span>, where <span class="math inline">\((t_i,\delta_i)\)</span> are realizations of the respective random variables <span class="math inline">\((T_i, \Delta_i)\)</span> and <span class="math inline">\(\mathbf{x}_i\)</span> is a <span class="math inline">\(p\)</span>-dimensional vector, <span class="math inline">\(\mathbf{x}_i = (x_{i;1} \ x_{i;2} \cdots x_{i;p})^\top\)</span> of features.</p>
<p>In this book, we will often refer to the unique, ordered event times. For example, given survival data, <span class="math inline">\((t_1=6, \delta_1=0), (t_2=1, \delta_2=1), (t_3=10,\delta_3=1), (t_4=5,\delta_4=0), (t_5 =10,\delta_5=1)\)</span> the ordered unique event times are <span class="math inline">\((t_{(1)},  t_{(2)}) = (1,10)\)</span> as the censored observations are removed, as are duplicated times.</p>
<p>Let <span class="math inline">\(m\leq n\)</span> be the number of unique, observed event times. We define the <strong>unique, ordered event times</strong> simply as</p>
<p><span id="eq-unique-ordered-event-times"><span class="math display">\[
t_{(1)}&lt;\cdots&lt;t_{(m)}, \qquad m\le n.
\tag{3.9}\]</span></span></p>
<p>A common short-hand in survival analysis is to assume the subscript <span class="math inline">\((k)\)</span> is an index referring to the observation who experienced the event at time <span class="math inline">\(t_{(k)}\)</span>. For example, <span class="math inline">\((\mathbf{x}_{(k)}, t_{(k)}, \delta_{(k)})\)</span> are the features and outcome for the observation that experienced the event at <span class="math inline">\(t_{(k)}\)</span>, the <span class="math inline">\(k\)</span>th ordered event time.</p>
<p>Finally, the following quantities are used frequently throughout this book and survival analysis literature more generally. Let <span class="math inline">\((t_i, \delta_i) \stackrel{i.i.d.}\sim(T,\Delta), i = 1,...,n\)</span>, be observed survival outcomes.</p>
<p>The <strong>risk set at <span class="math inline">\(\tau\)</span></strong>, is the index-set of observation units at risk for the event just before <span class="math inline">\(\tau\)</span></p>
<p><span id="eq-risk-set"><span class="math display">\[
\mathcal{R}_\tau := \{i: t_i \geq \tau\}
\tag{3.10}\]</span></span></p>
<p>where <span class="math inline">\(i\)</span> is the index of an observation in the data. For right-censored data, <span class="math inline">\(\mathcal{R}_0 = \{1,\ldots,n\}\)</span> and <span class="math inline">\(\mathcal{R}_{\tau} \subseteq \mathcal{R}_{\tau'}, \forall \tau &gt; \tau'\)</span>. Note that in a continuous setting, ‘just before’ refers to an infinitesimally smaller time than <span class="math inline">\(\tau\)</span>, in practice, as this is unobservable the risk set is defined at <span class="math inline">\(\tau\)</span>, hence an observation may both be at risk, and experience an event (or be censored) at <span class="math inline">\(\tau\)</span>.</p>
<p>The <strong>number of observations at risk at <span class="math inline">\(\tau\)</span></strong> is the cardinality of the risk set at <span class="math inline">\(\tau\)</span>, <!--  --> <span class="math display">\[
n_\tau := \sum_i \mathbb{I}(t_i \geq \tau) = |\mathcal{R}_\tau|
\]</span> <!--  --> Finally, the <strong>number of events at <span class="math inline">\(\tau\)</span></strong> is defined by, <!--  --> <span class="math display">\[
d_\tau := \sum_i \mathbb{I}(t_i = \tau, \delta_i = 1)
\]</span> <!--  --> For truly continuous variables, one might expect only one event to occur at each observed event time: <span class="math inline">\(d_{t_{(k)}} = 1,\forall k\)</span>. In practice, ties are often observed due to finite measurement precision, such that <span class="math inline">\(d_{\tau} &gt; 1\)</span> occurs frequently in real-world datasets.</p>
<p>The quantities <span class="math inline">\(\mathcal{R}_\tau\)</span>, <span class="math inline">\(n_\tau\)</span>, and <span class="math inline">\(d_\tau\)</span> underlie many models and measures in survival analysis. Several non-parametric and semi-parametric methods (<a href="P3C13_classical.html" class="quarto-xref"><span>Chapter 10</span></a>) like the Kaplan-Meier estimator (see <a href="#sec-surv-km" class="quarto-xref"><span>Section 3.5.2.1</span></a>) are based on the ratio <span class="math inline">\(d_\tau / n_\tau\)</span>.</p>
<p><a href="#tbl-surv-data-tumor" class="quarto-xref">Table&nbsp;<span>3.1</span></a> exemplifies an observed survival dataset, a subset of the <code>tumor</code> data <span class="citation" data-cites="pkgpammtools">(<a href="P5C26_references.html#ref-pkgpammtools" role="doc-biblioref">Bender and Scheipl 2018</a>)</span>, which contains the time until death in days after operation (<span class="math inline">\(\delta_i=1\)</span> if death occurred at the outcome time <span class="math inline">\(t_i\)</span> and <span class="math inline">\(\delta_i = 0\)</span> otherwise).</p>
<p>In this example, the above quantities would be:</p>
<ul>
<li><span class="math inline">\(\mathcal{R}_{\tau = 1217} = \{1, 3, 5\}\)</span> (these subjects’ outcome times are greater or equal to <span class="math inline">\(\tau = 1217\)</span> so they are at risk for the event at this time)</li>
<li><span class="math inline">\(n_{\tau = 1217} = |\mathcal{R}_{1217}| = 3\)</span></li>
<li><span class="math inline">\(d_{\tau = 1217} = 1\)</span>: As only <span class="math inline">\(i = 1\)</span> experienced the event (and not censoring) at this time.</li>
</ul>
<div id="tbl-surv-data-tumor" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-surv-data-tumor-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3.1: Subset of the <code>tumor</code> <span class="citation" data-cites="pkgpammtools">(<a href="P5C26_references.html#ref-pkgpammtools" role="doc-biblioref">Bender and Scheipl 2018</a>)</span> time-to-event dataset. Rows are individual observations (ID), <span class="math inline">\(\mathbf{x}_{;j}\)</span> columns are features, <span class="math inline">\(t\)</span> is observed time-to-event, <span class="math inline">\(\delta\)</span> is the event indicator.
</figcaption>
<div aria-describedby="tbl-surv-data-tumor-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<colgroup>
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 17%">
<col style="width: 34%">
<col style="width: 12%">
<col style="width: 17%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;"><strong>id</strong> (<span class="math inline">\(i\)</span>)</th>
<th style="text-align: right;"><strong>age</strong> <span class="math inline">\((\mathbf{x}_{;1})\)</span></th>
<th style="text-align: left;"><strong>sex</strong> <span class="math inline">\((\mathbf{x}_{;2})\)</span></th>
<th style="text-align: left;"><strong>complications</strong> <span class="math inline">\((\mathbf{x}_{;3})\)</span></th>
<th style="text-align: right;"><strong>days</strong> (<span class="math inline">\(\mathbf{t}\)</span>)</th>
<th style="text-align: right;"><strong>status</strong> (<span class="math inline">\(\boldsymbol{\delta}\)</span>)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">71</td>
<td style="text-align: left;">female</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">1217</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">70</td>
<td style="text-align: left;">male</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">519</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">67</td>
<td style="text-align: left;">female</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">2414</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: right;">58</td>
<td style="text-align: left;">male</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">397</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: right;">39</td>
<td style="text-align: left;">female</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1217</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: right;">6</td>
<td style="text-align: right;">59</td>
<td style="text-align: left;">female</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">268</td>
<td style="text-align: right;">1</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</section>
<section id="sec-types-of-censoring" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="sec-types-of-censoring"><span class="header-section-number">3.3</span> Types of Censoring</h2>
<p>Three types of censoring are commonly defined in survival analysis: right-censoring, left-censoring, and interval-censoring. The latter can be viewed as the most general case. Multiple types of censoring and/or truncation (<a href="#sec-truncation" class="quarto-xref"><span>Section 3.4</span></a>) can occur in any given data set and it is vital to identify which types are present in order to correctly select and specify models and measures for the data.</p>
<section id="right-censoring" class="level3" data-number="3.3.1">
<h3 data-number="3.3.1" class="anchored" data-anchor-id="right-censoring"><span class="header-section-number">3.3.1</span> Right-censoring</h3>
<p>Right-censoring is the most commonly assumed form of censoring in survival data. It occurs when the event of interest was not experienced during the observation period, which may happen because it was no longer observable (for example, due to withdrawal from the study) or because the event did not happen until study end. The exact event time is unknown but it <em>is</em> known that the event is after the observed censoring time, hence <em>right</em>-censoring (imagine a timeline from left to right as in <a href="#fig-survset-censor" class="quarto-xref">Figure&nbsp;<span>3.1</span></a>).</p>
<!-- Formally let $[\tau_l, \tau_u]$ be the study period for some, $\tau_l,\tau_u \in \NNReals$.
Then right-censoring occurs when either $Y > \tau_u$, or when $Y \in [\tau_l,\tau_u]$ and $C \leq Y$.
In the first case $T = C = \tau_u$ and censoring is due to the true time-to-event being unknown as the observation period has finished.
In the latter case, a separate censoring event, such as drop-out is observed. -->
<div id="fig-survset-censor" class="quarto-float quarto-figure quarto-figure-center anchored" alt="TODO">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-survset-censor-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survival/censoring.png" class="img-fluid figure-img" alt="TODO">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-survset-censor-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.1: Dead and censored subjects (y-axis) over time (x-axis). Black diamonds indicate true death times and white circles indicate censoring times. Diamonds at the end of dashed lines are hypothetical (unknown in reality). Vertical line is the study end time. Subjects 1 and 2 die in the study time. Subject 3 is censored in the study and (unknown) dies within the study time. Subject 4 is censored in the study and (unknown) dies after the study. Subject 5 is censored at study end and (unknown) dies after the end of the study.
</figcaption>
</figure>
</div>
<!-- FIXME: ADD REF TO THESIS IF WE KEEP THIS FIGURE -->
<p>Right-censoring can be further divided into Type-I, Type-II and random censoring. Type-I, or <em>administrative</em>, censoring occurs at the fixed, pre-defined end of an observation period <span class="math inline">\(\tau_u\)</span>, in which case the outcome is given by <span class="math inline">\((T_i = \min(Y_i, \tau_u), \Delta_i = \mathbb{I}(Y_i \leq \tau_u))\)</span>. Censored observations are therefore represented as <span class="math inline">\((\tau_u, 0)\)</span>. Type-II censoring also occurs when the observation period ends. However, in this case the study ends when a pre-defined number of subjects experienced the event of interest and hence <span class="math inline">\(\tau_u\)</span> is random.</p>
<p>Random censoring occurs when censoring times <em>randomly</em> follow an unknown distribution and one observes <span class="math inline">\((T_i=\min(Y_i,C_i), \Delta_i = \mathbb{I}(Y_i\leq C_i))\)</span>. Different types of right-censoring can, and sometimes do, co-occur in any given data set.</p>
<p>In practice, these different types of right-censoring are usually handled the same during modeling and evaluation and so this book refers to ‘right-censoring’ generally, which could occur from a combination of the above types.</p>
</section>
<section id="left--and-interval-censoring" class="level3" data-number="3.3.2">
<h3 data-number="3.3.2" class="anchored" data-anchor-id="left--and-interval-censoring"><span class="header-section-number">3.3.2</span> Left- and Interval-censoring</h3>
<p>Left-censoring occurs when the event is known to have happened at some unknown time before observation time, interval-censoring occurs when the event is known to have happened within some time span, but not the exact time.</p>
<p>Consider a survey about phone use where participants are asked: “How old were you when you used a smart phone for the first time?”. The possible answers are:</p>
<ul>
<li>exact age of first use</li>
<li>didn’t use a smart phone yet</li>
<li>did or does use a phone but doesn’t remember age of first time use</li>
<li>did or does use a phone, remembers a specific age range</li>
</ul>
<p>The first case represents an exactly observed event time, the second case the familiar right-censoring, as the event may occur later in life, but it is unknown when. The third case is referred to as left-censoring, we know the event occurred before the interview, but don’t know when. The fourth case is an example of interval-censoring, as we know the event occurred within some age span, but not the exact age.</p>
<p>Interval- and left-censoring often occur in medical contexts. For example, some guidelines suggest annual screenings for skin cancer (starting from a certain age). However, the initial age at which individuals do screenings and regularity of check ups varies widely. If cancer was detected between two screenings, the observation is interval-censored. If cancer is detected at first screening, the observation is left-censored (unless the ‘age’ of the cancer can be narrowed down based on size and other characteristics, in which case it would become interval-censored). Also of note, particularly in a medical context, is that interval censoring may be safely ignored depending on the time scale being modelled. Say a patient is screened in February and March 2025 and cancer is detected in the second screening, then they are technically interval censored as (February 2025, March 2025). However, if this data is modelled at an annual scale, then their outcome is known to be [2025] and they are not considered censored.</p>
</section>
<section id="censoring-notation" class="level3 unnumbered unlisted">
<h3 class="unnumbered unlisted anchored" data-anchor-id="censoring-notation">Censoring Notation</h3>
<p>In the presence of left- or interval-censoring the usual representation of survival outcomes as <span class="math inline">\((t_i, \delta_i)\)</span> is not sufficient to denote the different types of observations in the data set. Instead, we represent the data as intervals in which the event occurs. Formally, let <span class="math inline">\(Y_i\)</span> the random variable for time until the event of interest and <span class="math inline">\(L_i, R_i\)</span> random variables that define an interval <span class="math inline">\((L_i,R_i]\)</span> with realizations <span class="math inline">\(l_i, r_i\)</span>. Let further <span class="math inline">\(t_i\)</span> the time of observation (for example age at interview in the phone use example) or last-follow up time for subject <span class="math inline">\(i\)</span>. Then the event time of subject <span class="math inline">\(i\)</span> is</p>
<ul>
<li>left-censored if <span class="math inline">\(Y_i \in (L_i = 0, R_i = t_i]\)</span>;</li>
<li>right-censored at <span class="math inline">\(t_i\)</span> if <span class="math inline">\(Y_i \in (L_i = t_i, R_i = \infty)\)</span>;</li>
<li>interval-censored <span class="math inline">\(Y_i \in (L_i = l_i, R_i = r_i]\)</span>, <span class="math inline">\(l_i &lt; r_i \leq t_i\)</span></li>
<li>exactly observed if <span class="math inline">\(Y_i \in (L_i = t_i, R_i = t_i]\)</span></li>
</ul>
<p>In the cancer screening example from above it holds that <span class="math inline">\(r_i=t_i\)</span> and <span class="math inline">\(l_i\)</span> the last check-up before <span class="math inline">\(r_i\)</span>. In the phone use example, the participants might specify any age range between <span class="math inline">\(0\)</span> and <span class="math inline">\(t_i\)</span>.</p>
<p>To make this more concrete, consider phone use example data, where <span class="math inline">\(t_i\)</span> is the age at interview. In practice, such data is often stored by creating two variables representing the left (<span class="math inline">\(l_i\)</span>) and right border (<span class="math inline">\(r_i\)</span>) of the respective intervals (<span class="math inline">\(t_i\)</span> is not really needed here to define the outcome, but included for illustration).</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>id</th>
<th><span class="math inline">\(t_i\)</span></th>
<th><span class="math inline">\(l_i\)</span></th>
<th><span class="math inline">\(r_i\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>13</td>
<td>13</td>
<td><span class="math inline">\(\infty\)</span></td>
</tr>
<tr class="even">
<td>2</td>
<td>17</td>
<td>15</td>
<td>15</td>
</tr>
<tr class="odd">
<td>3</td>
<td>16</td>
<td>14</td>
<td>16</td>
</tr>
<tr class="even">
<td>4</td>
<td>16</td>
<td>13</td>
<td>15</td>
</tr>
<tr class="odd">
<td>5</td>
<td>18</td>
<td>0</td>
<td>18</td>
</tr>
</tbody>
</table>
<p>Here, the first subject is right-censored at 13 years (<span class="math inline">\(l_i=t_i = 13, r_i = \infty\)</span>), the second subject remembered exactly (<span class="math inline">\(l_i=r_i=15 &lt; t_i = 17\)</span>), the third subject remembers that it was after 14, but not exact age (<span class="math inline">\(l_i = 14 &lt; r_i = 16=t_i\)</span>), fourth subject remembers use after 13 years and latest at 15 years of age (<span class="math inline">\(l_i = 13 &lt; r_i = 15 &lt; t_i\)</span>), and the fifth subject uses a smart phone currently at age 18, but cannot specify further (<span class="math inline">\(l_i = 0 &lt; r_i = 18 = t_i\)</span>).</p>
<p>From the example above, it is clear, that right- and left-censoring are special cases of interval-censoring. However, if only right- or left-censoring is present, the likelihood and estimation simplifies (see <a href="#sec-surv-estimation" class="quarto-xref"><span>Section 3.5</span></a>). Also note that in case of left- and interval-censoring the event is known to have occurred, while for right-censoring the event didn’t occur during time under observation. For left- and right-censoring one might be tempted to consider it an event <span class="math inline">\(\delta_i = 1\)</span> without exact time, while right-censoring would be consider a non-event <span class="math inline">\(\delta_i = 0\)</span>. However, technically it is assumed that the event will always occur, if we wait long enough (for right-censored data in the interval <span class="math inline">\((t_i, \infty]\)</span>). Censoring therefore means having imprecise information about the time of event rather than information about the event occurring or not occurring.</p>
</section>
</section>
<section id="dependent-vs.-informative-censoring" class="level2 unnumbered unlisted">
<h2 class="unnumbered unlisted anchored" data-anchor-id="dependent-vs.-informative-censoring">Dependent vs.&nbsp;Informative Censoring</h2>
<p>Censoring may be defined as <em>uninformative</em> if <span class="math inline">\(Y \perp \!\!\! \perp C\)</span> and <em>informative</em> otherwise. However, these definitions can be misleading as the term ‘uninformative’ could imply that <span class="math inline">\(C\)</span> is independent of both <span class="math inline">\(X\)</span> <em>and</em> <span class="math inline">\(Y\)</span>, and not just <span class="math inline">\(Y\)</span>. To avoid misinterpretation, the following definitions are used in this book:</p>
<ul>
<li>If <span class="math inline">\(C \perp \!\!\! \perp X\)</span>, censoring is <em>feature-independent</em>, otherwise censoring is <em>feature-dependent</em>.</li>
<li>If <span class="math inline">\(C \perp \!\!\! \perp Y\)</span>, censoring is <em>event-independent</em>, otherwise censoring is <em>event-dependent</em>.</li>
<li>If <span class="math inline">\((C \perp \!\!\! \perp Y) | X\)</span>, censoring is conditionally independent of the event given covariates, or <em>conditionally event-independent</em>.</li>
<li>If <span class="math inline">\(C \perp \!\!\! \perp(X,Y)\)</span>, censoring is <em>uninformative</em>, otherwise censoring is <em>informative</em>.</li>
</ul>
<p>Uninformative censoring can generally be well-handled by models as the true underlying distribution of survival times is not affected by censoring. In fact, in this case one could even use regression models after removing censored observations (if they do not form a high proportion of the data).</p>
<p>In reality, censoring is rarely non-informative as reasons for drop-out or missingness in outcomes tend to be related to the study of interest. Event-dependent censoring is a tricky case that, if not handled appropriately (by a competing-risks framework), can easily lead to poor model development. Imagine a study is interested in predicting the time between relapses of stroke but a patient suffers a brain aneurysm due to some separate neurological condition. There is a high possibility that a stroke may have occurred if the aneurysm had not. A survival model is unlikely to distinguish the censoring event (aneurysm) from the event of interest (stroke) and will confuse predictions.</p>
<p>In practice, the majority of models and measures assume that censoring is conditionally event-independent and hence censoring patterns can be predicted/estimated based on the covariates. For example, if studying the survival time of ill pregnant patients in hospital, then dropping out of the study due to pregnancy is clearly dependent on how many weeks pregnant the patient is when the study starts (for the sake of argument assume no early/late pregnancy due to illness).</p>
</section>
<section id="sec-truncation" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="sec-truncation"><span class="header-section-number">3.4</span> Censoring vs.&nbsp;Truncation</h2>
<p>A common confusion is to conflate censoring and truncation, which is problematic as the methods to handle them differ substantially. Outside of time-to-event settings, truncation usually refers to truncating (or removing) an entire subject from a dataset. As discussed in <a href="P0C1_intro.html" class="quarto-xref"><span>Chapter 1</span></a>, truncation in survival analysis refers to partially truncating a period of time and is quite common in the more general event history setting (<a href="P1C5_eha.html" class="quarto-xref"><span>Chapter 4</span></a>).</p>
<p>While censored observations have incomplete information about the time-to-event, they are still part of the data set. Whereas truncation leads to observations not entering the data set (at least not at time 0). This will usually introduce bias that needs to be accounted for.</p>
<section id="sec-left-truncation" class="level3" data-number="3.4.1">
<h3 data-number="3.4.1" class="anchored" data-anchor-id="sec-left-truncation"><span class="header-section-number">3.4.1</span> Left-truncation</h3>
<p>Left-truncation often occurs when inclusion into a study is conditional on the occurrence of another event. Left-truncation plays an important role when modeling recurrent events or multi-state data (<a href="P1C5_eha.html" class="quarto-xref"><span>Chapter 4</span></a>), thus the concept will be introduced in more detail.</p>
<p>By example, consider a study from the 18th century <span class="citation" data-cites="brostrom.influence.1987">(<a href="P5C26_references.html#ref-brostrom.influence.1987" role="doc-biblioref">Broström 1987</a>)</span>, when childhood and maternal mortality were relatively high. The goal of the study was to establish the effect of a mother’s death on the survival of the infant. Since each death was reported to the authorities, an infant was added to the study if and when their mother died. To create a matched cohort, two other infants, whose mothers were alive, were matched into the study based on their age and other relevant features. Thus, groups of three infants within the study had identical features except for the status of the mother (alive or dead). Because of the study design, infants who died before their mothers could never enter into the study. A mother’s death is thus referred to as left-truncation event and the infant’s age at time of inclusion into the study is referred to as left-truncation time.</p>
<p>More formally, let <span class="math inline">\(t^L_i\)</span> the subject-specific left-truncation time. Then we only observe subjects with <span class="math inline">\(y_i &gt; t^L_i\)</span> and subjects with <span class="math inline">\(y_i &lt; t^L_i\)</span> never enter the data.</p>
<p>This is illustrated in <a href="#fig-left-truncation" class="quarto-xref">Figure&nbsp;<span>3.2</span></a>. Continuing with the example above, say Infant 1 dies at <span class="math inline">\(t_1\)</span>, while the mother dies at some later time point <span class="math inline">\(t_1^L\)</span>, therefore infant 1 never enters the study. The mother of Infant 2 dies at <span class="math inline">\(t_2^L\)</span>, at which point the infant is included in the study and experiences an event at <span class="math inline">\(t_2\)</span>. Finally, say Infant 3 enters the study at <span class="math inline">\(t_3^L\)</span> and is censored at 365 days when the study ends.</p>
<div id="fig-left-truncation" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Schematic illustration of left-truncation.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-left-truncation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survival/left-truncation.png" class="img-fluid figure-img" alt="Schematic illustration of left-truncation.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-left-truncation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.2: Illustration of left-truncation data. Subjects 2 and 3 enter the data set as the study entry condition (mother’s death) occurs before the event of interest occurs after the left-truncation time (left-truncation time shorter than event time). Subject 1 on the other hand never enters the data as the event occurs before the study entry condition (left-truncation time longer than event time).
</figcaption>
</figure>
</div>
<p>In this example, left-truncation biases the sample towards healthier or more robust infants, as frail infants on die earlier and thus on average before their mothers. This would bias the estimates if not properly taken into account (see <a href="#sec-non-param-lt" class="quarto-xref"><span>Section 3.5.2.3</span></a>).</p>
</section>
<section id="sec-right-truncation" class="level3" data-number="3.4.2">
<h3 data-number="3.4.2" class="anchored" data-anchor-id="sec-right-truncation"><span class="header-section-number">3.4.2</span> Right-truncation</h3>
<p>Right-truncation often occurs in retrospective sampling based on registry data, when data is queried for cases reported by a certain cut-off time (see for example <span class="citation" data-cites="vakulenko-lagun.inverse.2020">Vakulenko-Lagun, Mandel, and Betensky (<a href="P5C26_references.html#ref-vakulenko-lagun.inverse.2020" role="doc-biblioref">2020</a>)</span>). A common example is the estimation of the incubation period of an infectious disease, which is the time from infection to the disease onset. Only known, symptomatic (and/or tested) cases are entered into the database. At a time <span class="math inline">\(\tau\)</span>, one can only observe the subset of the infected population that has already experienced the disease, and not the population that is still incubating the disease, hence biasing the data to shorter incubation periods.</p>
<p>Formally, let <span class="math inline">\(t_i^r\)</span> be the right-truncation time (here time from infection until the time at which the database is queried), then subjects only enter the data set when <span class="math inline">\(t_i &lt; t_i^r\)</span>. This is illustrated in <a href="#fig-right-truncation" class="quarto-xref">Figure&nbsp;<span>3.3</span></a> using three subjects. All three subjects were infected during the observation period, however, the right-truncation time <span class="math inline">\(t_2^r\)</span> of subject 2 is shorter than the incubation period <span class="math inline">\(t_2\)</span> for this subject, thus at the time of querying the data base, this subject will not be included in the sample, as <span class="math inline">\(t_2 &gt; t_2^r\)</span>.</p>
<p>Note the difference to right-censoring. If subject 2 was right-censored, the subject would be in our sample and the time of infection would be known - the time of disease onset would be censored. In case of right-truncation on the other hand, the subject is not included in the sample at time of data extraction, as subjects are only included in the registry after disease onset. Overall this leads to a bias towards shorter incubation times and potentially feature values that lead to shorter incubation times.</p>
<!-- FIXME: When we remake diagram below should be lowercase 't' -->
<div id="fig-right-truncation" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Schematic illustration of right-truncation.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-right-truncation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survival/right-truncation.png" class="img-fluid figure-img" alt="Schematic illustration of right-truncation.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-right-truncation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.3: Illustration of right-truncation based on registry data. For subjects 1 and 3 the right-truncation time is longer than the incubation period, therefore they are included in the sample when the registry is queried. For subject 2 on the other hand the right-truncation time is shorter, therefore it’s excluded from the sample.
</figcaption>
</figure>
</div>
<p>As for left-truncation, ignoring right-truncation will lead to biased estimations. While for left-truncated data simple adjustments of the risk set work for non- and semi-parametric methods like the Kaplan-Meier and Cox-type estimators, this is not the case for right-truncated data. However, parametric methods can be employed (see <a href="#sec-surv-estimation" class="quarto-xref"><span>Section 3.5</span></a>) and generalised product limit estimators for right-truncated data exist <span class="citation" data-cites="akritasGeneralizedProductlimitEstimator2005">(<a href="P5C26_references.html#ref-akritasGeneralizedProductlimitEstimator2005" role="doc-biblioref">Akritas and LaValley 2005</a>)</span>.</p>
</section>
</section>
<section id="sec-surv-estimation" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="sec-surv-estimation"><span class="header-section-number">3.5</span> Estimation</h2>
<p>While details about estimation will be given later, when different models are introduced in Part III and Part IV of the book, it is worthwhile discussing some general concepts here, namely parametric and non-parametric approaches.</p>
<section id="sec-surv-estimation-param" class="level3" data-number="3.5.1">
<h3 data-number="3.5.1" class="anchored" data-anchor-id="sec-surv-estimation-param"><span class="header-section-number">3.5.1</span> Parametric estimation</h3>
<p>Consider for now uncensored data <span class="math inline">\((t_i, \delta_i=1), i=1,\ldots,n\)</span>. A standard approach would be to assume a suitable distribution for the event times <span class="math inline">\(t_i \stackrel{iid}{\sim} F_Y(\boldsymbol{\theta}), \boldsymbol{\theta}=(\theta_1,\theta_2,\ldots)^\top\)</span>, and define the likelihood of the data as <!--  --> <span id="eq-likelihood"><span class="math display">\[
\mathcal{L}(\boldsymbol{\theta}) = \prod_{i=1}^{n}f_Y(t_i|\boldsymbol{\theta})
\tag{3.11}\]</span></span></p>
<!--  -->
<p>where <span class="math inline">\(f_Y\)</span> is the pdf of <span class="math inline">\(F_Y\)</span>.</p>
<p>The model parameters can then be obtained by maximizing the likelihood such that</p>
<p><span class="math display">\[
\hat{\boldsymbol{\theta}} = \mathop{\mathrm{arg\,max}}_{\boldsymbol{\theta}} \mathcal{L}(\boldsymbol{\theta})
\]</span></p>
<p>However, in the presence of censoring <a href="#eq-likelihood" class="quarto-xref"><span>3.11</span></a> is incorrect, as the exact event time is only known for some subjects. For example, for right-censored data (<span class="math inline">\(\delta_i = 0\)</span>) we only know that the event occurred after observed censoring time <span class="math inline">\(t_i\)</span>. Thus the likelihood contribution for such data points is <span class="math inline">\(P(Y_i &gt; t_i) = S_Y(t_i)\)</span>, whereas for observed event times (<span class="math inline">\(\delta_i=0\)</span>) the likelihood contribution is <span class="math inline">\(f_Y(t_i)\)</span> as before.</p>
<p>Let now <span class="math inline">\(\mathcal{O}= \{i:\delta_i = 1\}\)</span> the index set of observed event times and <span class="math inline">\(\mathcal{RC} = \{i:\delta_i = 0\}\)</span> the index set of censored observations. The likelihood of this data can be written as</p>
<p><span id="eq-right-censoring-likelihood"><span class="math display">\[
\begin{aligned}
\mathcal{L}(\boldsymbol{\theta})
  &amp; \propto \prod_{i \in \mathcal{O}}f_Y(t_i|\boldsymbol{\theta})\prod_{i \in \mathcal{RC}}S_Y(t_i|\boldsymbol{\theta})\nonumber\\
  &amp; = \prod_{i=1}^n f_y(t_i|\boldsymbol{\theta})^{\delta_i}S_Y(t_i|\boldsymbol{\theta})^{1-\delta_i}\\
  &amp; = \prod_{i=1}^n \frac{f_y(t_i|\boldsymbol{\theta})^{\delta_i}}{S_Y(t_i|\boldsymbol{\theta})^{\delta_i}}S_Y(t_i|\boldsymbol{\theta})\\
  &amp; = \prod_{i=1}^n h_Y(t_i|\boldsymbol{\theta})^{\delta_i}S_Y(t_i|\boldsymbol{\theta})\nonumber,
\end{aligned}
\tag{3.12}\]</span></span></p>
<p>where the last equality follows from <a href="#eq-continuous-hazard" class="quarto-xref"><span>3.2</span></a>.</p>
<p>Similar adjustments to the likelihood can be made for other types of censoring and in the presence of truncation. Following <span class="citation" data-cites="Klein2003">Klein and Moeschberger (<a href="P5C26_references.html#ref-Klein2003" role="doc-biblioref">2003</a>)</span>, we can define the individual likelihood contributions for the different types of censoring as</p>
<ul>
<li>observed event at <span class="math inline">\(t_i\)</span>: <span class="math inline">\(f_Y(t_i|\boldsymbol{\theta})\)</span></li>
<li>right-censoring: <span class="math inline">\(P(Y_i &gt; t_i|\boldsymbol{\theta}) = S_Y(t_i|\boldsymbol{\theta})\)</span></li>
<li>left-censoring: <span class="math inline">\(P(Y_i &lt; t_i|\boldsymbol{\theta}) = F_Y(t_i|\boldsymbol{\theta}) = 1 - S_Y(t_i|\boldsymbol{\theta})\)</span></li>
<li>interval-censoring: <span class="math inline">\(P(l_i &lt; Y_i \leq r_i|\boldsymbol{\theta}) = S_Y(l_i|\boldsymbol{\theta}) - S_Y(r_i|\boldsymbol{\theta})\)</span></li>
</ul>
<p>Depending on which of the above contributions occur in the data set, we can now construct our likelihood accordingly. Let <span class="math inline">\(\mathcal{O}, \mathcal{RC}, \mathcal{LC}, \mathcal{IC}\)</span> non-overlapping subsets of the observed data <span class="math inline">\(\mathcal{D}\)</span> for subjects with observed event times, right-censoring, left-censoring and interval-censoring, respectively. Assuming independence between observations and in absence of truncation, the likelihood for the observed data can be defined as</p>
<p><span id="eq-objective-function"><span class="math display">\[
\mathcal{L}(\boldsymbol{\theta}) \propto \prod_{i \in \mathcal{O}}f_Y(t_i|\boldsymbol{\theta}) \prod_{i \in \mathcal{RC}}S_Y(t_i|\boldsymbol{\theta}) \prod_{i \in \mathcal{LC}}(1-S_Y(t_i|\boldsymbol{\theta}))\prod_{i \in \mathcal{IC}}(S_Y(l_i|\boldsymbol{\theta})-S_Y(r_i|\boldsymbol{\theta}))
\tag{3.13}\]</span></span></p>
<p>In case of truncation, the adjustments are made to all observations, as we have to condition on the event occurring after/before the truncation time. The truncation adjusted likelihood contributions (assuming independence of truncation and event/censoring times) would thus be given by</p>
<ul>
<li><p>left-truncation:</p>
<ul>
<li>event: <span class="math inline">\(f(Y_i = t_i|Y_i \geq t_i^L, \boldsymbol{\theta}) = \frac{f_Y(t_i|\boldsymbol{\theta})}{S_Y(t_i^L|\boldsymbol{\theta})}\)</span></li>
<li>left-censoring: <span class="math inline">\(P(Y_i &lt; t_i|Y_i \geq t_i^L, \boldsymbol{\theta}) = \frac{S_Y(t_i|\boldsymbol{\theta})}{S_Y(t_i^L|\boldsymbol{\theta})}\)</span></li>
<li>interval-censoring: <span class="math inline">\(P(l_i &lt; Y_i \leq r_i|Y_i \geq t_i^L,\boldsymbol{\theta}) = \frac{S_Y(l_i|\boldsymbol{\theta}) - S_Y(r_i|\boldsymbol{\theta})}{S_Y(t_i^L|\boldsymbol{\theta})}\)</span></li>
</ul></li>
<li><p>right-truncation:</p>
<ul>
<li>event: <span class="math inline">\(f(Y_i = t_i|Y_i \leq t_i^R, \boldsymbol{\theta}) = \frac{f_Y(t_i|\boldsymbol{\theta})}{F_Y(t_i^R|\boldsymbol{\theta})}\)</span></li>
<li>left-censoring: <span class="math inline">\(P(Y_i &lt; t_i|Y_i \leq t_i^R, \boldsymbol{\theta}) = \frac{S_Y(t_i|\boldsymbol{\theta})}{F_Y(t_i^R|\boldsymbol{\theta})}\)</span></li>
<li>interval-censoring: <span class="math inline">\(P(l_i &lt; Y_i \leq r_i|Y_i \leq t_i^R,\boldsymbol{\theta}) = \frac{S_Y(l_i|\boldsymbol{\theta}) - S_Y(r_i|\boldsymbol{\theta})}{F_Y(t_i^R|\boldsymbol{\theta})}\)</span></li>
</ul></li>
</ul>
<p>Note that in in practice, data sets will often not contain all types of censoring or truncation, in which case <a href="#eq-objective-function" class="quarto-xref"><span>3.13</span></a> will contain only a subset of the product terms. This has been illustrated in <a href="#eq-right-censoring-likelihood" class="quarto-xref"><span>3.12</span></a> under absence of truncation and <span class="math inline">\(\mathcal{LC}=\mathcal{IC}=\emptyset\)</span>.</p>
</section>
<section id="sec-surv-estimation-non-param" class="level3" data-number="3.5.2">
<h3 data-number="3.5.2" class="anchored" data-anchor-id="sec-surv-estimation-non-param"><span class="header-section-number">3.5.2</span> Non-parametric estimation</h3>
<p>As the name suggests, non-parametric estimation (and semi-parametric estimation) techniques do not make (strong) assumptions about the underlying distribution of event times.</p>
<p>A common principle for such techniques is to partition the follow-up into intervals (or to define specific time-points during the follow-up) and to estimate the continuous (<a href="#eq-continuous-hazard" class="quarto-xref"><span>3.2</span></a>) or discrete time (<a href="#eq-discrete-hazard" class="quarto-xref"><span>3.5</span></a>) hazards for each interval/time-point. Once the respective hazard estimates are obtained, estimates of other quantities, for example the survival probability, can be derived based on the relationship given in <a href="#sec-distributions-continuous" class="quarto-xref"><span>Section 3.1.1</span></a> or <a href="#sec-distributions-discrete" class="quarto-xref"><span>Section 3.1.2</span></a>.</p>
<section id="sec-surv-km" class="level4" data-number="3.5.2.1">
<h4 data-number="3.5.2.1" class="anchored" data-anchor-id="sec-surv-km"><span class="header-section-number">3.5.2.1</span> Kaplan-Meier estimator</h4>
<p>The Kaplan-Meier estimator <span class="citation" data-cites="Kaplan1958">(<a href="P5C26_references.html#ref-Kaplan1958" role="doc-biblioref">Kaplan and Meier 1958</a>)</span>, a simple, non-parametric estimator for the survival function (<a href="#eq-surv-prob" class="quarto-xref"><span>3.1</span></a>), is an example for this principle. While it is usually used for continuous time, its construction can be motivated from a discrete time persepective. Recall from <a href="#sec-data-rc" class="quarto-xref"><span>Section 3.2</span></a> the definition of unique, ordered, event time <span class="math inline">\(t_{(k)}, k=1,\ldots,m\)</span> (<a href="#eq-unique-ordered-event-times" class="quarto-xref"><span>3.9</span></a>). Define <span class="math inline">\(\tilde{Y} = t_{(k)} \Leftrightarrow Y \in (t_{(k-1)},t_{(k)}]\)</span> as discrete time representation of <span class="math inline">\(Y\)</span>. Using the quantities introduced in <a href="#sec-data-rc" class="quarto-xref"><span>Section 3.2</span></a>,the discrete time hazard <a href="#eq-discrete-hazard" class="quarto-xref"><span>3.5</span></a> is estimated as the ratio of the number of events occurring at <span class="math inline">\(t_{(k)}\)</span>, to the number of subjects at risk at <span class="math inline">\(t_{(k)}\)</span>:</p>
<!--  -->
<p><span id="eq-disc-haz-non-param"><span class="math display">\[
h^d(t_{(k)}) = P\left(Y\in (t_{(k-1)},t_{(k)}]|Y&gt;t_{(k-1)}\right)=\frac{d_{t_{(k)}}}{n_{t_{(k)}}}.
\tag{3.14}\]</span></span> <!--  --> The survival probability and the definition of the Kaplan-Meier estimator (<a href="#eq-km" class="quarto-xref"><span>3.15</span></a>) then follow from <a href="#eq-discrete-surv-prob" class="quarto-xref"><span>3.7</span></a>: <!--  --> <span id="eq-km"><span class="math display">\[
S(\tau) = \prod_{k:t_{(k)} \leq \tau}\left(1-\frac{d_{t_{(k)}}}{n_{t_{(k)}}}\right)
\tag{3.15}\]</span></span> <!--  --></p>
<p>which is a step-function at the observed ordered event times <span class="math inline">\(t_{(k)}, k=1,\ldots,m\)</span> with <span class="math inline">\(S_{KM}(\tau) = 1\ \forall \tau &lt; t_{(1)}\)</span>. It is usually estimated for all unique event times. In this book, <span class="math inline">\(\hat{S}_{KM}\)</span> specifically refers to the Kaplan-Meier estimator fit on some training data.</p>
<p>For illustration, we consider the <code>tumor</code> data set from <a href="#sec-data-rc" class="quarto-xref"><span>Section 3.2</span></a>. <a href="#fig-km-tumor" class="quarto-xref">Figure&nbsp;<span>3.4</span></a> shows the estimated survival probability obtained by applying the Kaplan-Meier estimator to the full data set, containing observations of <span class="math inline">\(n=776\)</span> subjects. By definition, the survival function starts at <span class="math inline">\(S(t)=1\)</span> at <span class="math inline">\(t=0\)</span> and monotonically decreases towards <span class="math inline">\(S(t)=0\)</span> for <span class="math inline">\(t\rightarrow \infty\)</span>. However, follow-ups are usually not infinite and in the presence of censoring, the survival function may not reach 0 by the end of the observation period. For the Kaplan-Meier estimator specifically, <span class="math inline">\(\hat{S}_{KM}(t_{(m)}) = 0\)</span> only if the last observed time is an event time and there are no censored observations at the same time.</p>
<p>Dotted lines indicate the median survival, defined as the time at which the survival function reaches <span class="math inline">\(S(t) = 0.5\)</span>. In this example, the median survival time is approximately 1500 days. This means that 50% of the subjects are expected die within 1500 days after operation.</p>
<div id="fig-km-tumor" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Illustration of the Kaplan-Meier estimate of the survival probability for the tumor data set. Median survival time is indicated by dotted lines.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-km-tumor-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survival/km-tumor.png" class="img-fluid figure-img" alt="Illustration of the Kaplan-Meier estimate of the survival probability for the tumor data set. Median survival time is indicated by dotted lines.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-km-tumor-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.4: Kaplan-Meier estimate for the <code>tumor</code> data <span class="citation" data-cites="pkgpammtools">(<a href="P5C26_references.html#ref-pkgpammtools" role="doc-biblioref">Bender and Scheipl 2018</a>)</span>. The estimated survival probabilities are given by the solid step-function. Dotted lines indicate the median survival time).
</figcaption>
</figure>
</div>
<p>While the Kaplan-Meier estimator does not directly support estimation of covariate effects on the estimated survival probabilities, it is often used for descriptive analysis by applying the estimator to different subgroups (referred to as stratification in survival analysis). For example, <a href="#fig-km-tumor-age-bin" class="quarto-xref">Figure&nbsp;<span>3.5</span></a> shows <span class="math inline">\(\hat{S}_{KM}\)</span> separately for subjects aged 50 years or older and subjects younger than 50 years, respectively. Dashed lines again illustrate median survival times. However, note that the median survival time does not exist for the younger age group, as their estimated survival function does not cross <span class="math inline">\(0.5\)</span>.</p>
<div id="fig-km-tumor-age-bin" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Illustration of the Kaplan-Meier estimate of the survival probability for the tumor data set, separately for two groups, patients below and above 50 years of age. Median survival time is indicated by dotted lines. Fot the younger age group, the line doesn't intersect with the survival curve.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-km-tumor-age-bin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survival/km-age-bin-tumor.png" class="img-fluid figure-img" alt="Illustration of the Kaplan-Meier estimate of the survival probability for the tumor data set, separately for two groups, patients below and above 50 years of age. Median survival time is indicated by dotted lines. Fot the younger age group, the line doesn't intersect with the survival curve.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-km-tumor-age-bin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.5: Kaplan-Meier estimate for the <code>tumor</code> data <span class="citation" data-cites="pkgpammtools">(<a href="P5C26_references.html#ref-pkgpammtools" role="doc-biblioref">Bender and Scheipl 2018</a>)</span> applied to subgroups of subjects of 50 or more years old and less than 50 years old, respectively. The estimated survival probabilities are given by the solid step-function. Dotted lines indicate the median survival time).
</figcaption>
</figure>
</div>
</section>
<section id="sec-surv-na" class="level4" data-number="3.5.2.2">
<h4 data-number="3.5.2.2" class="anchored" data-anchor-id="sec-surv-na"><span class="header-section-number">3.5.2.2</span> Nelson-Aalen</h4>
<p>As the Kaplan-Meier estimator (<a href="#sec-surv-km" class="quarto-xref"><span>Section 3.5.2.1</span></a>), the Nelson-Aalen estimator (<span class="citation" data-cites="Nelson1972">Nelson (<a href="P5C26_references.html#ref-Nelson1972" role="doc-biblioref">1972</a>)</span>,<span class="citation" data-cites="Aalen1978">Aalen (<a href="P5C26_references.html#ref-Aalen1978" role="doc-biblioref">1978</a>)</span>) is based on the discrete time hazards <a href="#eq-disc-haz-non-param" class="quarto-xref"><span>3.14</span></a>. Rather than estimating the survival probability, the Nelson-Aalen estimator estimates the discrete time cumulative hazard function <a href="#eq-discrete-cumu-hazard" class="quarto-xref"><span>3.6</span></a> and is defined as</p>
<p><span id="eq-nelson-aalen"><span class="math display">\[
H_{NA}(\tau) = \sum_{k:t_{(k)}\leq \tau} h^d(t_{(k)}) = \sum_{k:t_{(k)}\leq \tau} \frac{d_{t_{(k)}}}{n_{t_{(k)}}}.
\tag{3.16}\]</span></span></p>
<p>Note that we use a continuous time valued <span class="math inline">\(\tau\)</span> for the definition, which implies that <span class="math inline">\(H_{NA,e}(\tau) = H_{NA,e}(t_{(k)})\ \forall \tau \in [t_{(k)}, t_{(k+1)})\)</span>. In words: For time points between two unique event times we assume the previous value of the estimate.</p>
<p>Based on <a href="#eq-surv-haz" class="quarto-xref"><span>3.4</span></a> we can also define a survival probability estimator based on the Nelson-Aalen estimator as</p>
<p><span id="eq-surv-na"><span class="math display">\[
S_{NA}(\tau) = \exp(-H_{NA}(\tau)).
\tag{3.17}\]</span></span></p>
<p>Kaplan–Meier and Nelson–Aalen are consistent for survival and cumulative hazard respectively under independent right-censoring. Both these non-parametric estimators can be used as simple predictive methods (<a href="P3C13_classical.html#sec-classical-nonpar" class="quarto-xref"><span>Section 10.1</span></a>), and are often used as informative baseline models, which can be compared to sophisticated alternatives to improve interpretability in model evaluation <span class="citation" data-cites="Herrmann2020 Huang2020 Wang2017">(<a href="P5C26_references.html#ref-Herrmann2020" role="doc-biblioref">Herrmann et al. 2021</a>; <a href="P5C26_references.html#ref-Huang2020" role="doc-biblioref">Huang et al. 2020</a>; <a href="P5C26_references.html#ref-Wang2017" role="doc-biblioref">Wang, Li, and Reddy 2019</a>)</span>. Both methods can also be used for graphical calibration of models (<a href="P2C9_calib.html#sec-calib-km" class="quarto-xref"><span>Section 7.2.1</span></a>), components of complex models (<a href="P4C19_reductions.html" class="quarto-xref"><span>Chapter 15</span></a>), and other diagnostic graphical tools <span class="citation" data-cites="Habibi2018 Jager2008 Moghimi-dehkordi2008">(<a href="P5C26_references.html#ref-Habibi2018" role="doc-biblioref">Habibi et al. 2018</a>; <a href="P5C26_references.html#ref-Jager2008" role="doc-biblioref">Jager et al. 2008</a>; <a href="P5C26_references.html#ref-Moghimi-dehkordi2008" role="doc-biblioref">Moghimi-dehkordi et al. 2008</a>)</span>.</p>
</section>
<section id="sec-non-param-lt" class="level4" data-number="3.5.2.3">
<h4 data-number="3.5.2.3" class="anchored" data-anchor-id="sec-non-param-lt"><span class="header-section-number">3.5.2.3</span> Left truncation</h4>
<p>Non-and semi-parametric estimators can be easily extended to deal with left-truncation by adjusting the risk set definition (<a href="#eq-risk-set" class="quarto-xref"><span>3.10</span></a>) to the more general definition <span class="citation" data-cites="McGough2021">(<a href="P5C26_references.html#ref-McGough2021" role="doc-biblioref">McGough et al. 2021</a>)</span> <!--  --> <span id="eq-risk-lt"><span class="math display">\[
\mathcal{R}_\tau^L = \{i: t_i^L \ \leq \tau \leq t_i \},
\tag{3.18}\]</span></span> <!--  --> which excludes individuals from the risk set until their left-truncation time has occurred (<a href="#eq-risk-lt" class="quarto-xref"><span>3.18</span></a>).</p>
<p>The number of individuals at risk at <span class="math inline">\(\tau\)</span> is then <!--  --> <span id="eq-risk-lt-n"><span class="math display">\[
n^L_\tau = |\mathcal{R}_\tau^L| = \sum_i \mathbb{I}(t_i^L \leq \tau \leq t_i),
\tag{3.19}\]</span></span> <!--  --> where <span class="math inline">\(t_i, t_i^L\)</span> are the outcome time and left truncation time of observation <span class="math inline">\(i\)</span> respectively.</p>
<p>The Kaplan-Meier (<a href="#eq-km" class="quarto-xref"><span>3.15</span></a>) and Nelson-Aalen (<a href="#eq-nelson-aalen" class="quarto-xref"><span>3.16</span></a>) estimators can still be used with <span class="math inline">\(n_{t_{(k)}}\)</span> replaced by <span class="math inline">\(n^L_{t_{(k)}}\)</span> and <span class="math inline">\(d_{t_{(k)}}\)</span> calculated in the usual manner.</p>
<p>For illustration consider an excerpt from the infants data in <a href="#tbl-surv-data-infants" class="quarto-xref">Table&nbsp;<span>3.2</span></a> discussed in <a href="#sec-left-truncation" class="quarto-xref"><span>Section 3.4.1</span></a>, where <span class="math inline">\(t\)</span> is the observed event or censoring time and <span class="math inline">\(t^L\)</span> is the left-truncation time.</p>
<div id="tbl-surv-data-infants" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-surv-data-infants-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3.2: Excerpt of the <code>infants</code> <span class="citation" data-cites="pkgeha">(<a href="P5C26_references.html#ref-pkgeha" role="doc-biblioref">Broström 2024</a>)</span> time-to-event dataset. Rows are individual observations (id), <code>group</code> indicates matched infants, <span class="math inline">\(t^L\)</span> is the left-truncation time (time of inclusion into the study), <span class="math inline">\(t\)</span> is the observed time, <span class="math inline">\(\delta\)</span> is the event indicator.
</figcaption>
<div aria-describedby="tbl-surv-data-infants-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: right;">group</th>
<th style="text-align: right;">id</th>
<th style="text-align: right;"><span class="math inline">\(t^L\)</span></th>
<th style="text-align: right;"><span class="math inline">\(t\)</span></th>
<th style="text-align: right;"><span class="math inline">\(\delta\)</span></th>
<th style="text-align: left;">mother</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">dead</td>
</tr>
<tr class="even">
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">alive</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">alive</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">dead</td>
</tr>
<tr class="odd">
<td style="text-align: right;">2</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">alive</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">alive</td>
</tr>
<tr class="odd">
<td style="text-align: right;">4</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">dead</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">alive</td>
</tr>
<tr class="odd">
<td style="text-align: right;">4</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">365</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">alive</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p>In <a href="#tbl-surv-data-infants" class="quarto-xref">Table&nbsp;<span>3.2</span></a>, without stratifying according to mother’s status, the first observed event time is <span class="math inline">\(t_{(1)} = t_7=16\)</span>. Then, ignoring left-truncation,</p>
<ul>
<li><span class="math inline">\(\mathcal{R}_{t_{(1)}} = \{1,2,3,4,5,6,7,8,9\}\)</span></li>
<li><span class="math inline">\(d_{t_{(1)}} = 1\)</span></li>
<li><span class="math inline">\(n_{t_{(1)}} = 9\)</span></li>
<li><span class="math inline">\(S(t_{(1)}) = 1 - \frac{1}{9} \approx 0.9\)</span></li>
</ul>
<p>In contrast, when we take left-truncation into account, subjects only enter the risk set for the event after their left-truncation time (we already know they survived until <span class="math inline">\(t_i^L\)</span> so they are not at risk for the event before that time), thus</p>
<ul>
<li><span class="math inline">\(\mathcal{R}^L_{t_{(1)}} = \{4, 5, 6, 7, 8, 9\}\)</span></li>
<li><span class="math inline">\(d_{t_{(1)}} = 1\)</span></li>
<li><span class="math inline">\(n_{t_{(1)}}^L = 6\)</span></li>
<li><span class="math inline">\(S(t_{(1)}) = 1 - \frac{1}{6} \approx 0.8\)</span></li>
</ul>
<p>Thus, in the presence of left-truncation, <span class="math inline">\(n_{t_{(k)}}^L \leq  n_{t_{(k)}}\)</span> and therefore <span class="math inline">\(S(t_{(k)})^L \leq S(t_{(k)})\)</span>.</p>
<p><a href="#fig-infants-lt" class="quarto-xref">Figure&nbsp;<span>3.6</span></a> shows the difference between estimated survival probabilities when left-truncation is ignored (left panel) and taken into account (right panel), respectively. It is clear that the survival probabilities were underestimated in both groups, but more so in the group of infants whose mother died (thereby underestimating the effect of the mothers’ death on infant survival).</p>
<div id="fig-infants-lt" class="quarto-float quarto-figure quarto-figure-center anchored" alt="">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-infants-lt-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survival/km-infants.png" class="img-fluid figure-img" alt="">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-infants-lt-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.6: Left: Kaplan-Meier estimate of survival probabilities of infants depending on status of the mother ignoring left-truncation. Right: Kaplan-Meier estimate of survival probabilities adjusting for left-truncation.
</figcaption>
</figure>
</div>
</section>
<section id="interval-censoring" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="interval-censoring">Interval-censoring</h4>
<p>For interval-censoring, the Kaplan-Meier estimator is replaced by the nonparametric maximum likelihood estimator (NPMLE) <span class="citation" data-cites="Zhigang2010">(<a href="P5C26_references.html#ref-Zhigang2010" role="doc-biblioref">Zhang and Sun 2010</a>)</span>, which estimates the likelihood function under the assumption of left- or interval-censored data as in <a href="#sec-surv-estimation-param" class="quarto-xref"><span>Section 3.5.1</span></a>. The most common algorithm to compute this is the iterative Turnbull algorithm <span class="citation" data-cites="Turnbull1974">(<a href="P5C26_references.html#ref-Turnbull1974" role="doc-biblioref">Turnbull 1974</a>)</span>. The intuition behind this algorithm is that an observation, <span class="math inline">\(i\)</span>, should only contribute to the estimation at a given time-point, <span class="math inline">\(\tau\)</span>, if the event could have happened in the censoring interval, <span class="math inline">\(\tau \in (L_i, R_i]\)</span>.</p>
<p>In summary <span class="citation" data-cites="Giolo2004">(<a href="P5C26_references.html#ref-Giolo2004" role="doc-biblioref">Giolo 2004</a>)</span>:</p>
<p>Let <span class="math inline">\(t_{(k)}, k = 1,...,m\)</span> be the ordered times at which to evaluate the NPMLE.</p>
<ol type="1">
<li>Make an initial guess for <span class="math inline">\(\hat{S}^0(t_{(k)}), k = 1,...,m\)</span>.</li>
</ol>
<p>Where <span class="math inline">\(\hat{S}^0\)</span> is the initial guess. Let <span class="math inline">\(\hat{S}^j\)</span> be the estimated survival function at iteration <span class="math inline">\(j\)</span>.</p>
<ol type="1">
<li><strong>While</strong> <span class="math inline">\(\quad \lvert \hat{S}^j(t_{(k)}) - \hat{S}^{j-1}(t_{(k)}) \rvert &gt; \epsilon, \forall k = 1,...,m \quad\)</span> <strong>Do:</strong></li>
</ol>
<ol type="a">
<li>Compute the probability of an event at each time-point:</li>
</ol>
<p><span class="math display">\[
p_{t_{(k)}} = \hat{S}^{j-1}(t_{(k-1)}) - \hat{S}^{j-1}(t_{(k)}), \quad k = 1,...,m
\]</span></p>
<ol start="2" type="a">
<li>Estimate the number of events at each time-point:</li>
</ol>
<p><span class="math display">\[
d_{t_{(k)}} = \sum^n_{i=1} \frac{w_{ik}p_{t_{(k)}}}{\sum^m_{l=1} w_{il}p_{t_{(l)}}}, \quad k = 1,...,m
\]</span></p>
<p>Where <span class="math inline">\(w_{ik} = 1\)</span> if the interval <span class="math inline">\((t_{(k-1)}, t_{(k)}]\)</span> is contained within <span class="math inline">\((L_i, R_i]\)</span>, and <span class="math inline">\(w_{ik} = 0\)</span> otherwise; and <span class="math inline">\(L_i, R_i\)</span> are the left- and right-censoring times for observations <span class="math inline">\(i = 1,...,n\)</span>.</p>
<ol start="3" type="a">
<li>Compute the cumulative number of events at or after <span class="math inline">\(t_{(k)}\)</span>:</li>
</ol>
<p><span class="math display">\[
n_{t_{(k)}} = \sum^m_{l=k} d_{t_{(l)}}, \quad k = 1,...,m
\]</span></p>
<p>This serves a similar role to the usual ‘number at risk’ formula.</p>
<ol start="4" type="a">
<li>Calculate <span class="math inline">\(\hat{S}^j\)</span> using the Kaplan-Meier formula (<a href="#eq-km" class="quarto-xref"><span>3.15</span></a>), substituting the values in Steps 2b and 2c.</li>
</ol>
<p><span class="citation" data-cites="Giolo2004">Giolo (<a href="P5C26_references.html#ref-Giolo2004" role="doc-biblioref">2004</a>)</span> suggest using the Kaplan-Meier to find the initial guess in Step 1 and to use <span class="math inline">\(\epsilon = 10^{-3}\)</span>. A similar algorithm can be used for left-censored data.</p>
</section>
</section>
</section>
<section id="sec-censoring-distribution" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="sec-censoring-distribution"><span class="header-section-number">3.6</span> Estimation of the censoring distribution</h2>
<p>The techniques discussed in section <a href="#sec-surv-estimation" class="quarto-xref"><span>Section 3.5</span></a> can also be used to estimate the distribution of censoring times <span class="math inline">\((t_i, 1-\delta_i)\)</span>. This is often done in order to calculate the inverse probability of censoring weights (<a href="#sec-surv-ipcw" class="quarto-xref"><span>Section 3.6.2</span></a>) used in the evaluation of survival models (<a href="P2C8_rank.html#sec-eval-crank-conc" class="quarto-xref"><span>Section 6.1.1</span></a>) and in order to account for (covariate-)dependent censoring (<span class="citation" data-cites="willems2018correctingdependent">Willems et al. (<a href="P5C26_references.html#ref-willems2018correctingdependent" role="doc-biblioref">2018</a>)</span>).</p>
<section id="sec-surv-km-censoring" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="sec-surv-km-censoring"><span class="header-section-number">3.6.1</span> The Kaplan-Meier estimator for the censoring distribution</h3>
<p>The Kaplan-Meier estimator can be used to estimate the distribution of censoring times. The definition of the risk set (<a href="#eq-risk-set" class="quarto-xref"><span>3.10</span></a>) is thus the denominator <span class="math inline">\(n_{\tau}\)</span> in <a href="#eq-disc-haz-non-param" class="quarto-xref"><span>3.14</span></a> remains the same, but the numerator <span class="math inline">\(d_{\tau}\)</span> is now the number of censoring events at <span class="math inline">\(\tau\)</span>, which we denote as</p>
<p><span class="math display">\[
d_\tau^C = \sum_i \mathbb{I}(t_i = \tau, \delta_i = 0).
\]</span></p>
<p>The Kaplan-Meier estimator for the censoring distribution is then given by</p>
<p><span id="eq-km-censoring"><span class="math display">\[
G_{KM}(\tau) = \prod_{\ell:t_{(\ell)} &lt; \tau} \left(1-\frac{d^C_{t_{(\ell)}}}{n_{t_{(\ell)}}}\right),
\tag{3.20}\]</span></span></p>
<p>where <span class="math inline">\(t_{(\ell)},l=1,\ldots,g\)</span> are the unique, ordered censoring times. In this book, <span class="math inline">\(\hat{G}_{KM}\)</span> specifically refers to the Kaplan-Meier estimator fit to the observed censoring times <span class="math inline">\((t_i, 1-\delta_i)\)</span> from some training data.</p>
</section>
<section id="sec-surv-ipcw" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="sec-surv-ipcw"><span class="header-section-number">3.6.2</span> Inverse probability of censoring weights (IPCW)</h3>
<p>The inverse probability of censoring (IPC) weights are defined as</p>
<p><span id="eq-ipcw"><span class="math display">\[
w(\tau|\mathbf{x}) = \frac{1}{S(\tau|\mathbf{x})}
\tag{3.21}\]</span></span></p>
<p>When no relationship between the censoring time and the covariates is assumed, the IPC weights are in most cases estimated by plugging in the Kaplan-Meier estimate of <a href="#eq-km-censoring" class="quarto-xref"><span>3.20</span></a> into <a href="#eq-ipcw" class="quarto-xref"><span>3.21</span></a>, such that</p>
<p><span id="eq-ipcw-km"><span class="math display">\[
\hat{w}(\tau) = \frac{1}{\hat{G}_{KM}(\tau)}
\tag{3.22}\]</span></span></p>
</section>
</section>
<section id="conclusion" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">3.7</span> Conclusion</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Warning</span>Key takeaways
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>The first step in any survival analysis is to understand the data and the type of censoring and or truncation present.</li>
<li>Any data set can have a combination of different types of censoring and truncation</li>
<li>Non-parametric estimators are well equipped to deal with right-censoring and left-truncation.</li>
<li>Parametric estimation can naturally deal with different types of censoring and truncation, but requires assumptions about the distribution of the event times.</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Tip</span>Further reading
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><span class="citation" data-cites="KalbfleischPrentice1973">Kalbfleisch and Prentice (<a href="P5C26_references.html#ref-KalbfleischPrentice1973" role="doc-biblioref">1973</a>)</span> provide an early, theoretical treatment of the main concepts of survival analysis.</li>
<li><span class="citation" data-cites="Klein2003">Klein and Moeschberger (<a href="P5C26_references.html#ref-Klein2003" role="doc-biblioref">2003</a>)</span> is a standard reference for survival analysis and provides a comprehensive overview of the different types of censoring and truncation</li>
<li>For practical discussion about survival models in the context of right-censoring and left-truncation see <span class="citation" data-cites="McGough2021">McGough et al. (<a href="P5C26_references.html#ref-McGough2021" role="doc-biblioref">2021</a>)</span></li>
</ul>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Aalen1978" class="csl-entry" role="listitem">
Aalen, Odd. 1978. <span>“<span class="nocase">Nonparametric Inference for a Family of Counting Processes</span>.”</span> <em>The Annals of Statistics</em> 6 (4): 701–26.
</div>
<div id="ref-akritasGeneralizedProductlimitEstimator2005" class="csl-entry" role="listitem">
Akritas, Michael G., and Michael P. LaValley. 2005. <span>“A Generalized Product-Limit Estimator for Truncated Data.”</span> <em>Nonparametric Statistics</em>, September. <a href="https://doi.org/10.1080/10485250500038637">https://doi.org/10.1080/10485250500038637</a>.
</div>
<div id="ref-pkgpammtools" class="csl-entry" role="listitem">
Bender, Andreas, and Fabian Scheipl. 2018. <span>“<span class="nocase">pammtools: Piece-wise exponential Additive Mixed Modeling tools</span>.”</span> <em>arXiv:1806.01042 [Stat]</em>. <a href="http://arxiv.org/abs/1806.01042">http://arxiv.org/abs/1806.01042</a>.
</div>
<div id="ref-brostrom.influence.1987" class="csl-entry" role="listitem">
Broström, Göran. 1987. <span>“The <span>Influence</span> of <span>Mother</span>’s <span>Death</span> on <span>Infant Mortality</span>: <span>A Case Study</span> in <span>Matched Data Survival Analysis</span>.”</span> <em>Scandinavian Journal of Statistics</em> 14 (2): 113–23. <a href="https://www.jstor.org/stable/4616055">https://www.jstor.org/stable/4616055</a>.
</div>
<div id="ref-pkgeha" class="csl-entry" role="listitem">
———. 2024. <em>Eha: Event History Analysis</em>. <a href="https://cran.r-project.org/package=eha">https://cran.r-project.org/package=eha</a>.
</div>
<div id="ref-Giolo2004" class="csl-entry" role="listitem">
Giolo, Suely. 2004. <span>“Turnbull’s Nonparametric Estimator for Interval-Censored Data,”</span> January.
</div>
<div id="ref-Habibi2018" class="csl-entry" role="listitem">
Habibi, Danial, Mohammad Rafiei, Ali Chehrei, Zahra Shayan, and Soheil Tafaqodi. 2018. <span>“<span class="nocase">Comparison of Survival Models for Analyzing Prognostic Factors in Gastric Cancer Patients</span>.”</span> <em>Asian Pacific Journal of Cancer Prevention : APJCP</em> 19 (3): 749–53. <a href="https://doi.org/10.22034/APJCP.2018.19.3.749">https://doi.org/10.22034/APJCP.2018.19.3.749</a>.
</div>
<div id="ref-Herrmann2020" class="csl-entry" role="listitem">
Herrmann, Moritz, Philipp Probst, Roman Hornung, Vindi Jurinovic, and Anne-Laure Boulesteix. 2021. <span>“<span class="nocase">Large-scale benchmark study of survival prediction methods using multi-omics data</span>.”</span> <em>Briefings in Bioinformatics</em> 22 (3). <a href="https://doi.org/10.1093/bib/bbaa167">https://doi.org/10.1093/bib/bbaa167</a>.
</div>
<div id="ref-Huang2020" class="csl-entry" role="listitem">
Huang, Shigao, Jie Yang, Simon Fong, and Qi Zhao. 2020. <span>“<span class="nocase">Artificial intelligence in cancer diagnosis and prognosis: Opportunities and challenges</span>.”</span> <em>Cancer Letters</em> 471: 61–71. https://doi.org/<a href="https://doi.org/10.1016/j.canlet.2019.12.007">https://doi.org/10.1016/j.canlet.2019.12.007</a>.
</div>
<div id="ref-Jager2008" class="csl-entry" role="listitem">
Jager, Kitty J, Paul C van Dijk, Carmine Zoccali, and Friedo W Dekker. 2008. <span>“<span class="nocase">The analysis of survival data: the Kaplan–Meier method</span>.”</span> <em>Kidney International</em> 74 (5): 560–65. https://doi.org/<a href="https://doi.org/10.1038/ki.2008.217">https://doi.org/10.1038/ki.2008.217</a>.
</div>
<div id="ref-KalbfleischPrentice1973" class="csl-entry" role="listitem">
Kalbfleisch, J. D., and R. L. Prentice. 1973. <span>“<span class="nocase">Marginal likelihoods based on Cox’s regression and life model</span>.”</span> <em>Biometrika</em> 60 (2): 267–78. <a href="https://doi.org/10.1093/biomet/60.2.267">https://doi.org/10.1093/biomet/60.2.267</a>.
</div>
<div id="ref-Kaplan1958" class="csl-entry" role="listitem">
Kaplan, E. L., and Paul Meier. 1958. <span>“<span class="nocase">Nonparametric Estimation from Incomplete Observations</span>.”</span> <em>Journal of the American Statistical Association</em> 53 (282): 457–81. <a href="https://doi.org/10.2307/2281868">https://doi.org/10.2307/2281868</a>.
</div>
<div id="ref-Klein2003" class="csl-entry" role="listitem">
Klein, John P, and Melvin L Moeschberger. 2003. <em><span class="nocase">Survival analysis: techniques for censored and truncated data</span></em>. 2nd ed. Springer Science &amp; Business Media.
</div>
<div id="ref-McGough2021" class="csl-entry" role="listitem">
McGough, Sarah F., Devin Incerti, Svetlana Lyalina, Ryan Copping, Balasubramanian Narasimhan, and Robert Tibshirani. 2021. <span>“Penalized Regression for Left-Truncated and Right-Censored Survival Data.”</span> <em>Statistics in Medicine</em> 40 (25): 5487–5500. https://doi.org/<a href="https://doi.org/10.1002/sim.9136">https://doi.org/10.1002/sim.9136</a>.
</div>
<div id="ref-Moghimi-dehkordi2008" class="csl-entry" role="listitem">
Moghimi-dehkordi, Bijan, Azadeh Safaee, Mohamad Amin Pourhoseingholi, Reza Fatemi, Ziaoddin Tabeie, and Mohammad Reza Zali. 2008. <span>“<span class="nocase">Statistical Comparison of Survival Models for Analysis of Cancer Data</span>.”</span> <em>Asian Pacific Journal of Cancer Prevention</em> 9: 417–20.
</div>
<div id="ref-Nelson1972" class="csl-entry" role="listitem">
Nelson, Wayne. 1972. <span>“<span class="nocase">Theory and Applications of Hazard Plotting for Censored Failure Data</span>.”</span> <em>Technometrics</em> 14 (4): 945–66.
</div>
<div id="ref-Turnbull1974" class="csl-entry" role="listitem">
Turnbull, Bruce W. 1974. <span>“Nonparametric Estimation of a Survivorship Function with Doubly Censored Data.”</span> <em>Journal of the American Statistical Association</em> 69 (345): 169–73. <a href="https://doi.org/10.1080/01621459.1974.10480146">https://doi.org/10.1080/01621459.1974.10480146</a>.
</div>
<div id="ref-Tutz2016" class="csl-entry" role="listitem">
Tutz, Gerhard, and Matthias Schmid. 2016. <em><span class="nocase">Modeling Discrete Time-to-Event Data</span></em>. Springer Series in Statistics. Cham: Springer International Publishing. <a href="https://doi.org/10.1007/978-3-319-28158-2">https://doi.org/10.1007/978-3-319-28158-2</a>.
</div>
<div id="ref-vakulenko-lagun.inverse.2020" class="csl-entry" role="listitem">
Vakulenko-Lagun, Bella, Micha Mandel, and Rebecca A. Betensky. 2020. <span>“Inverse <span>Probability Weighting Methods</span> for <span>Cox Regression</span> with <span>Right-Truncated Data</span>.”</span> <em>Biometrics</em> 76 (2): 484–95. <a href="https://doi.org/10.1111/biom.13162">https://doi.org/10.1111/biom.13162</a>.
</div>
<div id="ref-Wang2017" class="csl-entry" role="listitem">
Wang, Ping, Yan Li, and Chandan K. Reddy. 2019. <span>“<span class="nocase">Machine Learning for Survival Analysis</span>.”</span> <em>ACM Computing Surveys</em> 51 (6): 1–36. <a href="https://doi.org/10.1145/3214306">https://doi.org/10.1145/3214306</a>.
</div>
<div id="ref-willems2018correctingdependent" class="csl-entry" role="listitem">
Willems, SJW, A Schat, MS van Noorden, and M Fiocco. 2018. <span>“Correcting for Dependent Censoring in Routine Outcome Monitoring Data by Applying the Inverse Probability Censoring Weighted Estimator.”</span> <em>Statistical Methods in Medical Research</em> 27 (2): 323–35. <a href="https://doi.org/10.1177/0962280216628900">https://doi.org/10.1177/0962280216628900</a>.
</div>
<div id="ref-Zhigang2010" class="csl-entry" role="listitem">
Zhang, Zhigang, and Jianguo Sun. 2010. <span>“Interval Censoring.”</span> <em>Statistical Methods in Medical Research</em> 19 (1): 53–70. <a href="https://doi.org/10.1177/0962280209105023">https://doi.org/10.1177/0962280209105023</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./P1C3_machinelearning.html" class="pagination-link" aria-label="Machine Learning">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Machine Learning</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./P1C5_eha.html" class="pagination-link" aria-label="Event-history Analysis">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>All content licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> <br> © Raphael Sonabend, Andreas Bender.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://www.mlsabook.com">Website</a> | <a href="https://github.com/mlsa-book/MLSA">GitHub</a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P1C4_survival.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P1C4_survival.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>