<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.27">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Raphael Sonabend and Andreas Bender">

<title>5&nbsp; Survival Task – Machine Learning in Survival Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./P2C8_rank.html" rel="next">
<link href="./P1C5_eha.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-14d2473a982606de670f0038afa36c2b.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-1008a5a8f5e3c2ed0fd245401abc5d00.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P1C3_machinelearning.html">Survival Analysis and Machine Learning</a></li><li class="breadcrumb-item"><a href="./P1C6_survtsk.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Survival Task</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Machine Learning in Survival Analysis</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/mlsa-book/MLSA/tree/main/book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./Machine-Learning-in-Survival-Analysis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C0_notation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Symbols and Notation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Survival Analysis and Machine Learning</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C3_machinelearning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C4_survival.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C5_eha.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C6_survtsk.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Survival Task</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C8_rank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Discrimination</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C9_calib.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Calibration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C10_rules.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Scoring Rules</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C11_time.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Survival Time Measures</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Models</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C13_classical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Traditional Survival Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C14_forests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Random Forests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C15_svm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C16_boosting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Boosting Methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C17_neural.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Neural Networks</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Reduction Techniques</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C19_reductions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Reduction Techniques for Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C20_ipcw.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">IPC weighted classification</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C21_pseudo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Pseudo-value regression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C22_partition-based-reduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Partition based reductions</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C24_eha.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Reductions for Event-History Analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C24_conclusions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">FAQs and Outlook</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C26_references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-survtsk-dist" id="toc-sec-survtsk-dist" class="nav-link active" data-scroll-target="#sec-survtsk-dist"><span class="header-section-number">5.1</span> Predicting Distributions</a></li>
  <li><a href="#sec-survtsk-risk" id="toc-sec-survtsk-risk" class="nav-link" data-scroll-target="#sec-survtsk-risk"><span class="header-section-number">5.2</span> Predicting Risks</a>
  <ul class="collapse">
  <li><a href="#distributions-and-risks" id="toc-distributions-and-risks" class="nav-link" data-scroll-target="#distributions-and-risks"><span class="header-section-number">5.2.1</span> Distributions and risks</a></li>
  </ul></li>
  <li><a href="#sec-survtsk-time" id="toc-sec-survtsk-time" class="nav-link" data-scroll-target="#sec-survtsk-time"><span class="header-section-number">5.3</span> Predicting Survival Times</a>
  <ul class="collapse">
  <li><a href="#times-and-risks" id="toc-times-and-risks" class="nav-link" data-scroll-target="#times-and-risks"><span class="header-section-number">5.3.1</span> Times and risks</a></li>
  <li><a href="#times-and-distributions" id="toc-times-and-distributions" class="nav-link" data-scroll-target="#times-and-distributions"><span class="header-section-number">5.3.2</span> Times and distributions</a></li>
  </ul></li>
  <li><a href="#sec-survtsk-PI" id="toc-sec-survtsk-PI" class="nav-link" data-scroll-target="#sec-survtsk-PI"><span class="header-section-number">5.4</span> Prognostic Index Predictions</a>
  <ul class="collapse">
  <li><a href="#prognostic-index-risks-and-times" id="toc-prognostic-index-risks-and-times" class="nav-link" data-scroll-target="#prognostic-index-risks-and-times"><span class="header-section-number">5.4.1</span> Prognostic index, risks, and times</a></li>
  <li><a href="#prognostic-index-and-distributions" id="toc-prognostic-index-and-distributions" class="nav-link" data-scroll-target="#prognostic-index-and-distributions"><span class="header-section-number">5.4.2</span> Prognostic index and distributions</a></li>
  </ul></li>
  <li><a href="#beyond-single-event" id="toc-beyond-single-event" class="nav-link" data-scroll-target="#beyond-single-event"><span class="header-section-number">5.5</span> Beyond single-event</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5.6</span> Conclusion</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P1C6_survtsk.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P1C6_survtsk.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P1C3_machinelearning.html">Survival Analysis and Machine Learning</a></li><li class="breadcrumb-item"><a href="./P1C6_survtsk.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Survival Task</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span id="sec-survtsk" class="quarto-section-identifier"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Survival Task</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This chapter introduces the different survival problems and formalises them as survival tasks. There are four prediction types in survival analysis: relative risks - predicting the risk of an event, survival times - predicting the time until an event happens, prognostic index - predicting a linear predictor to assess outcomes based on risk factors, and survival distributions - predicting the probability of an event taking place over time. These reduce to three formal survival tasks: deterministic (survival time), ranking (risks and prognostic index), and probabilistic (distribution). This separation of tasks helps create a taxonomy of survival models and losses that is used throughout the book.
  </div>
</div>


</header>


<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Warning</span>Minor changes expected!
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This page is a work in progress and minor changes will be made over time.</strong></p>
</div>
</div>
<p>The preceding chapters have focused on understanding survival data. This final chapter turns to the prediction tasks themselves.</p>
<p>Recall from <a href="P1C3_machinelearning.html#sec-ml-basics" class="quarto-xref"><span>Section 2.1</span></a> that a general survival prediction problem is one in which:</p>
<ol type="1">
<li>a survival dataset, <span class="math inline">\(\mathcal{D}\)</span>, is split for training, <span class="math inline">\(\mathcal{D}_{train}\)</span>, and testing, <span class="math inline">\(\mathcal{D}_{test}\)</span>;</li>
<li>a survival model is fit on <span class="math inline">\(\mathcal{D}_{train}\)</span>; and</li>
<li>the model predicts a representation of the unknown true survival time, <span class="math inline">\(Y\)</span>, given <span class="math inline">\(\mathcal{D}_{test}\)</span>.</li>
</ol>
<p>The process of model fitting is model-dependent and ranges from non-parametric methods to machine learning approaches; these are discussed in Part III. Discussing the typical ‘representations of <span class="math inline">\(Y\)</span>’ will be the focus of this chapter.</p>
<p>In general there are four common <em>prediction types</em> or <em>prediction problems</em> which codify different representations of <span class="math inline">\(Y\)</span>, these are:</p>
<ol type="1">
<li>Survival distributions;</li>
<li>Time-to-event;</li>
<li>Relative risks;</li>
<li>Prognostic index.</li>
</ol>
<p>The first of these is considered <em>probabilistic</em> as it predicts the probability of an event occurring over time. The second is <em>deterministic</em> as the prediction is a single point prediction. The final two do not directly predict the event of interest but instead provide different interpretations that both relate to the risk of the event. The relative risks and time-to-event prediction types can both be derived from a survival distribution prediction, and the prognostic index is often used as an interim step before making a probabilistic prediction (see <span class="quarto-unresolved-ref">?sec-surv-models-crank</span>); so all of these prediction types are closely related.</p>
<p>It is vital to clearly separate which prediction problem you are working with, as they are not directly compatible with one another. For example, it is not meaningful to compare a relative risk prediction from one model to a survival distribution prediction of another. Moreover, prediction types may look similar but have very different implications, for example if one erroneously interprets a survival time as a relative risk then they could assume an individual has a worse outcome than they do as longer survival times imply lower risk.</p>
<p>Each prediction type has its own particular advantages and disadvantages and should be used appropriately. For example, survival times are not advised in domains where over-confidence can be detrimental, but can be used when making less-critical predictions, such as a marathon runner’s finish time (<a href="#fig-survtsk-overview" class="quarto-xref">Figure&nbsp;<span>5.1</span></a>, top-left), which is a survival analysis problem as many people do not finish. Whilst clinicians will usually not present a probability distribution to a patient (<a href="#fig-survtsk-overview" class="quarto-xref">Figure&nbsp;<span>5.1</span></a>, top-right), it is common to relay information such as ‘5-year survival probability’, which is read from a predicted survival distribution. Finally, relative risk predictions are often used for triage and resource allocation, such as assigning patients to particular wards or beds (<a href="#fig-survtsk-overview" class="quarto-xref">Figure&nbsp;<span>5.1</span></a>, bottom).</p>
<div id="fig-survtsk-overview" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Cartoon illustration of predictive types. Top left shows a marathon runner with a predicted survival time '4 hours 13 min', top right shows a doctor presenting a graph to a patient representing distribution predictions, bottom shows a clinician looking at a chart that ranks patients.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-survtsk-overview-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survtsk/overview.png" class="img-fluid figure-img" style="width:60.0%" alt="Cartoon illustration of predictive types. Top left shows a marathon runner with a predicted survival time '4 hours 13 min', top right shows a doctor presenting a graph to a patient representing distribution predictions, bottom shows a clinician looking at a chart that ranks patients.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-survtsk-overview-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.1: Illustration of prediction types. Survival time predictions (top-left) are limited to domains where uncertainty is less important. Survival probabilities (top-right) are shared with patients as ‘T-year survival probabilities’. Relative risk predictions (bottom) are used for triage and resource allocation.
</figcaption>
</figure>
</div>
<!-- FIXME: FIGURE TO BE UPDATED -->
<p>Whilst one prediction type is necessary to define a model’s output, it is often the case that an algorithm might use another prediction type as an interim prediction. This chaining of predictions is called ‘composition’ as two components are composed to create a final prediction, this is commonly the case for distribution predictions. It is also possible to convert prediction types between one another. <a href="#tbl-survtsk-trafo" class="quarto-xref">Table&nbsp;<span>5.1</span></a> highlights how distribution and survival time predictions can be transformed to most other forms and how all forms can be transformed to a relative risk prediction. These transformations will be discussed in more detail in the rest of this chapter.</p>
<p>Throughout this chapter let <span class="math inline">\(\mathcal{X}\subseteq \mathbb{R}^{n \times p}\)</span> be a set representing covariates. Note that the type of censoring in the data is part of the estimation problem and does not affect the prediction types, hence this chapter does not differentiate between right, left, or interval censoring. The focus is primarily on the single-event setting, a brief section on other settings is included at the end of the chapter.</p>
<div id="tbl-survtsk-trafo" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-survtsk-trafo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.1: Converting prediction types between one another. Rows are the original prediction type and columns are the transformed prediction, entries are the transformation methods. ‘NA’ indicates transformation is not possible between the two prediction types without significant assumptions. <span class="math inline">\(\phi, \Phi\)</span> are arbitrary functions.
</figcaption>
<div aria-describedby="tbl-survtsk-trafo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>Distribution</th>
<th>Risk</th>
<th>Time</th>
<th>PI</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Distribution, <span class="math inline">\(\hat{S}\)</span></td>
<td><span class="math inline">\(\hat{S}\)</span></td>
<td><span class="math inline">\(\sum_t \hat{H}(t)\)</span></td>
<td><span class="math inline">\(\operatorname{RMST}(\tau)\)</span></td>
<td>NA</td>
</tr>
<tr class="even">
<td>Risk, <span class="math inline">\(\hat{r}\)</span></td>
<td><span class="math inline">\(\Phi(\hat{r})\)</span></td>
<td><span class="math inline">\(\hat{r}\)</span></td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>Time, <span class="math inline">\(\hat{y}\)</span></td>
<td><span class="math inline">\(\operatorname{Distr}(\hat{y}, \sigma)\)</span></td>
<td><span class="math inline">\(-\hat{y}\)</span></td>
<td><span class="math inline">\(\hat{y}\)</span></td>
<td>NA</td>
</tr>
<tr class="even">
<td>PI, <span class="math inline">\(\hat{\eta}\)</span></td>
<td><span class="math inline">\(\phi(\hat{\eta}, \hat{y})\)</span></td>
<td><span class="math inline">\(\hat{\eta}\)</span> or <span class="math inline">\(-\hat{\eta}\)</span></td>
<td>NA</td>
<td><span class="math inline">\(\hat{\eta}\)</span></td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<section id="sec-survtsk-dist" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="sec-survtsk-dist"><span class="header-section-number">5.1</span> Predicting Distributions</h2>
<p>Predicting a survival distribution means predicting the probability of an individual surviving over time from <span class="math inline">\(0\)</span> to <span class="math inline">\(\infty\)</span>. Ideally one would make predictions over the continuous <span class="math inline">\(\mathbb{R}_{\geq 0}\)</span>, however, in practice it is more common for predictions to be made over <span class="math inline">\(\mathbb{N}_0\)</span>. This is due to the majority of models using discrete non-parametric estimators to create a distribution prediction. Distributional prediction can, in theory, target any of the distribution defining functions introduced in <a href="P1C4_survival.html#sec-distributions" class="quarto-xref"><span>Section 3.1</span></a>, but predicting <span class="math inline">\(S(t)\)</span> and/or <span class="math inline">\(h(t)\)</span> is most common. This is a <em>probabilistic survival task</em> as uncertainty is explicit in the prediction. Mathematically, the task is defined by <span class="math inline">\(g: \mathcal{X}\rightarrow \mathcal{S}\)</span> where <span class="math inline">\(\mathcal{S}\subseteq \operatorname{Distr}(\mathbb{R}_{\geq 0})\)</span> is a set of distributions on <span class="math inline">\(\mathbb{R}_{\geq 0}\)</span>.</p>
<p>Practically, especially in healthcare, survival distribution predictions are often used to estimate ‘<span class="math inline">\(\tau\)</span>-year survival probabilities’, which is the probability of survival at a given point in time. Therefore a clinician is not likely to display a survival curve to a patient, but may use their individual features to compute probabilities of survival at key time-points (often, 5, 10 years) – an example of this in use is the PREDICT breast cancer tool <span class="citation" data-cites="Candido2017">(<a href="P5C26_references.html#ref-Candido2017" role="doc-biblioref">Candido dos Reis et al. 2017</a>)</span>. In other contexts, such as engineering, survival distributions might be used to establish thresholds for replacing components. For example, one could use a survival model to estimate the reliability of a plane engine over time, a threshold could be set such that the engine will be replaced when <span class="math inline">\(\hat{S}(t) &lt; 0.6\)</span>.</p>
<p>Predicting <span class="math inline">\(\tau\)</span>-year survival probabilities is often confused with a classification problem, which make probabilities for one or more events occurring at a fixed point in time. However, a classification model would be unable to use any observations that were censored before the time of interest and discarding these observations would bias any results <span class="citation" data-cites="DeRong2025">(<a href="P5C26_references.html#ref-DeRong2025" role="doc-biblioref">Loh et al. 2025</a>)</span>.</p>
<p>Another potential source of confusion can arise when trying to figure out what it actually means to predict a distribution that represents a single event occurring for a single observation. In reality, something either does or does not happen at a point in time, so what does it really mean to predict a distribution around this event? To make this clear, <a href="#fig-survtsk-heaviside" class="quarto-xref">Figure&nbsp;<span>5.2</span></a> visualizes what the survival task aims to achieve. The top-left image shows the real-world distribution when an event occurs, this is a heaviside function where the survival probability drops immediately from <span class="math inline">\(1\)</span> to <span class="math inline">\(0\)</span> when the event occurs; the top-right image highlights this curve for a single event. The bottom-left shows what we aim to achieve, which is a survival curve prediction that accurately captures the average survival distribution across all of these events. Finally, the bottom-right curve shows the same events split according to a single, binary covariate with one prediction per covariate value. In a machine learning task, this is essentially repeated for every possible combination of covariates.</p>
<div id="fig-survtsk-heaviside" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Four graphs with 'survival probability S(t)' on the y-axis and 't' on the x-axis. The top left shows 10 random heaviside functions. The top right shows 1 of these highlighted. The bottom left shows a smooth curve superimposed across all of the same lines. The bottom right shows the same 10 lines but split between red and blue with two smooth curves in the same colors superimposed on top.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-survtsk-heaviside-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survtsk/heavisides.png" class="img-fluid figure-img" style="width:80.0%" alt="Four graphs with 'survival probability S(t)' on the y-axis and 't' on the x-axis. The top left shows 10 random heaviside functions. The top right shows 1 of these highlighted. The bottom left shows a smooth curve superimposed across all of the same lines. The bottom right shows the same 10 lines but split between red and blue with two smooth curves in the same colors superimposed on top.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-survtsk-heaviside-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.2: Comparing theoretical real-world events to survival distribution predictions.
</figcaption>
</figure>
</div>
</section>
<section id="sec-survtsk-risk" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="sec-survtsk-risk"><span class="header-section-number">5.2</span> Predicting Risks</h2>
<p>Predicting risks is defined as predicting a continuous rank for an individual’s relative risk of experiencing the event. Therefore this might also be known as a ‘ranking’ problem or predicting ‘continuous rankings’. In machine learning terms, this task is the problem of estimating, <span class="math inline">\(g: \mathcal{X}\rightarrow \mathbb{R}\)</span>.</p>
<p>Interpretation of these rankings is more complex than might be imagined as the meaning differs between models, parametrizations, and even software implementations. To be consistent, in this book a larger risk value <em>always</em> corresponds to a higher risk of an event and lower values corresponding to lower risk.</p>
<p>A potential confusion that should be avoided is conflating this risk prediction with a prediction of absolute risk. Risk predictions are specifically <em>relative risks</em>, which means the risks are only comparable to other observations within the same sample. For example, given three subjects, <span class="math inline">\(\{i,j,k\}\)</span>, a risk prediction may be <span class="math inline">\(\{0.5, 10, 0.1\}\)</span> respectively. From these predictions, two primary types of conclusion can be drawn.</p>
<ol type="1">
<li>Conclusions comparing subjects:</li>
</ol>
<ul>
<li>The corresponding ranks for <span class="math inline">\(i,j,k\)</span> are <span class="math inline">\(2,3,1\)</span>;</li>
<li><span class="math inline">\(k\)</span> is at the least risk and <span class="math inline">\(j\)</span> is at the highest risk;</li>
<li>The risk of <span class="math inline">\(i\)</span> is slightly higher than that of <span class="math inline">\(k\)</span> but <span class="math inline">\(j\)</span>’s risk is considerably higher than both the others.</li>
</ul>
<ol start="2" type="1">
<li>Conclusions comparing risk groups:</li>
</ol>
<ul>
<li>Thresholding risks at <span class="math inline">\(0.4\)</span> means <span class="math inline">\(k\)</span> is at a low-risk but <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> are high-risk.</li>
<li>Thresholding risks at <span class="math inline">\(1.0\)</span> means <span class="math inline">\(i\)</span> and <span class="math inline">\(k\)</span> are low-risk but <span class="math inline">\(j\)</span> is high-risk.</li>
</ul>
<p>Whilst many important conclusions can be drawn from these predictions, the values themselves have no meaning when not compared to other individuals. Similarly, the values have no meanings across research, even if observation <span class="math inline">\(k\)</span> is at low-risk according to this sample, it may be high risk compared to another. Finally, as with relative risks in other domains, differences in relative risk should be interpreted with care. In the example above, <span class="math inline">\(j\)</span>’s relative risk is 100 times that of <span class="math inline">\(k\)</span>, but if <span class="math inline">\(k\)</span>’s overall probability of experiencing the event is <span class="math inline">\(0.0001\)</span>, then <span class="math inline">\(j\)</span>’s overall probability of experiencing the event is still only <span class="math inline">\(0.01\)</span>.</p>
<section id="distributions-and-risks" class="level3" data-number="5.2.1">
<h3 data-number="5.2.1" class="anchored" data-anchor-id="distributions-and-risks"><span class="header-section-number">5.2.1</span> Distributions and risks</h3>
<p>In general it is not possible to easily convert a risk prediction to a distribution. However, this may be possible if the risk prediction corresponds to a particular model form. This is discussed further in <a href="#sec-survtsk-PI" class="quarto-xref"><span>Section 5.4</span></a>.</p>
<p>More common is transformation from a distribution to a risk. Theoretically the simplest way to do so would be to take the mean or median of the distribution, however as will be discussed in detail in <a href="#sec-survtsk-time" class="quarto-xref"><span>Section 5.3</span></a> this is practically difficult. Instead, a stable approach is to use the ‘ensemble’ or ‘expected mortality’ to create a measure of risk <span class="citation" data-cites="Ishwaran2008">(<a href="P5C26_references.html#ref-Ishwaran2008" role="doc-biblioref">Ishwaran et al. 2008</a>)</span>. The expected mortality is defined by</p>
<p><span class="math display">\[
\sum_t -\log(\hat{S}_i(t)) = \sum_t \hat{H}_i(t)
\]</span></p>
<p>and is closely related to the calibration measure defined in <a href="P2C9_calib.html#sec-alpha" class="quarto-xref"><span>Section 7.1.2</span></a>. This represents the expected number of events for individuals with similar characteristics to <span class="math inline">\(i\)</span>. This is a measure of risk as a larger value indicates that among individuals with a similar profile, there is expected to be a larger number of events and therefore have greater risk than those with smaller ensemble mortality. For example, say for an individual, <span class="math inline">\(i\)</span>, we have: <span class="math inline">\((t, \hat{S}_i(t)) = (0, 1), (1, 0.8), (2, 0.4), (3, 0.15)\)</span> then, <span class="math inline">\((t, \hat{H}_i(t)) = (0, 0), (1, 0.10), (2, 0.40), (3, 0.82)\)</span>, then their relative risk prediction is <span class="math inline">\(\sum_t \hat{H}_i(t) = 0 + 0.10 + 0.4 + 0.82 = 1.32\)</span>.</p>
</section>
</section>
<section id="sec-survtsk-time" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="sec-survtsk-time"><span class="header-section-number">5.3</span> Predicting Survival Times</h2>
<p>Predicting a time-to-event is the problem of estimating when an individual will experience an event. Mathematically, the problem is the task of estimating <span class="math inline">\(g: \mathcal{X}\rightarrow \mathbb{R}_{&gt;0}\)</span>, that is, predicting a single value over <span class="math inline">\([0,\infty]\)</span>.</p>
<p>For practical purposes, the expected time-to-event would be the ideal prediction type as it is easy to interpret and communicate. However, evaluation of these predictions is tricky. Take the following example. Say someone is censored at <span class="math inline">\(\tau=5\)</span>, there is no way to know if they would have experienced the event at <span class="math inline">\(\tau=6\)</span> or <span class="math inline">\(\tau=600\)</span>, all that can be known is that the model correctly identified they did not experience the event before <span class="math inline">\(\tau=6\)</span>. Furthermore, if the model had predicted <span class="math inline">\(\tau=3\)</span>, metrics such as mean absolute error would be misleading, as an MAE of <span class="math inline">\(5 - 3 = 2\)</span> would make the model seem artificially better than it was, if the event actually occurred at <span class="math inline">\(\tau=600\)</span>. Therefore, when an observation is censored, the best one can do is evaluate a classification task: did the model predict that the event did not occur before the censoring time. As this is not the task of interest, one is left to evaluate uncensored observations only, which can introduce bias to the evaluation. Given this complexity, the time-to-event point prediction is rare in practice.</p>
<section id="times-and-risks" class="level3" data-number="5.3.1">
<h3 data-number="5.3.1" class="anchored" data-anchor-id="times-and-risks"><span class="header-section-number">5.3.1</span> Times and risks</h3>
<p>Converting a time-to-event prediction to a risk prediction is trivial as the former is a special case of the latter. An individual with a longer survival time will have a lower overall risk: if <span class="math inline">\(\hat{y}_i,\hat{y}_j\)</span> and <span class="math inline">\(\hat{r}_i,\hat{r}_j\)</span> are survival time and ranking predictions for subjects <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> respectively, then <span class="math inline">\(\hat{y}_i &gt; \hat{y}_j \Rightarrow \hat{r}_i &lt; \hat{r}_j\)</span>. It is not possible to make the transformation in the opposite direction without making significant assumptions as risk predictions are usually abstract quantities that rarely map to realistic survival times.</p>
</section>
<section id="times-and-distributions" class="level3" data-number="5.3.2">
<h3 data-number="5.3.2" class="anchored" data-anchor-id="times-and-distributions"><span class="header-section-number">5.3.2</span> Times and distributions</h3>
<p>Moving from a survival time to a distribution prediction is rare given the reasons outlined above. Theoretically one <em>could</em> make a prediction for the expected survival time, <span class="math inline">\(\hat{y}\)</span> and then assume a particular distributional form. For example, one could assume <span class="math inline">\(\operatorname{TruncatedNormal}(\hat{y}, \sigma, a=0, b=\infty)\)</span> where <span class="math inline">\(\hat{y}\)</span> is the predicted expected survival time, <span class="math inline">\(\sigma\)</span> is a parameter representing variance to be estimated or assumed, and <span class="math inline">\(\{a, b\}\)</span> is the distribution support. This method clearly has drawbacks given the number of required assumptions and as such is not commonly seen in practice.</p>
<p>In the other direction, it is common to to reduce a distribution prediction to a survival time prediction by <em>attempting</em> to compute the mean or median of the distribution. When there is no censoring, one can calculate the expectation from the predicted survival function using the ‘Darth Vader rule’ <span class="citation" data-cites="Muldowney2012">(<a href="P5C26_references.html#ref-Muldowney2012" role="doc-biblioref">Muldowney, Ostaszewski, and Wojdowski 2012</a>)</span>:</p>
<p><span id="eq-darth"><span class="math display">\[
\mathbb{E}[Y] = \int^\infty_0 S_Y(y) \ \text{d}y
\tag{5.1}\]</span></span></p>
<p>However, this rule is rarely usable in practice as censoring results in estimated survival distributions being ‘improper’. A valid probability distribution for a random variable <span class="math inline">\(Y\)</span> satisfies: <span class="math inline">\(\int f_Y = 1\)</span>, <span class="math inline">\(S_Y(0) = 1\)</span> and <span class="math inline">\(S_Y(\infty) = 0\)</span>. This last condition is often violated in survival distribution predictions, which often use semi-parametric methodology (<span class="quarto-unresolved-ref">?sec-surv-models-crank</span>), resulting in improper distributions. Recall from <a href="P1C4_survival.html#sec-surv-km" class="quarto-xref"><span>Section 3.5.2.1</span></a> that the Kaplan-Meier estimator is defined as:</p>
<p><span class="math display">\[
\hat{S}_{KM}(\tau) = \prod_{k:t_{(k)} \leq \tau}\left(1-\frac{d_{t_{(k)}}}{n_{t_{(k)}}}\right)
\]</span></p>
<p>This only reaches zero if every individual at risk at the last observed time-point experiences the event: <span class="math inline">\(d_{t_{(k)}} = n_{t_{(k)}}\)</span>. In practice, due to administrative censoring, there will almost always be censoring at the final time-point. As a result, <span class="math inline">\(d_{t_{(k)}} &lt; n_{t_{(k)}}\)</span> and <span class="math inline">\(\hat{S}(\infty) &gt; 0\)</span>. Heuristics have been proposed to address this including linear extrapolation to zero, or dropping the curve to zero at the final time-point, however these can introduce significant bias in the estimated survival time <span class="citation" data-cites="Han2022 Sonabend2022">(<a href="P5C26_references.html#ref-Han2022" role="doc-biblioref">Han and Jung 2022</a>; <a href="P5C26_references.html#ref-Sonabend2022" role="doc-biblioref">Sonabend, Bender, and Vollmer 2022</a>)</span>. Another possibility is to instead report the median survival time, but this is only defined if the survival curves drop below <span class="math inline">\(0.5\)</span> within the observed period, which is not guaranteed <span class="citation" data-cites="Haider2020">(<a href="P5C26_references.html#ref-Haider2020" role="doc-biblioref">Haider et al. 2020</a>)</span>.</p>
<p>One alternative is to estimate the <em>restricted mean survival time</em> (RMST) <span class="citation" data-cites="Han2022 andersen.regression.2004">(<a href="P5C26_references.html#ref-Han2022" role="doc-biblioref">Han and Jung 2022</a>; <a href="P5C26_references.html#ref-andersen.regression.2004" role="doc-biblioref">Andersen, Hansen, and Klein 2004</a>)</span>. In contrast to <a href="#eq-darth" class="quarto-xref"><span>5.1</span></a>, which integrates the survival curve over the entire time axis, the RMST places an upper-bound on the integral:</p>
<p><span id="eq-rmst"><span class="math display">\[
\operatorname{RMST}(\tau) = \int^\tau_0 S_Y(y) \ \text{d}y
\tag{5.2}\]</span></span></p>
<p>Clearly then <span class="math inline">\(\operatorname{RMST}(\infty) = \mathbb{E}[Y]\)</span> by <a href="#eq-darth" class="quarto-xref"><span>5.1</span></a>. Whereas <a href="#eq-darth" class="quarto-xref"><span>5.1</span></a> represents the average survival over <span class="math inline">\([0,\infty)\)</span>, <a href="#eq-rmst" class="quarto-xref"><span>5.2</span></a> represents the average survival time up to <span class="math inline">\(\tau\)</span>. Equivalently, the RMST is the average amount of time each individual spends without experiencing the event up to <span class="math inline">\(\tau\)</span>. The RMST treats all events happening after <span class="math inline">\(\tau\)</span> as if they happened at <span class="math inline">\(\tau\)</span>. Hence, it is a truncated expectation, estimating: <span class="math inline">\(\mathbb{E}[\min(Y, \tau)]\)</span></p>
<p>For example, say individuals are observed over times <span class="math inline">\([0,100]\)</span> and administrative censoring is present in the data, then <span class="math inline">\(\hat{S}(\tau)\)</span> is unknown for <span class="math inline">\(\tau &gt; 100\)</span> and <span class="math inline">\(\mathbb{E}[Y]\)</span> cannot be reliably computed. However, one can compute <span class="math inline">\(\operatorname{RMST}(100)\)</span> which provides an interpretable lower bound for the mean survival time: <span class="math inline">\(\operatorname{RMST}(100) \leq \mathbb{E}[Y]\)</span>. This avoids assumptions beyond <span class="math inline">\(\tau=100\)</span> and offers an interpretable prediction: “the average survival time is at least <span class="math inline">\(\operatorname{RMST}(100)\)</span>” (<a href="#fig-survtsk-rmst" class="quarto-xref">Figure&nbsp;<span>5.3</span></a>). The RMST avoids the pitfalls of computing the mean and median by remaining valid even when the predicted distribution is improper, provided that <span class="math inline">\(\tau\)</span> is chosen within the range of observed follow-up times.</p>
<p>As a worked example, say for an individual, <span class="math inline">\(i\)</span>, we have: <span class="math inline">\((t, \hat{S}_i(t)) = (0, 1), (1, 0.8), (2, 0.4), (3, 0.15)\)</span>. Using a Riemann sum approximation, <span class="math inline">\(\operatorname{RMST}(3) \approx 1 + 0.8 + 0.4 = 2.2\)</span> and <span class="math inline">\(\operatorname{RMST}(4) \approx 1 + 0.8 + 0.4 + 0.15 = 2.35\)</span>. <span class="math inline">\(\operatorname{RMST}(3)\)</span> reflects the expected time without experiencing the event up to <span class="math inline">\(\tau=3\)</span>, which means treating all observations as if the event occurred at <span class="math inline">\(\tau=3\)</span> (unless it occurred sooner). <span class="math inline">\(\operatorname{RMST}(4)\)</span> incorporates one more unit of follow-up, including the fourth survival probability in this case. In this way, the <span class="math inline">\(\operatorname{RMST}(\tau)\)</span> summarizes the average survival experience up until <span class="math inline">\(\tau\)</span>, smaller values of <span class="math inline">\(\tau\)</span> emphasize near-term survival whereas larger values approach the full mean survival time (if estimable). The actual choice of <span class="math inline">\(\tau\)</span> varies by the use-case.</p>
<div id="fig-survtsk-rmst" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Graph with 'survival probability S(t)' on the y-axis and 'Time' on the x-axis. A vertical red-line at x=100 represents the cut-off point. The area under the curve before the cut-off point is highlighted in blue, representing the RMST(100).">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-survtsk-rmst-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/survtsk/rmst.png" class="img-fluid figure-img" style="width:80.0%" alt="Graph with 'survival probability S(t)' on the y-axis and 'Time' on the x-axis. A vertical red-line at x=100 represents the cut-off point. The area under the curve before the cut-off point is highlighted in blue, representing the RMST(100).">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-survtsk-rmst-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.3: Illustrating the RMST at the final observed time-point. The RMST is the area under the survival curve up to the point of interest. The red vertical line represents the final observed time-point, the survival curve (blue) after that point is unknown.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="sec-survtsk-PI" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="sec-survtsk-PI"><span class="header-section-number">5.4</span> Prognostic Index Predictions</h2>
<p>In medical terminology (often used in survival analysis), a prognostic index is a tool that predicts outcomes based on risk factors. Given covariates, <span class="math inline">\(\mathbf{X}\in \mathbb{R}^{n \times p}\)</span>, and coefficients, <span class="math inline">\(\boldsymbol{\beta}\in \mathbb{R}^p\)</span>, the <em>linear predictor</em> is defined as <span class="math inline">\(\boldsymbol{\eta}:= \mathbf{X}\boldsymbol{\beta}\)</span>. Applying some function <span class="math inline">\(g\)</span>, which could simply be the identity function <span class="math inline">\(g(x) = x\)</span>, yields a <em>prognostic index</em>, <span class="math inline">\(g(\boldsymbol{\eta})\)</span>. A prognostic index can serve several purposes, including:</p>
<ol type="1">
<li>Scaling or normalization: simple functions to scale the linear predictor can better support interpretation and visualisation;</li>
<li>Ensuring meaningful results: for example the Cox PH model (<a href="P3C13_classical.html" class="quarto-xref"><span>Chapter 10</span></a>) applies the transformation <span class="math inline">\(g(\boldsymbol{\eta}) = \exp(\boldsymbol{\eta})\)</span>, which guarantees positivity of hazards as well as making the covariates multiplicative;</li>
<li>Aiding in interpretability: in some cases this could simply be <span class="math inline">\(g(\boldsymbol{\eta}) = -\boldsymbol{\eta}\)</span> to ensure the ‘higher value implies higher risk’ interpretation.</li>
</ol>
<section id="prognostic-index-risks-and-times" class="level3" data-number="5.4.1">
<h3 data-number="5.4.1" class="anchored" data-anchor-id="prognostic-index-risks-and-times"><span class="header-section-number">5.4.1</span> Prognostic index, risks, and times</h3>
<p>A prognostic index is a special case of the survival ranking task, assuming that there is a one-to-one mapping between the prediction and expected survival times. For example, one could define <span class="math inline">\(g(\boldsymbol{\eta}) = 1 \text{ if } \eta &lt; \text{5 or } 0 \text{ otherwise }\)</span> however this would prevent the prognostic index being used as a ranking method and it would defeat the point of a prognostic index.</p>
<p>As with a risk prediction, there is no general relationship between the prognostic index and survival time. However, there are cases where there may be a clear relationship between the two. For example, one can theoretically estimate a survival time from the linear predictor used in an accelerated failure time model (<span class="quarto-unresolved-ref">?sec-surv-models</span>). <!-- FIXME: UPDATE REFERENCE ABOVE WITH UPDATED VERSION WHEN MERGED --></p>
</section>
<section id="prognostic-index-and-distributions" class="level3" data-number="5.4.2">
<h3 data-number="5.4.2" class="anchored" data-anchor-id="prognostic-index-and-distributions"><span class="header-section-number">5.4.2</span> Prognostic index and distributions</h3>
<p>A general prognostic index cannot be generated from a distribution, however the reverse transformation is very common in survival models. The vast majority of survival models are composed from a group-wise survival time estimate and an estimation of an individual’s prognostic index. This will be seen in detail in future chapters (particularly in <a href="P3C16_boosting.html" class="quarto-xref"><span>Chapter 13</span></a> and <a href="P3C17_neural.html" class="quarto-xref"><span>Chapter 14</span></a>) but in general it is very common for models to assume data follows either proportional hazards or accelerated failure time (<a href="P3C13_classical.html" class="quarto-xref"><span>Chapter 10</span></a>). In this case models are formed of two components:</p>
<ol type="1">
<li>A ‘simple’ model that can estimate the baseline hazard, <span class="math inline">\(\hat{h}_0\)</span>, or baseline survival function, <span class="math inline">\(\hat{S}_0\)</span>. Discussed further in <a href="P3C13_classical.html" class="quarto-xref"><span>Chapter 10</span></a>, these are group-wise estimates that apply to the entire sample.</li>
<li>A sophisticated algorithm that estimates the individual linear prediction, <span class="math inline">\(\hat{\eta}\)</span>.</li>
</ol>
<p>These are composed as either:</p>
<ol type="1">
<li><span class="math inline">\(\hat{h}_0(t)\exp(\hat{\eta})\)</span> if proportional hazards is assumed; or</li>
<li><span class="math inline">\(\hat{h}_0(\exp(-\hat{\eta})t)\exp(-\hat{\eta})\)</span> for accelerated failure time models.</li>
</ol>
</section>
</section>
<section id="beyond-single-event" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="beyond-single-event"><span class="header-section-number">5.5</span> Beyond single-event</h2>
<p>In <a href="P1C5_eha.html" class="quarto-xref"><span>Chapter 4</span></a>, competing risks and multi-state models were introduced with a focus on estimating probability distributions.</p>
<p>In the competing risks setting, ‘survival time’ is ill-defined, it is ambiguous whether this refers to the time until a specific event or until any event takes place. In principle, one could apply the RMSE approach to the all-cause survival function, though this is uncommon in practice. Analogous to the single-event setting, a cause-specific prognostic index can be derived from a cause-specific cumulative incidence function, and vice versa. One might also consider transformations based on all-cause risk, but again such approaches are rarely used in applied settings.</p>
<p>In multi-state models, the primary prediction targets are transition probabilities between states over time (<a href="P1C5_eha.html#sec-multi-state" class="quarto-xref"><span>Section 4.3</span></a>). Unlike standard survival settings, there is no single notion of a ‘survival time’. Instead, one can estimate the sojourn time, which is the expected time spent in a given state, using the estimated transition probabilities. This is particularly well-defined in Markov or semi-Markov models, where sojourn times follow from standard stochastic process theory. These derivations are beyond the scope of this book; for further detail, see for example <span class="citation" data-cites="Ibe2013">(<a href="P5C26_references.html#ref-Ibe2013" role="doc-biblioref">Ibe 2013</a>)</span>.</p>
</section>
<section id="conclusion" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">5.6</span> Conclusion</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Warning</span>Key takeaways
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Survival prediction tasks fall into three categories: probabilistic (predicting survival distributions), deterministic (predicting survival times), and ranking (predicting relative risks or prognostic indices).</li>
<li>Prediction types are not directly interchangeable; transforming between them requires assumptions and often changes what is being estimated.</li>
<li>Distribution predictions are the most general, enabling derivation of other forms.</li>
<li>Survival times are intuitive but prone to overconfidence in censored data.</li>
<li>Ranking predictions are the most common but their interpretation can be complicated and their utility is limited to within-sample comparisons.</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Tip</span>Further reading
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><span class="citation" data-cites="Sonabend2022">Sonabend, Bender, and Vollmer (<a href="P5C26_references.html#ref-Sonabend2022" role="doc-biblioref">2022</a>)</span> reviews methods to transform distribution and ranking/survival time predictions, advantages and disadvantages of methods is discussed in brief in the appendix of <span class="citation" data-cites="Haider2020">Haider et al. (<a href="P5C26_references.html#ref-Haider2020" role="doc-biblioref">2020</a>)</span>.</li>
<li>For more on the RMST see <span class="citation" data-cites="Uno2014">Uno et al. (<a href="P5C26_references.html#ref-Uno2014" role="doc-biblioref">2014</a>)</span>, <span class="citation" data-cites="Han2022">Han and Jung (<a href="P5C26_references.html#ref-Han2022" role="doc-biblioref">2022</a>)</span>, and <span class="citation" data-cites="Royston2011">Royston and Parmar (<a href="P5C26_references.html#ref-Royston2011" role="doc-biblioref">2011</a>)</span>.</li>
</ul>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-andersen.regression.2004" class="csl-entry" role="listitem">
Andersen, Per Kragh, Mette Gerster Hansen, and John P. Klein. 2004. <span>“Regression <span>Analysis</span> of <span>Restricted Mean Survival Time Based</span> on <span>Pseudo-Observations</span>.”</span> <em>Lifetime Data Analysis</em> 10 (4): 335–50. <a href="https://doi.org/10.1007/s10985-004-4771-0">https://doi.org/10.1007/s10985-004-4771-0</a>.
</div>
<div id="ref-Candido2017" class="csl-entry" role="listitem">
Candido dos Reis, Francisco J, Gordon C Wishart, Ed M Dicks, David Greenberg, Jem Rashbass, Marjanka K Schmidt, Alexandra J van den Broek, et al. 2017. <span>“<span class="nocase">An updated PREDICT breast cancer prognostication and treatment benefit prediction model with independent validation</span>.”</span> <em>Breast Cancer Research</em> 19 (1): 58. <a href="https://doi.org/10.1186/s13058-017-0852-3">https://doi.org/10.1186/s13058-017-0852-3</a>.
</div>
<div id="ref-Haider2020" class="csl-entry" role="listitem">
Haider, Humza, Bret Hoehn, Sarah Davis, and Russell Greiner. 2020. <span>“<span class="nocase">Effective ways to build and evaluate individual survival distributions</span>.”</span> <em>Journal of Machine Learning Research</em> 21 (85): 1–63.
</div>
<div id="ref-Han2022" class="csl-entry" role="listitem">
Han, Kyunghwa, and Inkyung Jung. 2022. <span>“Restricted <span>Mean Survival Time</span> for <span>Survival Analysis</span>: <span>A Quick Guide</span> for <span>Clinical Researchers</span>.”</span> <em>Korean Journal of Radiology</em> 23 (5): 495–99. <a href="https://doi.org/10.3348/kjr.2022.0061">https://doi.org/10.3348/kjr.2022.0061</a>.
</div>
<div id="ref-Ibe2013" class="csl-entry" role="listitem">
Ibe, Oliver C. 2013. <em>Markov Processes for Stochastic Modeling</em>. 2nd ed. Academic Press. <a href="https://www.sciencedirect.com/book/9780124077959/markov-processes-for-stochastic-modeling">https://www.sciencedirect.com/book/9780124077959/markov-processes-for-stochastic-modeling</a>.
</div>
<div id="ref-Ishwaran2008" class="csl-entry" role="listitem">
Ishwaran, By Hemant, Udaya B Kogalur, Eugene H Blackstone, and Michael S Lauer. 2008. <span>“<span class="nocase">Random survival forests</span>.”</span> <em>The Annals of Statistics</em> 2 (3): 841–60. <a href="https://doi.org/10.1214/08-AOAS169">https://doi.org/10.1214/08-AOAS169</a>.
</div>
<div id="ref-DeRong2025" class="csl-entry" role="listitem">
Loh, De Rong, Elliot D Hill, Nan Liu, Geraldine Dawson, and Matthew M Engelhard. 2025. <span>“Limitations of Binary Classification for Long-Horizon Diagnosis Prediction and Advantages of a Discrete-Time Time-to-Event Approach: Empirical Analysis.”</span> <em>JMIR AI</em> 4 (March): e62985. <a href="https://doi.org/10.2196/62985">https://doi.org/10.2196/62985</a>.
</div>
<div id="ref-Muldowney2012" class="csl-entry" role="listitem">
Muldowney, Pat, Krzysztof Ostaszewski, and Wojciech Wojdowski. 2012. <span>“The Darth Vader Rule.”</span> <em>Tatra Mountains Mathematical Publications</em> 52 (1): 53–63. <a href="https://doi.org/10.2478/v10127-012-0025-9">https://doi.org/10.2478/v10127-012-0025-9</a>.
</div>
<div id="ref-Royston2011" class="csl-entry" role="listitem">
Royston, Patrick, and Mahesh K. B. Parmar. 2011. <span>“The Use of Restricted Mean Survival Time to Estimate the Treatment Effect in Randomized Clinical Trials When the Proportional Hazards Assumption Is in Doubt.”</span> <em>Statistics in Medicine</em> 30 (19): 2409–21. https://doi.org/<a href="https://doi.org/10.1002/sim.4274">https://doi.org/10.1002/sim.4274</a>.
</div>
<div id="ref-Sonabend2022" class="csl-entry" role="listitem">
Sonabend, Raphael, Andreas Bender, and Sebastian Vollmer. 2022. <span>“<span class="nocase">Avoiding C-hacking when evaluating survival distribution predictions with discrimination measures</span>.”</span> Edited by Zhiyong Lu. <em>Bioinformatics</em> 38 (17): 4178–84. <a href="https://doi.org/10.1093/bioinformatics/btac451">https://doi.org/10.1093/bioinformatics/btac451</a>.
</div>
<div id="ref-Uno2014" class="csl-entry" role="listitem">
Uno, Hajime, Brian Claggett, Lu Tian, Eisuke Inoue, Paul Gallo, Toshio Miyata, Deborah Schrag, et al. 2014. <span>“Moving Beyond the Hazard Ratio in Quantifying the Between-Group Difference in Survival Analysis.”</span> <em>Journal of Clinical Oncology</em> 32 (22): 2380–85. <a href="https://doi.org/10.1200/JCO.2014.55.2208">https://doi.org/10.1200/JCO.2014.55.2208</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./P1C5_eha.html" class="pagination-link" aria-label="Event-history Analysis">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./P2C8_rank.html" class="pagination-link" aria-label="Discrimination">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Discrimination</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>All content licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> <br> © Raphael Sonabend, Andreas Bender.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://www.mlsabook.com">Website</a> | <a href="https://github.com/mlsa-book/MLSA">GitHub</a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P1C6_survtsk.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P1C6_survtsk.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>