<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.24">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Raphael Sonabend and Andreas Bender">

<title>10&nbsp; Classical Models – Machine Learning in Survival Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./P3C14_forests.html" rel="next">
<link href="./P2C11_time.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-16365895d012c6821ad2015e20b8b579.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-be89772f0c623d959023f561edd8d78b.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P3C13_classical.html">Models</a></li><li class="breadcrumb-item"><a href="./P3C13_classical.html"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Classical Models</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Machine Learning in Survival Analysis</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/mlsa-book/MLSA/tree/main/book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./Machine-Learning-in-Survival-Analysis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C0_notation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Symbols and Notation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Survival Analysis and Machine Learning</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C3_machinelearning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C4_survival.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C5_eha.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C6_survtsk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Survival Task</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C8_rank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Discrimination</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C9_calib.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Calibration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C10_rules.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Scoring Rules</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C11_time.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Survival Time Measures</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Models</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C13_classical.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Classical Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C14_forests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Random Forests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C15_svm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C16_boosting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Boosting Methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C17_neural.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Neural Networks</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Reduction Techniques</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C19_reductions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Reductions</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C20_competing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Competing Risks Pipelines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C21_discrete.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Discrete Time Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C22_poisson.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Connections to Poisson Regression and Processes</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C23_pseudo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Connections to Regression and Imputation</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C24_conclusions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">FAQs and Outlook</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C25_exercises.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exercises</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C26_references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-surv-models" id="toc-sec-surv-models" class="nav-link active" data-scroll-target="#sec-surv-models"><span class="header-section-number">10.1</span> A Review of Classical Survival Models</a>
  <ul class="collapse">
  <li><a href="#sec-surv-models-uncond" id="toc-sec-surv-models-uncond" class="nav-link" data-scroll-target="#sec-surv-models-uncond"><span class="header-section-number">10.1.1</span> Non-Parametric Distribution Estimators</a></li>
  <li><a href="#sec-surv-models-crank" id="toc-sec-surv-models-crank" class="nav-link" data-scroll-target="#sec-surv-models-crank"><span class="header-section-number">10.1.2</span> Continuous Ranking and Semi-Parametric Models: Cox PH</a></li>
  <li><a href="#sec-surv-models-param" id="toc-sec-surv-models-param" class="nav-link" data-scroll-target="#sec-surv-models-param"><span class="header-section-number">10.1.3</span> Conditional Distribution Predictions: Parametric Linear Models</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P3C13_classical.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P3C13_classical.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P3C13_classical.html">Models</a></li><li class="breadcrumb-item"><a href="./P3C13_classical.html"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Classical Models</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span id="sec-models-classical" class="quarto-section-identifier"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Classical Models</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    TODO (150-200 WORDS)
  </div>
</div>


</header>


<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>Major changes expected!
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This page is a work in progress and major changes will be made over time.</strong></p>
</div>
</div>
<hr>
<p>TODO</p>
<ul>
<li>change chapter name (don’t use “classical”)</li>
<li></li>
</ul>
<hr>
<section id="sec-surv-models" class="level2" data-number="10.1">
<h2 data-number="10.1" class="anchored" data-anchor-id="sec-surv-models"><span class="header-section-number">10.1</span> A Review of Classical Survival Models</h2>
<p>This chapter provides a brief review of classical survival models before later chapters move on to machine learning models. ‘Classical’ models are defined with a very narrow scope in this book: low-complexity models that are either non-parametric or have parameters that can be fit with maximum likelihood estimation (or an equivalent method). In contrast, ‘machine learning’ (ML) models require more intensive model fitting procedures such as recursion or iteration. The classical models in this paper are fast to fit and highly interpretable, though can be inflexible and may make unreasonable assumptions. Whereas the ML models are more flexible with hyper-parameters however are computationally more intensive (both in terms of speed and storage), require tuning to produce ‘good’ results, and are often a ‘black-box’ with difficult interpretation.</p>
<p>As classical survival models have been studied extensively for decades, these are only discussed briefly here, primarily these are of interest as many of these models will be seen to influence machine learning extensions. The scope of the models discussed in this chapter is limited to the general book scope (<span class="quarto-unresolved-ref">?sec-surv-scope</span>), i.e.&nbsp;single event with right-censoring and no competing-risks, though in some cases these are discussed.</p>
<p>There are several possible taxonomies for categorising statistical models, these include:</p>
<ul>
<li>Parametrisation Type: One of non-, semi-, or fully-parametric. \ Non-parametric models assume that the data distribution cannot be specified with a finite set of parameters. In contrast, fully-parametric models assume the distribution can be specified with a finite set of parameters. Semi-parametric models are a hybrid of the two and are formed of a finite set of parameters <em>and</em> an infinite-dimensional ‘nuisance’ parameter.</li>
<li>Conditionality Type: One of unconditional or conditional. A conditional prediction is one that makes use of covariates in order to condition the prediction on each observation. Unconditional predictors, which are referred to below as ‘estimators’, ignore covariate data and make the same prediction for all individuals.</li>
<li>Prediction Type: One of ranking, survival time, or distribution (<a href="P1C6_survtsk.html" class="quarto-xref"><span>Chapter 5</span></a>).</li>
</ul>
<p><a href="#tbl-surv-models" class="quarto-xref">Table&nbsp;<span>10.1</span></a> summarises the models discussed below into the taxonomies above for reference. Note that the Cox model is listed as predicting a continuous ranking, and not a survival distribution, which may appear inconsistent with other definitions. The reason for this is elaborated upon in <a href="P4C19_reductions.html" class="quarto-xref"><span>Chapter 15</span></a>. Though the predict-type taxonomy is favoured throughout this book, it is clearer to review classical models in increasing complexity, beginning with unconditional estimators before moving onto semi-parametric continuous ranking predictions, and finally conditional distribution predictors. The review is brief with mathematics limited to the model fundamentals but not including methods for parameter estimation. Also the review is limited to the ‘basic’ model specification and common extensions such as regularization are not discussed though they do exist for many of these models.</p>
<p>All classical models are highly transparent and accessible, with decades of research and many off-shelf implementations. Predictive performance of each model is briefly discussed as part of the review and then again in <span class="citation" data-cites="Sonabend2021b">(<a href="P5C26_references.html#ref-Sonabend2021b" role="doc-biblioref">R. E. B. Sonabend 2021</a>)</span>.</p>
<div id="tbl-surv-models" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-surv-models-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;10.1: Table of models discussed in this literature review, classified by parametrisation, prediction type, and conditionality.
</figcaption>
<div aria-describedby="tbl-surv-models-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<thead>
<tr class="header">
<th>Model<span class="math inline">\(^1\)</span></th>
<th>Parametrisation<span class="math inline">\(^2\)</span></th>
<th>Prediction<span class="math inline">\(^3\)</span></th>
<th>Conditionality</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Kaplan-Meier</td>
<td>Non</td>
<td>Distr.</td>
<td>Unconditional</td>
</tr>
<tr class="even">
<td>Nelson-Aalen</td>
<td>Non</td>
<td>Distr.</td>
<td>Unconditional</td>
</tr>
<tr class="odd">
<td>Akritas</td>
<td>Non</td>
<td>Distr.</td>
<td>Conditional</td>
</tr>
<tr class="even">
<td>Cox PH</td>
<td>Semi</td>
<td>Rank</td>
<td>Conditional</td>
</tr>
<tr class="odd">
<td>Parametric PH</td>
<td>Fully</td>
<td>Distr.</td>
<td>Conditional</td>
</tr>
<tr class="even">
<td>Accelerated Failure Time</td>
<td>Fully</td>
<td>Distr.</td>
<td>Conditional</td>
</tr>
<tr class="odd">
<td>Proportional Odds</td>
<td>Fully</td>
<td>Distr.</td>
<td>Conditional</td>
</tr>
<tr class="even">
<td>Flexible Spline</td>
<td>Fully</td>
<td>Distr.</td>
<td>Conditional</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p><sup> * 1. All models are implemented in the <span class="math inline">\(\textsf{R}\)</span> package <span class="math inline">\(\textbf{survival}\)</span> <span class="citation" data-cites="pkgsurvival">(<a href="P5C26_references.html#ref-pkgsurvival" role="doc-biblioref">Therneau 2015</a>)</span> with the exception of flexible splines, implemented in <span class="math inline">\(\textbf{flexsurv}\)</span> <span class="citation" data-cites="pkgflexsurv">(<a href="P5C26_references.html#ref-pkgflexsurv" role="doc-biblioref">Jackson 2016</a>)</span>, and the Akritas estimator in <span class="math inline">\(\textbf{survivalmodels}\)</span> <span class="citation" data-cites="pkgsurvivalmodels">(<a href="P5C26_references.html#ref-pkgsurvivalmodels" role="doc-biblioref">R. Sonabend 2020</a>)</span>. * 2. Non = non-parametric, Semi = semi-parametric, Fully = fully-parametric. * 3. Distr. = distribution, Rank = ranking. </sup></p>
<section id="sec-surv-models-uncond" class="level3" data-number="10.1.1">
<h3 data-number="10.1.1" class="anchored" data-anchor-id="sec-surv-models-uncond"><span class="header-section-number">10.1.1</span> Non-Parametric Distribution Estimators</h3>
<section id="unconditional-estimators" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="unconditional-estimators">Unconditional Estimators</h4>
<p>Unconditional non-parametric survival models assume no distribution for survival times and estimate the survival function using simple algorithms based on observed outcomes and no covariate data. The two most common methods are the Kaplan-Meier estimator <span class="citation" data-cites="KaplanMeier1958">(<a href="P5C26_references.html#ref-KaplanMeier1958" role="doc-biblioref"><strong>KaplanMeier1958?</strong></a>)</span>, which estimates the average survival function of a training dataset, and the Nelson-Aalen estimator <span class="citation" data-cites="Aalen1978 Nelson1972">(<a href="P5C26_references.html#ref-Aalen1978" role="doc-biblioref">Aalen 1978</a>; <a href="P5C26_references.html#ref-Nelson1972" role="doc-biblioref">Nelson 1972</a>)</span>, which estimates the average cumulative hazard function of a training dataset.</p>
<p>The Kaplan-Meier estimator of the survival function is given by <span id="eq-km-classical"><span class="math display">\[
\hat{S}_{KM}(\tau|\mathcal{D}_{train}) = \prod_{t \in \mathcal{U}_O, t \leq \tau} \Big(1 - \frac{d_t}{n_t}\Big)
\tag{10.1}\]</span></span></p>
<p>As this estimate is so important in survival models, this book will always use the symbol <span class="math inline">\(\hat{S}_{KM}\)</span> to refer to the Kaplan-Meier estimate of the average survival function fit on training data <span class="math inline">\((T_i, \Delta_i)\)</span>. Another valuable function is the Kaplan-Meier estimate of the average survival function of the <em>censoring</em> distribution, which is the same as above but estimated on <span class="math inline">\((T_i, 1 - \Delta_i)\)</span>, this will be denoted by <span class="math inline">\(\hat{G}_{KM}\)</span>.</p>
<p>The Nelson-Aalen estimator for the cumulative hazard function is given by <span id="eq-na"><span class="math display">\[
\hat{H}(\tau|\mathcal{D}_{train}) = \sum_{t \in \mathcal{U}_O, t \leq \tau} \frac{d_t}{n_t}
\tag{10.2}\]</span></span></p>
<p>The primary advantage of these models is that they rely on heuristics from empirical outcomes only and don’t require any assumptions about the form of the data. To train the models they only require <span class="math inline">\((T_i,\Delta_i)\)</span> and both return a prediction of <span class="math inline">\(\mathcal{S}\subseteq \operatorname{Distr}(\mathcal{T})\)</span> (<span class="citation" data-cites="box-task-surv">(<a href="P5C26_references.html#ref-box-task-surv" role="doc-biblioref"><strong>box-task-surv?</strong></a>)</span>). In addition, both simply account for censoring and can be utilised in fitting other models or to estimate unknown censoring distributions. The Kaplan-Meier and Nelson-Aalen estimators are both consistent estimators for the survival and cumulative hazard functions respectively.</p>
<p>Utilising the relationships provided in (<a href="P1C4_survival.html#sec-distributions-continuous" class="quarto-xref"><span>Section 3.1.1</span></a>), one could write the Nelson-Aalen estimator in terms of the survival function as <span class="math inline">\(\hat{S}_{NA} = \exp(-\hat{H}(\tau|\mathcal{D}_{train}))\)</span>. It has been demonstrated that <span class="math inline">\(\hat{S}_{NA}\)</span> and <span class="math inline">\(\hat{S}_{KM}\)</span> are asymptotically equivalent, but that <span class="math inline">\(\hat{S}_{NA}\)</span> will provide larger estimates than <span class="math inline">\(\hat{S}_{KM}\)</span> in smaller samples <span class="citation" data-cites="Colosimo2002">(<a href="P5C26_references.html#ref-Colosimo2002" role="doc-biblioref">Colosimo et al. 2002</a>)</span>. In practice, the Kaplan-Meier is the most widely utilised non-parametric estimator in survival analysis and is the simplest estimator that yields consistent estimation of a survival distribution; it is therefore a natural, and commonly utilised, ‘baseline’ model <span class="citation" data-cites="Binder2008 Herrmann2020 Huang2020 Wang2017">(<a href="P5C26_references.html#ref-Binder2008" role="doc-biblioref">Binder and Schumacher 2008</a>; <a href="P5C26_references.html#ref-Herrmann2020" role="doc-biblioref">Herrmann et al. 2021</a>; <a href="P5C26_references.html#ref-Huang2020" role="doc-biblioref">Huang et al. 2020</a>; <a href="P5C26_references.html#ref-Wang2017" role="doc-biblioref">Wang, Li, and Reddy 2019</a>)</span>: estimators that other models should be ‘judged’ against to ascertain their overall performance (<span class="quarto-unresolved-ref">?sec-eval</span>).</p>
<p>Not only can these estimators be used for analytical comparison, but they also provide intuitive methods for graphical calibration of models (<a href="P2C9_calib.html#sec-calib-prob" class="quarto-xref"><span>Section 7.2</span></a>). These models are never stuidied for prognosis directly but as baselines, components of complex models (<a href="P4C19_reductions.html" class="quarto-xref"><span>Chapter 15</span></a>), or graphical tools <span class="citation" data-cites="Habibi2018 Jager2008 Moghimi-dehkordi2008">(<a href="P5C26_references.html#ref-Habibi2018" role="doc-biblioref">Habibi et al. 2018</a>; <a href="P5C26_references.html#ref-Jager2008" role="doc-biblioref">Jager et al. 2008</a>; <a href="P5C26_references.html#ref-Moghimi-dehkordi2008" role="doc-biblioref">Moghimi-dehkordi et al. 2008</a>)</span>. The reason for this is due to them having poor predictive performance as a result of omitting explanatory variables in fitting. Moreover, if the data follows a particular distribution, parametric methods will be more efficient <span class="citation" data-cites="Wang2017">(<a href="P5C26_references.html#ref-Wang2017" role="doc-biblioref">Wang, Li, and Reddy 2019</a>)</span>.</p>
</section>
<section id="conditional-estimators" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="conditional-estimators">Conditional Estimators</h4>
<p>The Kaplan-Meier and Nelson-Aalen estimators are simple to compute and provide good estimates for the survival time distribution but in many cases they may be overly-simplistic. Conditional non-parametric estimators include the advantages described above (no assumptions about underlying data distribution) but also allow for conditioning the estimation on the covariates. This is particularly useful when estimating a censoring distribution that may depend on the data (<span class="quarto-unresolved-ref">?sec-eval</span>). However predictive performance of conditional non-parametric estimators decreases as the number of covariates increases, and these models are especially poor when censoring is feature-dependent <span class="citation" data-cites="Gerds2006">(<a href="P5C26_references.html#ref-Gerds2006" role="doc-biblioref">Gerds and Schumacher 2006</a>)</span>.</p>
<p>The most widely used conditional non-parametric estimator for survival analysis is the Akritas estimator <span class="citation" data-cites="Akritas1994">(<a href="P5C26_references.html#ref-Akritas1994" role="doc-biblioref">Akritas 1994</a>)</span> defined by <span class="math display">\[
\hat{S}(\tau|X^*, \mathcal{D}_{train}, \lambda) = \prod_{j:T_j \leq \tau, \Delta_j = 1} \Big(1 - \frac{K(X^*, X_j|\lambda)}{\sum_{l = 1}^n K(X^*, X_l|\lambda)\mathbb{I}(T_l \geq T_j)}\Big)
\]</span> where <span class="math inline">\(K\)</span> is a kernel function, usually <span class="math inline">\(K(x,y|\lambda) = \mathbb{I}(\lvert \hat{F}_X(x) - \hat{F}_X(y)\rvert &lt; \lambda), \lambda \in (0, 1]\)</span>, <span class="math inline">\(\hat{F}_X\)</span> is the empirical distribution function of the training data, <span class="math inline">\(X_1,...,X_n\)</span>, and <span class="math inline">\(\lambda\)</span> is a hyper-parameter. The estimator can be interpreted as a conditional Kaplan-Meier estimator which is computed on a neighbourhood of subjects closest to <span class="math inline">\(X^*\)</span> <span class="citation" data-cites="Blanche2013">(<a href="P5C26_references.html#ref-Blanche2013" role="doc-biblioref">Blanche, Dartigues, and Jacqmin-Gadda 2013</a>)</span>. To account for tied survival times, the following adaptation of the estimator is utilised <span class="citation" data-cites="Blanche2013">(<a href="P5C26_references.html#ref-Blanche2013" role="doc-biblioref">Blanche, Dartigues, and Jacqmin-Gadda 2013</a>)</span></p>
<p><span id="eq-akritas"><span class="math display">\[
\hat{S}(\tau|X^*, \mathcal{D}_{train}, \lambda) = \prod_{t \in \mathcal{U}_O, t \leq \tau} \Big(1 - \frac{\sum^n_{j=1} K(X^*,X_j|\lambda)\mathbb{I}(T_j = t, \Delta_j = 1)}{\sum^n_{j=1} K(X^*,X_j|\lambda)\mathbb{I}(T_j \geq t)}\Big)
\tag{10.3}\]</span></span> If <span class="math inline">\(\lambda = 1\)</span> then <span class="math inline">\(K(\cdot|\lambda) = 1\)</span> and the estimator is identical to the Kaplan-Meier estimator.</p>
<p>The non-parametric nature of the model is highlighted in (<a href="#eq-akritas" class="quarto-xref">Equation&nbsp;<span>10.3</span></a>), in which both the fitting and predicting stages are combined into a single equation. A new observation, <span class="math inline">\(X^*\)</span>, is compared to its nearest neighbours from a training dataset, <span class="math inline">\(\mathcal{D}_{train}\)</span>, without a separated fitting procedure. One could consider splitting fitting and predicting in order to clearly separate between training and testing data. In this case, the fitting procedure is the estimation of <span class="math inline">\(\hat{F}_X\)</span> on training data and the prediction is given by (<a href="#eq-akritas" class="quarto-xref">Equation&nbsp;<span>10.3</span></a>) with <span class="math inline">\(\hat{F}_X\)</span> as an argument. This separated fit/predict method is implemented in <span class="math inline">\(\textbf{survivalmodels}\)</span> <span class="citation" data-cites="pkgsurvivalmodels">(<a href="P5C26_references.html#ref-pkgsurvivalmodels" role="doc-biblioref">R. Sonabend 2020</a>)</span>. As with other non-parametric estimators, the Akritas estimator can still be considered transparent and accessible. With respect to predictive performance, the Akritas estimator has more explanatory power than non-parametric estimators due to conditioning on covariates, however this is limited to a very small number of variables and therefore this estimator is still best placed as a conditional baseline.</p>
</section>
</section>
<section id="sec-surv-models-crank" class="level3" data-number="10.1.2">
<h3 data-number="10.1.2" class="anchored" data-anchor-id="sec-surv-models-crank"><span class="header-section-number">10.1.2</span> Continuous Ranking and Semi-Parametric Models: Cox PH</h3>
<p>The Cox Proportional Hazards (CPH) <span class="citation" data-cites="Cox1972">(<a href="P5C26_references.html#ref-Cox1972" role="doc-biblioref">Cox 1972</a>)</span>, or Cox model, is likely the most widely known semi-parametric model and the most studied survival model <span class="citation" data-cites="Habibi2018 Moghimi-dehkordi2008 Reid1994 Wang2017">(<a href="P5C26_references.html#ref-Habibi2018" role="doc-biblioref">Habibi et al. 2018</a>; <a href="P5C26_references.html#ref-Moghimi-dehkordi2008" role="doc-biblioref">Moghimi-dehkordi et al. 2008</a>; <a href="P5C26_references.html#ref-Reid1994" role="doc-biblioref">Reid 1994</a>; <a href="P5C26_references.html#ref-Wang2017" role="doc-biblioref">Wang, Li, and Reddy 2019</a>)</span>. The Cox model assumes that the hazard for a subject is proportionally related to their explanatory variables, <span class="math inline">\(X_1,...,X_n\)</span>, via some baseline hazard that all subjects in a given dataset share (‘the PH assumption’). The hazard function in the Cox PH model is defined by <span class="math display">\[
h(\tau|X_i)= h_0(\tau)\exp(X_i\beta)
\]</span> where <span class="math inline">\(h_0\)</span> is the non-negative <em>baseline hazard function</em> and <span class="math inline">\(\beta = \beta_1,...,\beta_p\)</span> where <span class="math inline">\(\beta_i \in \mathbb{R}\)</span> are coefficients to be fit. Note the proportional hazards (PH) assumption can be seen as the estimated hazard, <span class="math inline">\(h(\tau|X_i)\)</span>, is directly proportional to the model covariates <span class="math inline">\(\exp(X_i\beta)\)</span>. Whilst a form is assumed for the ‘risk’ component of the model, <span class="math inline">\(\exp(X_i\beta)\)</span>, no assumptions are made about the distribution of <span class="math inline">\(h_0\)</span>, hence the model is semi-parametric.</p>
<p>The coefficients, <span class="math inline">\(\beta\)</span>, are estimated by maximum likelihood estimation of the ‘partial likelihood’ <span class="citation" data-cites="Cox1975">(<a href="P5C26_references.html#ref-Cox1975" role="doc-biblioref">Cox 1975</a>)</span>, which only makes use of ordered event times and does not utilise all data available (hence being ‘partial’). The partial likelihood allows study of the informative <span class="math inline">\(\beta\)</span>-parameters whilst ignoring the nuisance <span class="math inline">\(h_0\)</span>. The predicted linear predictor, <span class="math inline">\(\hat{\eta} := X^*\hat{\beta}\)</span>, can be computed from the estimated <span class="math inline">\(\hat{\beta}\)</span> to provide a ranking prediction.</p>
<p>Inspection of the model is also useful without specifying the full hazard by interpreting the coefficients as ‘hazard ratios’. Let <span class="math inline">\(p = 1\)</span> and <span class="math inline">\(\hat{\beta} \in \mathbb{R}\)</span> and let <span class="math inline">\(X_i,X_j \in \mathbb{R}\)</span> be the covariates of two training observations, then the <em>hazard ratio</em> for these observations is the ratio of their hazard functions, <span class="math display">\[
\frac{h(\tau|X_i)}{h(\tau|X_j)} = \frac{h_0(\tau)\exp(X_i\hat{\beta})}{h_0(\tau)\exp(X_j\hat{\beta})} =  \exp(\hat{\beta})^{X_i - X_j}
\]</span></p>
<p>If <span class="math inline">\(\exp(\hat{\beta}) = 1\)</span> then <span class="math inline">\(h(\tau|X_i) = h(\tau|X_j)\)</span> and thus the covariate has no effect on the hazard. If <span class="math inline">\(\exp(\hat{\beta}) &gt; 1\)</span> then <span class="math inline">\(X_i &gt; X_j \rightarrow h(\tau|X_i) &gt; h(\tau|X_i)\)</span> and therefore the covariate is positively correlated with the hazard (increases risk of event). Finally if <span class="math inline">\(\exp(\hat{\beta}) &lt; 1\)</span> then <span class="math inline">\(X_i &gt; X_j \rightarrow h(\tau|X_i) &lt; h(\tau|X_i)\)</span> and the covariate is negatively correlated with the hazard (decreases risk of event).</p>
<p>Interpreting hazard ratios is known to be a challenge, especially by clinicians who require simple statistics to communicate to patients <span class="citation" data-cites="Sashegyi2017 Spruance2004">(<a href="P5C26_references.html#ref-Sashegyi2017" role="doc-biblioref">Sashegyi and Ferry 2017</a>; <a href="P5C26_references.html#ref-Spruance2004" role="doc-biblioref">Spruance et al. 2004</a>)</span>. For example the full interpretation of a hazard ratio of ‘2’ for binary covariate <span class="math inline">\(X\)</span> would be: ‘assuming that the risk of death is constant at all time-points then the instantaneous risk of death is twice as high in a patient with <span class="math inline">\(X\)</span> than without’. Simple conclusions are limited to stating if patients are at more or less risk than others in their cohort. Further disadvantages of the model also lie in its lack of real-world interpretabilitity, these include <span class="citation" data-cites="Reid1994">(<a href="P5C26_references.html#ref-Reid1994" role="doc-biblioref">Reid 1994</a>)</span>:</p>
<ul>
<li>the PH assumption may not be realistic and the risk of event may not be constant over time;</li>
<li>the estimated baseline hazard from a non-parametric estimator is a discrete step-function resulting in a discrete survival distribution prediction despite time being continuous; and</li>
<li>the estimated baseline hazard will be constant after the last observed time-point in the training set <span class="citation" data-cites="Gelfand2000">(<a href="P5C26_references.html#ref-Gelfand2000" role="doc-biblioref">Gelfand et al. 2000</a>)</span>.</li>
</ul>
<p>Despite these disadvantages, the model has been demonstrated to have excellent predictive performance and routinely outperforms (or at least does not underperform) sophisticated ML models <span class="citation" data-cites="Gensheimer2018 Luxhoj1997 VanBelle2011b">(<a href="P5C26_references.html#ref-Gensheimer2018" role="doc-biblioref">Gensheimer and Narasimhan 2018</a>; <a href="P5C26_references.html#ref-Luxhoj1997" role="doc-biblioref">Luxhoj and Shyur 1997</a>; <a href="P5C26_references.html#ref-VanBelle2011b" role="doc-biblioref">Van Belle et al. 2011</a>)</span> (and <span class="citation" data-cites="Sonabend2021b">(<a href="P5C26_references.html#ref-Sonabend2021b" role="doc-biblioref">R. E. B. Sonabend 2021</a>)</span>). It’s simple form and wide popularity mean that it is also highly transparent and accessible.</p>
<p>The next class of models address some of the Cox model disadvantages by making assumptions about the baseline hazard.</p>
</section>
<section id="sec-surv-models-param" class="level3" data-number="10.1.3">
<h3 data-number="10.1.3" class="anchored" data-anchor-id="sec-surv-models-param"><span class="header-section-number">10.1.3</span> Conditional Distribution Predictions: Parametric Linear Models</h3>
<section id="parametric-proportional-hazards" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="parametric-proportional-hazards">Parametric Proportional Hazards</h4>
<p>The CPH model can be extended to a fully parametric PH model by substituting the unknown baseline hazard, <span class="math inline">\(h_0\)</span>, for a particular parameterisation. Common choices for distributions are Exponential, Weibull and Gompertz <span class="citation" data-cites="Kalbfleisch2011 Wang2017">(<a href="P5C26_references.html#ref-Kalbfleisch2011" role="doc-biblioref">Kalbfleisch and Prentice 2011</a>; <a href="P5C26_references.html#ref-Wang2017" role="doc-biblioref">Wang, Li, and Reddy 2019</a>)</span>; their hazard functions are summarised in (<span class="citation" data-cites="tab-survivaldists">(<a href="P5C26_references.html#ref-tab-survivaldists" role="doc-biblioref"><strong>tab-survivaldists?</strong></a>)</span>) along with the respective parametric PH model. Whilst an Exponential assumption leads to the simplest hazard function, which is constant over time, this is often not realistic in real-world applications. As such the Weibull or Gompertz distributions are often preferred. Moreover, when the shape parameter, <span class="math inline">\(\gamma\)</span>, is <span class="math inline">\(1\)</span> in the Weibull distribution or <span class="math inline">\(0\)</span> in the Gompertz distribution, their hazards reduce to a constant risk ((<a href="#fig-survhazards" class="quarto-xref">Figure&nbsp;<span>10.1</span></a>)). As this model is fully parametric, the model parameters can be fit with maximum likelihood estimation, with the likelihood dependent on the chosen distribution.</p>
<div id="tbl-survivaldists" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-survivaldists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;10.2: Exponential, Weibull, and Gompertz hazard functions and PH specification.
</figcaption>
<div aria-describedby="tbl-survivaldists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<colgroup>
<col style="width: 28%">
<col style="width: 28%">
<col style="width: 42%">
</colgroup>
<thead>
<tr class="header">
<th>Distribution<span class="math inline">\(^1\)</span></th>
<th><span class="math inline">\(h_0(\tau)^2\)</span></th>
<th><span class="math inline">\(h(\tau|X_i)^3\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\operatorname{Exp}(\lambda)\)</span></td>
<td><span class="math inline">\(\lambda\)</span></td>
<td><span class="math inline">\(\lambda\exp(X_i\beta)\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(\operatorname{Weibull}(\gamma, \lambda)\)</span></td>
<td><span class="math inline">\(\lambda\gamma \tau^{\gamma-1}\)</span></td>
<td><span class="math inline">\(\lambda\gamma \tau^{\gamma-1}\exp(X_i\beta)\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\operatorname{Gompertz}(\gamma, \lambda)\)</span></td>
<td><span class="math inline">\(\lambda \exp(\gamma \tau)\)</span></td>
<td><span class="math inline">\(\lambda \exp(\gamma \tau)\exp(X_i\beta)\)</span></td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p><sup> * 1. Distribution choices for baseline hazard. <span class="math inline">\(\gamma,\lambda\)</span> are shape and scale parameters respectively. * 2. Baseline hazard function, which is the (unconditional) hazard of the distribution. * 3. PH hazard function, <span class="math inline">\(h(\tau|X_i) = h_0(\tau)\exp(X_i\beta)\)</span>. </sup></p>
<div id="fig-survhazards" class="quarto-float quarto-figure quarto-figure-center anchored" alt="TODO">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-survhazards-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/classical/hazards.png" class="img-fluid figure-img" alt="TODO">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-survhazards-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.1: Comparing the hazard curves under Weibull and Gompertz distributions for varying values of the shape parameter; scale parameters are set so that each parametrisation has a median of 20. x-axes are time and y-axes are Weibull (top) and Gompertz (bottom) hazards as a function of time.
</figcaption>
</figure>
</div>
<p>In the literature, the Weibull distribution tends to be favoured as the initial assumption for the survival distribution <span class="citation" data-cites="Gensheimer2018 Habibi2018 Hielscher2010 CoxSnell1968 Rahman2017">(<a href="P5C26_references.html#ref-Gensheimer2018" role="doc-biblioref">Gensheimer and Narasimhan 2018</a>; <a href="P5C26_references.html#ref-Habibi2018" role="doc-biblioref">Habibi et al. 2018</a>; <a href="P5C26_references.html#ref-Hielscher2010" role="doc-biblioref">Hielscher et al. 2010</a>; <a href="P5C26_references.html#ref-CoxSnell1968" role="doc-biblioref">R. and J. 1968</a>; <a href="P5C26_references.html#ref-Rahman2017" role="doc-biblioref">Rahman et al. 2017</a>)</span>, though Gompertz is often tested in death-outcome models for its foundations in modelling human mortality <span class="citation" data-cites="Gompertz1825">(<a href="P5C26_references.html#ref-Gompertz1825" role="doc-biblioref">Gompertz 1825</a>)</span>. There exist many tests for checking the goodness-of-model-fit (<span class="quarto-unresolved-ref">?sec-eval-insample</span>) and the distribution choice can even be treated as a model hyper-parameter. Moreover it transpires that model inference and predictions are largely insensitive to the choice of distribution <span class="citation" data-cites="Collett2014 Reid1994">(<a href="P5C26_references.html#ref-Collett2014" role="doc-biblioref">Collett 2014</a>; <a href="P5C26_references.html#ref-Reid1994" role="doc-biblioref">Reid 1994</a>)</span>. In contrast to the Cox model, fully parametric PH models can predict absolutely continuous survival distributions, they do not treat the baseline hazard as a nuisance, and in general will result in more precise and interpretable predictions if the distribution is correctly specified <span class="citation" data-cites="Reid1994 RoystonParmar2002">(<a href="P5C26_references.html#ref-Reid1994" role="doc-biblioref">Reid 1994</a>; <a href="P5C26_references.html#ref-RoystonParmar2002" role="doc-biblioref">Royston and Parmar 2002</a>)</span>.</p>
<p>Whilst misspecification of the distribution tends not to affect predictions too greatly, PH models will generally perform worse when the PH assumption is not valid. PH models can be extended to include time-varying coefficients or model stratification <span class="citation" data-cites="Cox1972">(<a href="P5C26_references.html#ref-Cox1972" role="doc-biblioref">Cox 1972</a>)</span> but even with these adaptations the model may not reflect reality. For example, the predicted hazard in a PH model will be either monotonically increasing or decreasing but there are many scenarios where this is not realistic, such as when recovering from a major operation where risks tends to increase in the short-term before decreasing. Accelerated failure time models overcome this disadvantage and allow more flexible modelling, discussed next.</p>
</section>
<section id="accelerated-failure-time" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="accelerated-failure-time">Accelerated Failure Time</h4>
<p>In contrast to the PH assumption, where a unit increase in a covariate is a multiplicative increase in the hazard rate, the Accelerated Failure Time (AFT) assumption means that a unit increase in a covariate results in an acceleration or deceleration towards death (expanded on below). The hazard representation of an AFT model demonstrates how the interpretation of covariates differs from PH models, <span class="math display">\[
h(\tau|X_i)= h_0(\exp(-X_i\beta)\tau)\exp(-X_i\beta)
\]</span> where <span class="math inline">\(\beta = (\beta_1,...,\beta_p)\)</span> are model coefficients. In contrast to PH models, the ‘risk’ component, <span class="math inline">\(\exp(-X_i\beta)\)</span>, is the exponential of the <em>negative</em> linear predictor and therefore an increase in a covariate value results in a decrease of the predicted hazard. This representation also highlights how AFT models are more flexible than PH as the predicted hazard can be non-monotonic. For example the hazard of the Log-logistic distribution ((<a href="#fig-litreview-logloghaz" class="quarto-xref">Figure&nbsp;<span>10.2</span></a>)) is highly flexible depending on chosen parameters. Not only can the AFT model offer a wider range of shapes for the hazard function but it is more interpretable. Whereas covariates in a PH model act on the hazard, in an AFT they act on time, which is most clearly seen in the log-linear representation, <span class="math display">\[
\log Y_i = \mu + \alpha_1X_{i1} + \alpha_2X_{i2} + ... + \alpha_pX_{ip} + \sigma\epsilon_i
\]</span> where <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\sigma\)</span> are location and scale parameters respectively, <span class="math inline">\(\alpha_1,...,\alpha_p\)</span> are model coefficients, and <span class="math inline">\(\epsilon_i\)</span> is a random error term. In this case a one unit increase in covariate <span class="math inline">\(X_{ij}\)</span> means a <span class="math inline">\(\alpha_j\)</span> increase in the logarithmic survival time. For example if <span class="math inline">\(\exp(X_i\alpha) = 0.5\)</span> then <span class="math inline">\(i\)</span> ‘ages’ at double the baseline ‘speed’. Or less abstractly if studying the time until death from cancer then <span class="math inline">\(\exp(X_i\alpha) = 0.5\)</span> can be interpreted as ‘the entire process from developing tumours to metastasis and eventual death in subject <span class="math inline">\(i\)</span> is twice as fast than the normal’, where ‘normal’ refers to the baseline when all covariates are <span class="math inline">\(0\)</span>.</p>
<p>Specifying a particular distribution for <span class="math inline">\(\epsilon_i\)</span> yields a fully-parametric AFT model. Common distribution choices include Weibull, Exponential, Log-logistic, and Log-Normal <span class="citation" data-cites="Kalbfleisch2011 Wang2017">(<a href="P5C26_references.html#ref-Kalbfleisch2011" role="doc-biblioref">Kalbfleisch and Prentice 2011</a>; <a href="P5C26_references.html#ref-Wang2017" role="doc-biblioref">Wang, Li, and Reddy 2019</a>)</span>. The Buckley-James estimator <span class="citation" data-cites="Buckley1979">(<a href="P5C26_references.html#ref-Buckley1979" role="doc-biblioref">Buckley and James 1979</a>)</span> is a semi-parametric AFT model that non-parametrically estimates the distribution of the errors however this model has no theoretical justification and is rarely fit in practice <span class="citation" data-cites="Wei1992">(<a href="P5C26_references.html#ref-Wei1992" role="doc-biblioref">Wei 1992</a>)</span>. The fully-parametric model has theoretical justifications, natural interpretability, and can often provide a better fit than a PH model, especially when the PH assumption is violated <span class="citation" data-cites="Patel2006 Qi2009 Zare2015">(<a href="P5C26_references.html#ref-Patel2006" role="doc-biblioref">Patel, Kay, and Rowell 2006</a>; <a href="P5C26_references.html#ref-Qi2009" role="doc-biblioref">Qi 2009</a>; <a href="P5C26_references.html#ref-Zare2015" role="doc-biblioref">Zare et al. 2015</a>)</span>.</p>
<div id="fig-litreview-logloghaz" class="quarto-float quarto-figure quarto-figure-center anchored" alt="TODO">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-litreview-logloghaz-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/classical/llog_hazard.png" class="img-fluid figure-img" alt="TODO">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-litreview-logloghaz-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10.2: Log-logistic hazard curves with a fixed scale parameter of 1 and a changing shape parameter. x-axis is time and y-axis is the log-logistic hazard as a function of time.
</figcaption>
</figure>
</div>
</section>
<section id="proportional-odds" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="proportional-odds">Proportional Odds</h4>
<p>Proportional odds (PO) models <span class="citation" data-cites="Bennett1983">(<a href="P5C26_references.html#ref-Bennett1983" role="doc-biblioref">Bennett 1983</a>)</span> fit a proportional relationship between covariates and the odds of survival beyond a time <span class="math inline">\(\tau\)</span>, <span class="math display">\[
O_i(\tau) = \frac{S_i(\tau)}{F_i(\tau)} = O_0(\tau)\exp(X_i\beta)
\]</span> where <span class="math inline">\(O_0\)</span> is the baseline odds.</p>
<p>In this model, a unit increase in a covariate is a multiplicative increase in the odds of survival after a given time and the model can be interpreted as estimating the log-odds ratio. There is no simple closed form expression for the partial likelihood of the PO model and hence in practice a Log-logistic distribution is usually assumed for the baseline odds and the model is fit by maximum likelihood estimation on the full likelihood <span class="citation" data-cites="Bennett1983">(<a href="P5C26_references.html#ref-Bennett1983" role="doc-biblioref">Bennett 1983</a>)</span>.</p>
<p>Perhaps the most useful feature of the model is convergence of hazard functions <span class="citation" data-cites="Kirmani2001">(<a href="P5C26_references.html#ref-Kirmani2001" role="doc-biblioref">Kirmani and Gupta 2001</a>)</span>, which states <span class="math inline">\(h_i(\tau)/h_0(\tau) \rightarrow 1\)</span> as <span class="math inline">\(\tau \rightarrow \infty\)</span>. This property accurately reflects real-world scenarios, for example if comparing chemotherapy treatment on advanced cancer survival rates, then it is expected that after a long period (say 10 years) the difference in risk between groups is likely to be negligible. This is in contrast to the PH model that assumes the hazard ratios are constant over time, which is rarely a reflection of reality.</p>
<p>In practice, the PO model is harder to fit and is less flexible than PH and AFT models, both of which can also produce odds ratios. This may be a reason for the lack of popularity of the PO model, in addition there is limited off-shelf implementations <span class="citation" data-cites="Collett2014">(<a href="P5C26_references.html#ref-Collett2014" role="doc-biblioref">Collett 2014</a>)</span>. Despite PO models not being commonly utilised, they have formed useful components of neural networks (<span class="quarto-unresolved-ref">?sec-surv-ml-models-nn</span>) and flexible parametric models (below).</p>
</section>
<section id="flexible-parametric-models-splines" class="level4 unnumbered unlisted">
<h4 class="unnumbered unlisted anchored" data-anchor-id="flexible-parametric-models-splines">Flexible Parametric Models – Splines</h4>
<p>Royston-Parmar flexible parametric models <span class="citation" data-cites="RoystonParmar2002">(<a href="P5C26_references.html#ref-RoystonParmar2002" role="doc-biblioref">Royston and Parmar 2002</a>)</span> extend PH and PO models by estimating the baseline hazard with natural cubic splines. The model was designed to keep the form of the PH or PO methods but without the semi-parametric problem of estimating a baseline hazard that does not reflect reality (see above), or the parametric problem of misspecifying the survival distribution.</p>
<p>To provide an interpretable, informative and smooth hazard, natural cubic splines are fit in place of the baseline hazard. The crux of the method is to use splines to model time on a log-scale and to either estimate the log cumulative Hazard for PH models, <span class="math inline">\(\log H(\tau|X_i) = \log H_0(\tau) + X_i\beta\)</span>, or the log Odds for PO models, <span class="math inline">\(\log O(\tau|X_i) = \log O_0(\tau) + X_i\beta\)</span>, where <span class="math inline">\(\beta\)</span> are model coefficients to fit, <span class="math inline">\(H_0\)</span> is the baseline cumulative hazard function and <span class="math inline">\(O_0\)</span> is the baseline odds function. For the flexible PH model, a Weibull distribution is the basis for the baseline distribution and a Log-logistic distribution for the baseline odds in the flexible PO model. <span class="math inline">\(\log H_0(\tau)\)</span> and <span class="math inline">\(\log O_0(\tau)\)</span> are estimated by natural cubic splines with coefficients fit by maximum likelihood estimation. The standard full likelihood is optimised, full details are not provided here. Between one and three internal knots are recommended for the splines and the placement of knots does not greatly impact upon the fitted model <span class="citation" data-cites="RoystonParmar2002">(<a href="P5C26_references.html#ref-RoystonParmar2002" role="doc-biblioref">Royston and Parmar 2002</a>)</span>.</p>
<p>Advantages of the model include being: interpretable, flexible, can be fit with time-dependent covariates, and it returns a continuous function. Moreover many of the parameters, including the number and position of knots, are tunable, although Royston and Parmar advised against tuning and suggest often only one internal knot is required <span class="citation" data-cites="RoystonParmar2002">(<a href="P5C26_references.html#ref-RoystonParmar2002" role="doc-biblioref">Royston and Parmar 2002</a>)</span>. A recent simulation study demonstrated that even with an increased number of knots (up to seven degrees of freedom), there was little bias in estimation of the survival and hazard functions <span class="citation" data-cites="Bower2019">(<a href="P5C26_references.html#ref-Bower2019" role="doc-biblioref">Bower et al. 2019</a>)</span>. Despite its advantages, a 2018 review <span class="citation" data-cites="Ng2018">(<a href="P5C26_references.html#ref-Ng2018" role="doc-biblioref">Ng et al. 2018</a>)</span> found only twelve instances of published flexible parametric models since Royston and Parmar’s 2002 paper, perhaps because it is more complex to train, has a less intuitive fitting procedure than alternatives, and has limited off-shelf implementations; i.e.&nbsp;is less transparent and accessible than parametric alternatives.</p>
<p>The PH and AFT models are both very transparent and accessible, though require slightly more expert knowledge than the CPH in order to specify the ‘correct’ underlying probability distribution. Interestingly whilst there are many papers comparing PH and AFT models to one another using in-sample metrics (<span class="quarto-unresolved-ref">?sec-eval-insample</span>) such as AIC <span class="citation" data-cites="Georgousopoulou2015 Habibi2018 Moghimi-dehkordi2008 Zare2015">(<a href="P5C26_references.html#ref-Georgousopoulou2015" role="doc-biblioref">Georgousopoulou et al. 2015</a>; <a href="P5C26_references.html#ref-Habibi2018" role="doc-biblioref">Habibi et al. 2018</a>; <a href="P5C26_references.html#ref-Moghimi-dehkordi2008" role="doc-biblioref">Moghimi-dehkordi et al. 2008</a>; <a href="P5C26_references.html#ref-Zare2015" role="doc-biblioref">Zare et al. 2015</a>)</span>, no benchmark experiments could be found for out-of-sample performance. PO and spline models are less transparent than PH and AFT models and are even less accessible, with very few implementations of either. No conclusions can be drawn about the predictive performance of PO or spline models due to a lack of suitable benchmark experiments.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Aalen1978" class="csl-entry" role="listitem">
Aalen, Odd. 1978. <span>“<span class="nocase">Nonparametric Inference for a Family of Counting Processes</span>.”</span> <em>The Annals of Statistics</em> 6 (4): 701–26.
</div>
<div id="ref-Akritas1994" class="csl-entry" role="listitem">
Akritas, Michael G. 1994. <span>“<span class="nocase">Nearest Neighbor Estimation of a Bivariate Distribution Under Random Censoring</span>.”</span> <em>Ann. Statist.</em> 22 (3): 1299–1327. <a href="https://doi.org/10.1214/aos/1176325630">https://doi.org/10.1214/aos/1176325630</a>.
</div>
<div id="ref-Bennett1983" class="csl-entry" role="listitem">
Bennett, Steve. 1983. <span>“<span class="nocase">Analysis of survival data by the proportional odds model</span>.”</span> <em>Statistics in Medicine</em> 2 (2): 273–77. https://doi.org/<a href="https://doi.org/10.1002/sim.4780020223">https://doi.org/10.1002/sim.4780020223</a>.
</div>
<div id="ref-Binder2008" class="csl-entry" role="listitem">
Binder, Harald, and Martin Schumacher. 2008. <span>“<span class="nocase">Allowing for mandatory covariates in boosting estimation of sparse high-dimensional survival models</span>.”</span> <em>BMC Bioinformatics</em> 9 (1): 14. <a href="https://doi.org/10.1186/1471-2105-9-14">https://doi.org/10.1186/1471-2105-9-14</a>.
</div>
<div id="ref-Blanche2013" class="csl-entry" role="listitem">
Blanche, Paul, Jean-François Dartigues, and Hélène Jacqmin-Gadda. 2013. <span>“<span class="nocase">Review and comparison of ROC curve estimators for a time-dependent outcome with marker-dependent censoring</span>.”</span> <em>Biometrical Journal</em> 55 (5): 687–704. <a href="https://doi.org/10.1002/bimj.201200045">https://doi.org/10.1002/bimj.201200045</a>.
</div>
<div id="ref-Bower2019" class="csl-entry" role="listitem">
Bower, Hannah, Michael J Crowther, Mark J Rutherford, Therese M.-L. Andersson, Mark Clements, Xing-Rong Liu, Paul W Dickman, and Paul C Lambert. 2019. <span>“<span class="nocase">Capturing simple and complex time-dependent effects using flexible parametric survival models: A simulation study</span>.”</span> <em>Communications in Statistics - Simulation and Computation</em>, July, 1–17. <a href="https://doi.org/10.1080/03610918.2019.1634201">https://doi.org/10.1080/03610918.2019.1634201</a>.
</div>
<div id="ref-Buckley1979" class="csl-entry" role="listitem">
Buckley, Jonathan, and Ian James. 1979. <span>“<span class="nocase">Linear Regression with Censored Data</span>.”</span> <em>Biometrika</em> 66 (3): 429–36. <a href="https://doi.org/10.2307/2335161">https://doi.org/10.2307/2335161</a>.
</div>
<div id="ref-Collett2014" class="csl-entry" role="listitem">
Collett, David. 2014. <em><span class="nocase">Modelling Survival Data in Medical Research</span></em>. 3rd ed. CRC.
</div>
<div id="ref-Colosimo2002" class="csl-entry" role="listitem">
Colosimo, Enrico, Fla´vio Ferreira, Maristela Oliveira, and Cleide Sousa. 2002. <span>“<span class="nocase">Empirical comparisons between Kaplan-Meier and Nelson-Aalen survival function estimators</span>.”</span> <em>Journal of Statistical Computation and Simulation</em> 72 (4): 299–308. <a href="https://doi.org/10.1080/00949650212847">https://doi.org/10.1080/00949650212847</a>.
</div>
<div id="ref-Cox1972" class="csl-entry" role="listitem">
Cox, D. R. 1972. <span>“<span class="nocase">Regression Models and Life-Tables</span>.”</span> <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 34 (2): 187–220.
</div>
<div id="ref-Cox1975" class="csl-entry" role="listitem">
———. 1975. <span>“<span>Partial Likelihood</span>.”</span> <em>Biometrika</em> 62 (2): 269–76. <a href="https://doi.org/10.1080/03610910701884021">https://doi.org/10.1080/03610910701884021</a>.
</div>
<div id="ref-Gelfand2000" class="csl-entry" role="listitem">
Gelfand, Alan E, Sujit K Ghosh, Cindy Christiansen, Stephen B Soumerai, and Thomas J McLaughlin. 2000. <span>“<span class="nocase">Proportional hazards models: a latent competing risk approach</span>.”</span> <em>Journal of the Royal Statistical Society: Series C (Applied Statistics)</em> 49 (3): 385–97. https://doi.org/<a href="https://doi.org/10.1111/1467-9876.00199">https://doi.org/10.1111/1467-9876.00199</a>.
</div>
<div id="ref-Gensheimer2018" class="csl-entry" role="listitem">
Gensheimer, Michael F., and Balasubramanian Narasimhan. 2018. <span>“<span class="nocase">A Simple Discrete-Time Survival Model for Neural Networks</span>,”</span> 1–17. <a href="https://doi.org/arXiv:1805.00917v3">https://doi.org/arXiv:1805.00917v3</a>.
</div>
<div id="ref-Georgousopoulou2015" class="csl-entry" role="listitem">
Georgousopoulou, Ekavi N, Christos Pitsavos, Christos Mary Yannakoulia, and Demosthenes B Panagiotakos. 2015. <span>“<span class="nocase">Comparisons between Survival Models in Predicting Cardiovascular Disease Events : Application in the ATTICA Study ( 2002-2012 ).</span>”</span> <em>Journal of Statistics Applications &amp; Probability</em> 4 (2): 203–10.
</div>
<div id="ref-Gerds2006" class="csl-entry" role="listitem">
Gerds, Thomas A, and Martin Schumacher. 2006. <span>“<span class="nocase">Consistent Estimation of the Expected Brier Score in General Survival Models with Right-Censored Event Times</span>.”</span> <em>Biometrical Journal</em> 48 (6): 1029–40. <a href="https://doi.org/10.1002/bimj.200610301">https://doi.org/10.1002/bimj.200610301</a>.
</div>
<div id="ref-Gompertz1825" class="csl-entry" role="listitem">
Gompertz, Benjamin. 1825. <span>“<span class="nocase">On the Nature of the Function Expressive of the Law of Human Mortality, and on a New Mode of Determining the Value of Life Contingencies</span>.”</span> <em>Philosophical Transactions of the Royal Society of London</em> 115: 513–83.
</div>
<div id="ref-Habibi2018" class="csl-entry" role="listitem">
Habibi, Danial, Mohammad Rafiei, Ali Chehrei, Zahra Shayan, and Soheil Tafaqodi. 2018. <span>“<span class="nocase">Comparison of Survival Models for Analyzing Prognostic Factors in Gastric Cancer Patients</span>.”</span> <em>Asian Pacific Journal of Cancer Prevention : APJCP</em> 19 (3): 749–53. <a href="https://doi.org/10.22034/APJCP.2018.19.3.749">https://doi.org/10.22034/APJCP.2018.19.3.749</a>.
</div>
<div id="ref-Herrmann2020" class="csl-entry" role="listitem">
Herrmann, Moritz, Philipp Probst, Roman Hornung, Vindi Jurinovic, and Anne-Laure Boulesteix. 2021. <span>“<span class="nocase">Large-scale benchmark study of survival prediction methods using multi-omics data</span>.”</span> <em>Briefings in Bioinformatics</em> 22 (3). <a href="https://doi.org/10.1093/bib/bbaa167">https://doi.org/10.1093/bib/bbaa167</a>.
</div>
<div id="ref-Hielscher2010" class="csl-entry" role="listitem">
Hielscher, Thomas, Manuela Zucknick, Wiebke Werft, and Axel Benner. 2010. <span>“<span class="nocase">On the Prognostic Value of Gene Expression Signatures for Censored Data BT - Advances in Data Analysis, Data Handling and Business Intelligence</span>.”</span> In, edited by Andreas Fink, Berthold Lausen, Wilfried Seidel, and Alfred Ultsch, 663–73. Berlin, Heidelberg: Springer Berlin Heidelberg.
</div>
<div id="ref-Huang2020" class="csl-entry" role="listitem">
Huang, Shigao, Jie Yang, Simon Fong, and Qi Zhao. 2020. <span>“<span class="nocase">Artificial intelligence in cancer diagnosis and prognosis: Opportunities and challenges</span>.”</span> <em>Cancer Letters</em> 471: 61–71. https://doi.org/<a href="https://doi.org/10.1016/j.canlet.2019.12.007">https://doi.org/10.1016/j.canlet.2019.12.007</a>.
</div>
<div id="ref-pkgflexsurv" class="csl-entry" role="listitem">
Jackson, Christopher. 2016. <span>“<span class="nocase">flexsurv: A Platform for Parametric Survival Modeling in R</span>.”</span> <em>Journal of Statistical Software</em> 70 (8): 1–33.
</div>
<div id="ref-Jager2008" class="csl-entry" role="listitem">
Jager, Kitty J, Paul C van Dijk, Carmine Zoccali, and Friedo W Dekker. 2008. <span>“<span class="nocase">The analysis of survival data: the Kaplan–Meier method</span>.”</span> <em>Kidney International</em> 74 (5): 560–65. https://doi.org/<a href="https://doi.org/10.1038/ki.2008.217">https://doi.org/10.1038/ki.2008.217</a>.
</div>
<div id="ref-Kalbfleisch2011" class="csl-entry" role="listitem">
Kalbfleisch, John D, and Ross L Prentice. 2011. <em><span class="nocase">The statistical analysis of failure time data</span></em>. Vol. 360. John Wiley &amp; Sons.
</div>
<div id="ref-Kirmani2001" class="csl-entry" role="listitem">
Kirmani, S N U A, and Ramesh C Gupta. 2001. <span>“<span class="nocase">On the Proportional Odds Model in Survival Analysis</span>.”</span> <em>Annals of the Institute of Statistical Mathematics</em> 53 (2): 203–16. <a href="https://doi.org/10.1023/A:1012458303498">https://doi.org/10.1023/A:1012458303498</a>.
</div>
<div id="ref-Luxhoj1997" class="csl-entry" role="listitem">
Luxhoj, James T., and Huan Jyh Shyur. 1997. <span>“<span class="nocase">Comparison of proportional hazards models and neural networks for reliability estimation</span>.”</span> <em>Journal of Intelligent Manufacturing</em> 8 (3): 227–34. <a href="https://doi.org/10.1023/A:1018525308809">https://doi.org/10.1023/A:1018525308809</a>.
</div>
<div id="ref-Moghimi-dehkordi2008" class="csl-entry" role="listitem">
Moghimi-dehkordi, Bijan, Azadeh Safaee, Mohamad Amin Pourhoseingholi, Reza Fatemi, Ziaoddin Tabeie, and Mohammad Reza Zali. 2008. <span>“<span class="nocase">Statistical Comparison of Survival Models for Analysis of Cancer Data</span>.”</span> <em>Asian Pacific Journal of Cancer Prevention</em> 9: 417–20.
</div>
<div id="ref-Nelson1972" class="csl-entry" role="listitem">
Nelson, Wayne. 1972. <span>“<span class="nocase">Theory and Applications of Hazard Plotting for Censored Failure Data</span>.”</span> <em>Technometrics</em> 14 (4): 945–66.
</div>
<div id="ref-Ng2018" class="csl-entry" role="listitem">
Ng, Ryan, Kathy Kornas, Rinku Sutradhar, Walter P. Wodchis, and Laura C. Rosella. 2018. <span>“<span class="nocase">The current application of the Royston-Parmar model for prognostic modeling in health research: a scoping review</span>.”</span> <em>Diagnostic and Prognostic Research</em> 2 (1): 4. <a href="https://doi.org/10.1186/s41512-018-0026-5">https://doi.org/10.1186/s41512-018-0026-5</a>.
</div>
<div id="ref-Patel2006" class="csl-entry" role="listitem">
Patel, Katie, Richard Kay, and Lucy Rowell. 2006. <span>“<span class="nocase">Comparing proportional hazards and accelerated failure time models: An application in influenza</span>.”</span> <em>Pharmaceutical Statistics</em> 5 (3): 213–24. <a href="https://doi.org/10.1002/pst.213">https://doi.org/10.1002/pst.213</a>.
</div>
<div id="ref-Qi2009" class="csl-entry" role="listitem">
Qi, Jiezhi. 2009. <span>“<span class="nocase">Comparison of Proportional Hazards and Accelerated Failure Time Models</span>.”</span> PhD thesis.
</div>
<div id="ref-CoxSnell1968" class="csl-entry" role="listitem">
R., Cox, and Snell J. 1968. <span>“<span class="nocase">A General Definition of Residuals</span>.”</span> <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 30 (2): 248–75.
</div>
<div id="ref-Rahman2017" class="csl-entry" role="listitem">
Rahman, M. Shafiqur, Gareth Ambler, Babak Choodari-Oskooei, and Rumana Z. Omar. 2017. <span>“<span class="nocase">Review and evaluation of performance measures for survival prediction models in external validation settings</span>.”</span> <em>BMC Medical Research Methodology</em> 17 (1): 1–15. <a href="https://doi.org/10.1186/s12874-017-0336-2">https://doi.org/10.1186/s12874-017-0336-2</a>.
</div>
<div id="ref-Reid1994" class="csl-entry" role="listitem">
Reid, Nancy. 1994. <span>“<span class="nocase">A Conversation with Sir David Cox</span>.”</span> <em>Statistical Science</em> 9 (3): 439–55. <a href="https://doi.org/10.1214/aos/1176348654">https://doi.org/10.1214/aos/1176348654</a>.
</div>
<div id="ref-RoystonParmar2002" class="csl-entry" role="listitem">
Royston, Patrick, and Mahesh K. B. Parmar. 2002. <span>“<span class="nocase">Flexible parametric proportional-hazards and proportional-odds models for censored survival data, with application to prognostic modelling and estimation of treatment effects</span>.”</span> <em>Statistics in Medicine</em> 21 (15): 2175–97. <a href="https://doi.org/10.1002/sim.1203">https://doi.org/10.1002/sim.1203</a>.
</div>
<div id="ref-Sashegyi2017" class="csl-entry" role="listitem">
Sashegyi, Andreas, and David Ferry. 2017. <span>“<span class="nocase">On the Interpretation of the Hazard Ratio and Communication of Survival Benefit</span>.”</span> <em>The Oncologist</em> 22 (4): 484–86. <a href="https://doi.org/10.1634/theoncologist.2016-0198">https://doi.org/10.1634/theoncologist.2016-0198</a>.
</div>
<div id="ref-pkgsurvivalmodels" class="csl-entry" role="listitem">
Sonabend, Raphael. 2020. <span>“<span class="nocase">survivalmodels: Models for Survival Analysis</span>.”</span> CRAN. <a href="https://raphaels1.r-universe.dev/ui#package:survivalmodels">https://raphaels1.r-universe.dev/ui#package:survivalmodels</a>.
</div>
<div id="ref-Sonabend2021b" class="csl-entry" role="listitem">
Sonabend, Raphael Edward Benjamin. 2021. <span>“<span class="nocase">A Theoretical and Methodological Framework for Machine Learning in Survival Analysis: Enabling Transparent and Accessible Predictive Modelling on Right-Censored Time-to-Event Data</span>.”</span> PhD, University College London (UCL). <a href="https://discovery.ucl.ac.uk/id/eprint/10129352/">https://discovery.ucl.ac.uk/id/eprint/10129352/</a>.
</div>
<div id="ref-Spruance2004" class="csl-entry" role="listitem">
Spruance, Spotswood L, Julia E Reid, Michael Grace, and Matthew Samore. 2004. <span>“<span class="nocase">Hazard ratio in clinical trials</span>.”</span> <em>Antimicrobial Agents and Chemotherapy</em> 48 (8): 2787–92. <a href="https://doi.org/10.1128/AAC.48.8.2787-2792.2004">https://doi.org/10.1128/AAC.48.8.2787-2792.2004</a>.
</div>
<div id="ref-pkgsurvival" class="csl-entry" role="listitem">
Therneau, Terry M. 2015. <span>“<span class="nocase">A Package for Survival Analysis in S</span>.”</span> <a href="https://cran.r-project.org/package=survival">https://cran.r-project.org/package=survival</a>.
</div>
<div id="ref-VanBelle2011b" class="csl-entry" role="listitem">
Van Belle, Vanya, Kristiaan Pelckmans, Sabine Van Huffel, and Johan A. K. Suykens. 2011. <span>“<span class="nocase">Support vector methods for survival analysis: A comparison between ranking and regression approaches</span>.”</span> <em>Artificial Intelligence in Medicine</em> 53 (2): 107–18. <a href="https://doi.org/10.1016/j.artmed.2011.06.006">https://doi.org/10.1016/j.artmed.2011.06.006</a>.
</div>
<div id="ref-Wang2017" class="csl-entry" role="listitem">
Wang, Ping, Yan Li, and Chandan K. Reddy. 2019. <span>“<span class="nocase">Machine Learning for Survival Analysis</span>.”</span> <em>ACM Computing Surveys</em> 51 (6): 1–36. <a href="https://doi.org/10.1145/3214306">https://doi.org/10.1145/3214306</a>.
</div>
<div id="ref-Wei1992" class="csl-entry" role="listitem">
Wei, L J. 1992. <span>“<span class="nocase">The Accelerated Failure Time Model: A Useful Alternative to the Cox Regression Model in Survival Analysis</span>.”</span> <em>Statistics in Medicine</em> 11: 1871–79.
</div>
<div id="ref-Zare2015" class="csl-entry" role="listitem">
Zare, Ali, Mostafa Hosseini, Mahmood Mahmoodi, Kazem Mohammad, Hojjat Zeraati, and Kourosh Holakouie Naieni. 2015. <span>“<span class="nocase">A Comparison between Accelerated Failure-time and Cox Proportional Hazard Models in Analyzing the Survival of Gastric Cancer Patients.</span>”</span> <em>Iranian Journal of Public Health</em> 44 (8): 1095–1102. <a href="https://doi.org/10.1007/s00606-006-0435-8">https://doi.org/10.1007/s00606-006-0435-8</a>.
</div>
</div>
</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./P2C11_time.html" class="pagination-link" aria-label="Survival Time Measures">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Survival Time Measures</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./P3C14_forests.html" class="pagination-link" aria-label="Random Forests">
        <span class="nav-page-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Random Forests</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>All content licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> <br> © Raphael Sonabend, Andreas Bender.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://www.mlsabook.com">Website</a> | <a href="https://github.com/mlsa-book/MLSA">GitHub</a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P3C13_classical.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P3C13_classical.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>