<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Raphael Sonabend and Andreas Bender">

<title>13&nbsp; Support Vector Machines – Machine Learning in Survival Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./P3C16_boosting.html" rel="next">
<link href="./P3C14_forests.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-0d45b1ff1595a53868627e64e30aef28.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-b6db0a1bf8162d09ed5006fd111a6427.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P3C13_classical.html">Models</a></li><li class="breadcrumb-item"><a href="./P3C15_svm.html"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Machine Learning in Survival Analysis</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/mlsa-book/MLSA/tree/main/book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./Machine-Learning-in-Survival-Analysis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C0_notation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Symbols and Notation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P0C1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Survival Analysis and Machine Learning</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C2_preview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">MLSA From Start to Finish</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C3_machinelearning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C4_survival.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C5_eha.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Event-history Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P1C6_survtsk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Survival Task</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C8_rank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Discrimination</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C9_calib.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Calibration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C10_rules.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Scoring Rules</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P2C11_time.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Survival Time Measures</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Models</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C13_classical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Classical Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C14_forests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Random Forests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C15_svm.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C16_boosting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Boosting Methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P3C17_neural.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Neural Networks</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Reduction Techniques</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C19_reductions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Reductions</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C20_competing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Competing Risks Pipelines</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C21_discrete.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Discrete Time Survival Analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C22_poisson.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Connections to Poisson Regression and Processes</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P4C23_pseudo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Connections to Regression and Imputation</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C24_conclusions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">FAQs and Outlook</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C25_exercises.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exercises</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./P5C26_references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#svms-for-regression" id="toc-svms-for-regression" class="nav-link active" data-scroll-target="#svms-for-regression"><span class="header-section-number">13.1</span> SVMs for Regression</a></li>
  <li><a href="#sec-surv-ml-models-svm-surv" id="toc-sec-surv-ml-models-svm-surv" class="nav-link" data-scroll-target="#sec-surv-ml-models-svm-surv"><span class="header-section-number">13.2</span> SVMs for Survival Analysis</a>
  <ul class="collapse">
  <li><a href="#survival-time-ssvms" id="toc-survival-time-ssvms" class="nav-link" data-scroll-target="#survival-time-ssvms"><span class="header-section-number">13.2.1</span> Survival time SSVMs</a></li>
  <li><a href="#ranking-ssvms" id="toc-ranking-ssvms" class="nav-link" data-scroll-target="#ranking-ssvms"><span class="header-section-number">13.2.2</span> Ranking SSVMs</a></li>
  <li><a href="#hybrid-ssvms" id="toc-hybrid-ssvms" class="nav-link" data-scroll-target="#hybrid-ssvms"><span class="header-section-number">13.2.3</span> Hybrid SSVMs</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">13.3</span> Conclusion</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P3C15_svm.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P3C15_svm.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./P3C13_classical.html">Models</a></li><li class="breadcrumb-item"><a href="./P3C15_svm.html"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span id="sec-surv-ml-models-svm" class="quarto-section-identifier"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Support Vector Machines</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This chapter introduces support vector machines (SVMs) for regression and then describes the extensions to survival analysis. Regression SVMs extend simple linear methods by estimating flexible, non-linear hyperplanes that minimise the difference between predictions and the truth for individual observations. In survival analysis, SVMs may make survival time or ranking predictions, however there is no current formulation for survival distribution predictions. The chapter begins by discussing survival time SVMs and then ranking models before concluding with a hybrid formulation that combines both model forms. This primarily covers the work of Shivaswamy and Van Belle. SVMs are a powerful method for estimating non-linear relationships in data and have proven to be well-performing models in regression and classification. However, SVMs are less developed in survival analysis and have been shown to perform worse than other models in experiments.
  </div>
</div>


</header>


<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Minor changes expected!
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This page is a work in progress and minor changes will be made over time.</strong></p>
</div>
</div>
<p>Support vector machines are a popular class of models in regression and classification settings due to their ability to make accurate predictions for complex high-dimensional, non-linear data. Survival support vector machines (SSVMs) predict continuous responses that can be used as ranking predictions with some formulations that provide survival time interpretations. This chapter starts with SVMs in the regression setting before moving to adaptions for survival analysis.</p>
<section id="svms-for-regression" class="level2" data-number="13.1">
<h2 data-number="13.1" class="anchored" data-anchor-id="svms-for-regression"><span class="header-section-number">13.1</span> SVMs for Regression</h2>
<p>In simple linear regression, the aim is to estimate the line <span class="math inline">\(y = \alpha + x\beta_1\)</span> by estimating the <span class="math inline">\(\alpha,\beta_1\)</span> coefficients. As the number of coefficients increases, the goal is to instead estimate the <em>hyperplane</em>, which divides the higher-dimensional space into two separate parts. To visualize a hyperplane, imagine looking at a room from a birds eye view that has a dividing wall cutting the room into two halves (<a href="#fig-svm-hyper" class="quarto-xref">Figure&nbsp;<span>13.1</span></a>). In this view, the room appears to have two dimensions (x=left-right, y=top-bottom) and the divider is a simple line of the form <span class="math inline">\(y = \alpha + x\beta_1\)</span>. In reality, this room is actually three dimensional and has a third dimension (z=up-down) and the divider is therefore a hyperplane of the form <span class="math inline">\(y = \alpha + x\beta_1 + z\beta_2\)</span>.</p>
<div id="fig-svm-hyper" class="quarto-float quarto-figure quarto-figure-center anchored" alt="AI generated image of a birds-eye view of an office cut in half by a dividing wall.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-svm-hyper-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/svm/hyperplane.png" class="img-fluid figure-img" alt="AI generated image of a birds-eye view of an office cut in half by a dividing wall." width="400">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-svm-hyper-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.1: Visualising a hyperplane by viewing a 3D room in two-dimensions with a wall that is now seen as a simple line. When standing in this room, the wall will clearly exist in three dimensional space.
</figcaption>
</figure>
</div>
<p>Continuing the linear regression example, consider a simple model where the objective is to find the <span class="math inline">\(\boldsymbol{\beta} = ({\beta}_1 \ {\beta}_2 \cdots {\beta}_{p})^\top\)</span> coefficients that minimize <span class="math inline">\(\sum^n_{i=1} (g(\mathbf{x}_i) - y_i)^2\)</span> where <span class="math inline">\(g(\mathbf{x}_i) = \alpha + \mathbf{x}_i^\top\boldsymbol{\beta}\)</span> and <span class="math inline">\((\mathbf{X}, \mathbf{y})\)</span> is training data such that <span class="math inline">\(\mathbf{X}\in \mathbb{R}^{n \times p}\)</span> and <span class="math inline">\(\mathbf{y}\in \mathbb{R}^n\)</span>. In a higher-dimensional space, a penalty term can be added for variable selection to reduce model complexity, commonly of the form</p>
<p><span class="math display">\[
\frac{1}{2} \sum^n_{i=1} (g(\mathbf{x}_i) - y_i)^2 + \frac{\lambda}{2} \|\boldsymbol{\beta}\|^2
\]</span></p>
<p>for some penalty term <span class="math inline">\(\lambda \in \mathbb{R}\)</span>. Minimizing this error function effectively minimizes the <em>average</em> difference between all predictions and true outcomes, resulting in a hyperplane that represents the best <em>linear</em> relationship between coefficients and outcomes.</p>
<p>Similarly to linear regression, support vector machines (SVMs) <span class="citation" data-cites="CortesVapnik1995">(<a href="P5C26_references.html#ref-CortesVapnik1995" role="doc-biblioref">Cortes and Vapnik 1995</a>)</span> also fit a hyperplane, <span class="math inline">\(g\)</span>, on given training data, <span class="math inline">\(\mathbf{X}\)</span>. However, in SVMs, the goal is to fit a <em>flexible</em> (non-linear) hyperplane that minimizes the difference between predictions and the truth for <em>individual</em> observations. A core feature of SVMs is that one does not try to fit a hyperplane that makes perfect predictions as this would overfit the training data and is unlikely to perform well on unseen data. Instead, SVMs use a regularized error function, which allows incorrect predictions (errors) for some observations, with the magnitude of error controlled by an <span class="math inline">\(\epsilon&gt;0\)</span> parameter as well as slack parameters, <span class="math inline">\(\boldsymbol{\xi'} = ({\xi'}_1 \ {\xi'}_2 \cdots {\xi'}_{n})^\top\)</span> and <span class="math inline">\(\boldsymbol{\xi^*} = ({\xi^*}_1 \ {\xi^*}_2 \cdots {\xi^*}_{n})^\top\)</span> :</p>
<p><span id="eq-svm-opt"><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta},\alpha, \boldsymbol{\xi}', \boldsymbol{\xi}^*} \frac{1}{2} \|\boldsymbol{\beta}\|^2 + C \sum^n_{i=1}(\xi'_i + \xi_i^*) \\
&amp; \textrm{subject to}
\begin{dcases}
g(\mathbf{x}_i) &amp; \geq y_i -\epsilon - \xi'_i \\
g(\mathbf{x}_i) &amp; \leq y_i + \epsilon + \xi_i^* \\
\xi'_i, \xi_i^* &amp; \geq 0 \\
\end{dcases}
\end{aligned}
\tag{13.1}\]</span></span></p>
<p><span class="math inline">\(\forall i\in 1,...,n\)</span> where <span class="math inline">\(g(\mathbf{x}_i) = \alpha + \mathbf{x}_i^\top\boldsymbol{\beta}\)</span> for model weights <span class="math inline">\(\boldsymbol{\beta}\in \mathbb{R}^p\)</span> and <span class="math inline">\(\alpha \in \mathbb{R}\)</span> and the same training data <span class="math inline">\((\mathbf{X}, \mathbf{y})\)</span> as above.</p>
<p><a href="#fig-svm-regr" class="quarto-xref">Figure&nbsp;<span>13.2</span></a> visualizes a support vector regression model in two dimensions. The red circles are values within the <span class="math inline">\(\epsilon\)</span>-tube and are thus considered to have a negligible error. In fact, the red circles do not affect the fitting of the optimal line <span class="math inline">\(g\)</span> and even if they moved around, as long as they remain within the tube, the shape of <span class="math inline">\(g\)</span> would not change. In contrast the blue diamonds have an unacceptable margin of error – as an example the top blue diamond will have <span class="math inline">\(\xi'_i = 0\)</span> but <span class="math inline">\(\xi_i^* &gt; 0\)</span>, thus influencing the estimation of <span class="math inline">\(g\)</span>. Points on or outside the epsilon tube are referred to as <em>support vectors</em> as they affect the construction of the hyperplane. The <span class="math inline">\(C \in \mathbb{R}_{&gt;0}\)</span> hyperparameter controls the slack parameters and thus as <span class="math inline">\(C\)</span> increases, the number of errors (and subsequently support vectors) is allowed to increase resulting in low variance but higher bias, in contrast a lower <span class="math inline">\(C\)</span> is more likely to introduce overfitting with low bias but high variance <span class="citation" data-cites="Hastie2001">(<a href="P5C26_references.html#ref-Hastie2001" role="doc-biblioref">Hastie, Tibshirani, and Friedman 2001</a>)</span>. <span class="math inline">\(C\)</span> should be tuned to control this trade-off.</p>
<div id="fig-svm-regr" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Line graph with g(x) on the y-axis and 'x' on the x-axis. A solid black line labelled 'y' runs along g(x)=x, i.e., from the bottom-left to the top-right of the graph. Parallel to this line, above and below, are two dotted lines labelled 'y+epsilon' and 'y-epsilon' respectively. These dotted lines form the boundaries of the epsilon-tube. Red dots lie between the dotted lines and blue diamonds are outside the lines. One blue dot above the top dotted line is labelled to reflect that it represents the distance xi* and one blue dot below the bottom line is the distance xi.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-svm-regr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/svm/regression.png" class="img-fluid figure-img" alt="Line graph with g(x) on the y-axis and 'x' on the x-axis. A solid black line labelled 'y' runs along g(x)=x, i.e., from the bottom-left to the top-right of the graph. Parallel to this line, above and below, are two dotted lines labelled 'y+epsilon' and 'y-epsilon' respectively. These dotted lines form the boundaries of the epsilon-tube. Red dots lie between the dotted lines and blue diamonds are outside the lines. One blue dot above the top dotted line is labelled to reflect that it represents the distance xi* and one blue dot below the bottom line is the distance xi." width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-svm-regr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.2: Visualising a support vector machine with an <span class="math inline">\(\epsilon\)</span>-tube and slack parameters <span class="math inline">\(\boldsymbol{\xi}'\)</span> and <span class="math inline">\(\boldsymbol{\xi}^*\)</span>. Red circles are values within the <span class="math inline">\(\epsilon\)</span>-tube and blue diamonds are support vectors on and outside the tube. x-axis is single covariate, <span class="math inline">\(x\)</span>, and y-axis is <span class="math inline">\(g(x) = x\beta_1 + \alpha\)</span>.
</figcaption>
</figure>
</div>
<p>The other core feature of SVMs is exploiting the <em>kernel trick</em>, which uses functions known as <em>kernels</em> to allow the model to learn a non-linear hyperplane whilst keeping the computations limited to lower-dimensional settings. Once the model coefficients have been estimated using the optimization above, predictions for a new observation <span class="math inline">\(\mathbf{x}^*\)</span> can be made using a function of the form</p>
<p><span id="eq-svm-pred"><span class="math display">\[
\hat{g}(\mathbf{x}^*) = \sum^n_{i=1} \mu_iK(\mathbf{x}^*,\mathbf{x}_i) + \alpha
\tag{13.2}\]</span></span></p>
<p>Details (including estimation) of the <span class="math inline">\(\mu_i\)</span> Lagrange multipliers are beyond the scope of this book, references are given at the end of this chapter for the interested reader. <span class="math inline">\(K\)</span> is a kernel function, with common functions including the linear kernel, <span class="math inline">\(K(x^*,x_i) = \sum^p_{j=1} x_{ij}x^*_j\)</span>, radial kernel, <span class="math inline">\(K(x^*,x_i) = \exp(-\omega\sum^p_{j=1} (x_{ij} - x^*_j)^2)\)</span> for some <span class="math inline">\(\omega \in \mathbb{R}_{&gt;0}\)</span>, and polynomial kernel, <span class="math inline">\(K(x^*,x_i) = (1 + \sum^p_{j=1} x_{ij}x^*_j)^d\)</span> for some <span class="math inline">\(d \in \mathbb{N}_{&gt; 0}\)</span>.</p>
<p>The choice of kernel and its parameters, the regularization parameter <span class="math inline">\(C\)</span>, and the acceptable error <span class="math inline">\(\epsilon\)</span>, are all tunable hyper-parameters, which makes the support vector machine a highly adaptable and often well-performing machine learning method. The parameters <span class="math inline">\(C\)</span> and <span class="math inline">\(\epsilon\)</span> often have no clear apriori meaning (especially true in the survival setting predicting abstract rankings) and thus require tuning over a great range of values; no tuning usually results in a poor model fit <span class="citation" data-cites="Probst2019">(<a href="P5C26_references.html#ref-Probst2019" role="doc-biblioref">Probst, Boulesteix, and Bischl 2019</a>)</span>.</p>
</section>
<section id="sec-surv-ml-models-svm-surv" class="level2" data-number="13.2">
<h2 data-number="13.2" class="anchored" data-anchor-id="sec-surv-ml-models-svm-surv"><span class="header-section-number">13.2</span> SVMs for Survival Analysis</h2>
<p>Extending SVMs to the survival domain (SSVMs) is a case of: i) identifying the quantity to predict; and ii) updating the optimization problem (<a href="#eq-svm-opt" class="quarto-xref">Equation&nbsp;<span>13.1</span></a>) and prediction function (<a href="#eq-svm-pred" class="quarto-xref">Equation&nbsp;<span>13.2</span></a>) to accommodate for censoring. In the first case, SSVMs can be used to either make survival time or ranking predictions, which are discussed in turn. The notation above is reused below for SSVMs, with additional notation introduced when required and now using the survival training data <span class="math inline">\((\mathbf{X}, \mathbf{t}, \boldsymbol{\delta})\)</span>.</p>
<section id="survival-time-ssvms" class="level3" data-number="13.2.1">
<h3 data-number="13.2.1" class="anchored" data-anchor-id="survival-time-ssvms"><span class="header-section-number">13.2.1</span> Survival time SSVMs</h3>
<p>To begin, consider the objective for support vector regression with the <span class="math inline">\(y\)</span> variable replaced with the usual survival time outcome <span class="math inline">\(t\)</span>, for all <span class="math inline">\(i\in 1,...,n\)</span>:</p>
<p><span id="eq-svm-survnaive"><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta}, \alpha, \boldsymbol{\xi}', \boldsymbol{\xi}^*} \frac{1}{2} \|\boldsymbol{\beta}\|^2 + C \sum^n_{i=1}(\xi'_i + \xi_i^*) \\
&amp; \textrm{subject to}
\begin{dcases}
g(\mathbf{x}_i) &amp; \geq t_i -\epsilon - \xi'_i \\
g(\mathbf{x}_i) &amp; \leq t_i + \epsilon + \xi_i^* \\
\xi'_i, \xi_i^* &amp; \geq 0 \\
\end{dcases}
\end{aligned}
\tag{13.3}\]</span></span></p>
<p>In survival analysis, this translates to fitting a hyperplane in order to predict the true survival time. However, as with all adaptations from regression to survival analysis, there needs to be a method for incorporating censoring.</p>
<p>Recall the <span class="math inline">\((t_l, t_u)\)</span> notation to describe censoring as introduced in <a href="P1C4_survival.html" class="quarto-xref"><span>Chapter 4</span></a> such that the outcome occurs within the range <span class="math inline">\((t_l, t_u)\)</span>. <!-- FIXME: MAKE CROSS-REFERENCE MORE SPECIFIC --> Let <span class="math inline">\(\tau \in \mathbb{R}_{&gt;0}\)</span> be some known time-point, then an observation is:</p>
<ul>
<li>left-censored if the survival time is less than <span class="math inline">\(\tau\)</span>: <span class="math inline">\((t_l, t_u) = (-\infty, \tau)\)</span>;</li>
<li>right-censored if the true survival time is greater than <span class="math inline">\(\tau\)</span>: <span class="math inline">\((t_l, t_u) = (\tau, \infty)\)</span>; or</li>
<li>uncensored if the true survival time is known to be <span class="math inline">\(\tau\)</span>: <span class="math inline">\((t_l, t_u) = (\tau, \tau)\)</span>.</li>
</ul>
<p>Define <span class="math inline">\(\mathcal{L}= \{i: t_i &gt; -\infty\}\)</span> as the set of observations with a finite lower-bounded time, which can be seen above to be those that are right-censored or uncensored. Define <span class="math inline">\(\mathcal{U}= \{i: t_i &lt; \infty\}\)</span> as the analogous set of observations with a finite upper-bounded time, which are those that are left-censored or uncensored.</p>
<p>Consider these definitions in the context of the constraints in <a href="#eq-svm-survnaive" class="quarto-xref">Equation&nbsp;<span>13.3</span></a>. The first constraint ensures the hyperplane is greater than some lower-bound created by subtracting the slack parameter from the true outcome – given the set definitions above this constraint only has meaning for observations with a finite lower-bound, <span class="math inline">\(i \in \mathcal{L}\)</span>, otherwise the constraint would include <span class="math inline">\(g(\mathbf{x}_i) \geq -\infty\)</span>, which is clearly not useful. Similarly the second constraint ensures the hyperplane is less than some upper-bound, which again can only be meaningful for observations <span class="math inline">\(i \in \mathcal{U}\)</span>. Restricting the constraints in this way leads to the optimization problem <span class="citation" data-cites="Shivaswamy2007">(<a href="P5C26_references.html#ref-Shivaswamy2007" role="doc-biblioref">Shivaswamy, Chu, and Jansche 2007</a>)</span> below and visualised in <a href="#fig-svm-surv" class="quarto-xref">Figure&nbsp;<span>13.3</span></a>:</p>
<p><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta}, \alpha, \boldsymbol{\xi}', \boldsymbol{\xi}^*} \frac{1}{2}\|\boldsymbol{\beta}\|^2 + C\Big(\sum_{i \in \mathcal{U}} \xi_i + \sum_{i \in \mathcal{L}} \xi_i^*\Big) \\
&amp; \textrm{subject to}
\begin{dcases}
g(\mathbf{x}_i) &amp; \geq t_i -\xi'_i, i \in \mathcal{L}\\
g(\mathbf{x}_i) &amp; \leq t_i + \xi^*_i, i \in \mathcal{U}\\
\xi'_i \geq 0, \forall i\in \mathcal{L}; \xi^*_i \geq 0, \forall i \in \mathcal{U}
\end{dcases}
\end{aligned}
\]</span></p>
<p>If no observations are censored then the optimization becomes the regression optimization in (<a href="#eq-svm-opt" class="quarto-xref">Equation&nbsp;<span>13.1</span></a>). Note that in SSVMs, the <span class="math inline">\(\epsilon\)</span> parameters are typically removed to better accommodate censoring and to help prevent the same penalization of over- and under-predictions. In contrast to this formulation, one <em>could</em> introduce more <span class="math inline">\(\epsilon\)</span> and <span class="math inline">\(C\)</span> parameters to separate between under- and over-predictions and to separate right- and left-censoring, however this leads to eight tunable hyperparameters, which is inefficient and may increase overfitting <span class="citation" data-cites="pkgsurvivalsvm Land2011">(<a href="P5C26_references.html#ref-pkgsurvivalsvm" role="doc-biblioref">Fouodo et al. 2018</a>; <a href="P5C26_references.html#ref-Land2011" role="doc-biblioref">Land et al. 2011</a>)</span>. The algorithm can be simplified to right-censoring only by removing the second constraint completely for anyone censored:</p>
<p><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta}, \alpha, \boldsymbol{\xi}', \boldsymbol{\xi}^*} \frac{1}{2}\|\boldsymbol{\beta}\|^2 + C \sum_{i = 1}^n (\xi'_i + \xi_i^*) \\
&amp; \textrm{subject to}
\begin{dcases}
g(\mathbf{x}_i) &amp; \geq t_i - \xi^*_i \\
g(\mathbf{x}_i) &amp; \leq t_i + \xi'_i, i:\delta_i=1 \\
\xi'_i, \xi_i^* &amp; \geq 0 \\
\end{dcases}
\end{aligned}
\]</span></p>
<p><span class="math inline">\(\forall i\in 1,...,n\)</span>. With the prediction for a new observation <span class="math inline">\(\mathbf{x}^*\)</span> calculated as,</p>
<p><span class="math display">\[
\hat{g}(\mathbf{x}^*) = \sum^n_{i=1} \mu^*_i K(\mathbf{x}_i, \mathbf{x}^*) - \delta_i\mu_i'K(\mathbf{x}_i, \mathbf{x}^*) + \alpha
\]</span></p>
<p>Where again <span class="math inline">\(K\)</span> is a kernel function and the calculation of the Lagrange multipliers is beyond the scope of this book.</p>
<div id="fig-svm-surv" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Line graph with g(x) on the y-axis and 'x' on the x-axis. A solid black line labelled 'y' runs along g(x)=x, i.e., from the bottom-left to the top-right of the graph. Red circles and blue dots lie on both sides of the line. Blue dots above the line represent observations with finite upper bounds and blue dots below the line represent observations with finite lower bounds. The blue dots are labelled to show they represent the distance xi* and xi.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-svm-surv-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/svm/survival.png" class="img-fluid figure-img" alt="Line graph with g(x) on the y-axis and 'x' on the x-axis. A solid black line labelled 'y' runs along g(x)=x, i.e., from the bottom-left to the top-right of the graph. Red circles and blue dots lie on both sides of the line. Blue dots above the line represent observations with finite upper bounds and blue dots below the line represent observations with finite lower bounds. The blue dots are labelled to show they represent the distance xi* and xi." width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-svm-surv-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.3: Visualising a survival time SVM. Blue diamonds are influential support vectors, which are uncensored or left-censored when <span class="math inline">\(g(\mathbf{x})&lt;t\)</span> or uncensored or right-censored when <span class="math inline">\(g(\mathbf{x})&gt;t\)</span>. Red circles are non-influential observations.
</figcaption>
</figure>
</div>
</section>
<section id="ranking-ssvms" class="level3" data-number="13.2.2">
<h3 data-number="13.2.2" class="anchored" data-anchor-id="ranking-ssvms"><span class="header-section-number">13.2.2</span> Ranking SSVMs</h3>
<p>Support vector machines can be used to estimate rankings by penalizing predictions that result in disconcordant predictions. Recall the definition of concordance from <a href="P2C8_rank.html" class="quarto-xref"><span>Chapter 7</span></a>: ranking predictions for a pair of comparable observations <span class="math inline">\((i, j)\)</span> where <span class="math inline">\(t_i &lt; t_j \cap \delta_i = 1\)</span>, are called concordant if <span class="math inline">\(r_i &gt; r_j\)</span> where <span class="math inline">\(r_i, r_j\)</span> are the predicted ranks for observations <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> respectively and a higher value implies greater risk. Using the prognostic index as a ranking prediction (<a href="P1C6_survtsk.html#sec-survtsk-PI" class="quarto-xref"><span>Section 6.3</span></a>), a pair of observations is concordant if <span class="math inline">\(g(\mathbf{x}_i) &gt; g(\mathbf{x}_j)\)</span> when <span class="math inline">\(t_i &lt; t_j\)</span>, leading to:</p>
<p><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta}, \alpha, \boldsymbol{\xi}} \frac{1}{2}\|\boldsymbol{\beta}\|^2 + \gamma\sum_{i =1}^n \xi_i \\
&amp; \textrm{subject to}
\begin{dcases}
g(\mathbf{x}_i) - g(\mathbf{x}_j) &amp; \geq \xi_i, \forall i,j \in CP \\
\xi_i &amp; \geq 0, i = 1,...,n \\
\end{dcases}
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(CP\)</span> is the set of comparable pairs defined by <span class="math inline">\(CP = \{(i, j) : t_i &lt; t_j \wedge \delta_i = 1\}\)</span>. Given the number of pairs, the optimization problem quickly becomes difficult to solve with a very long runtime. To solve this problem <span class="citation" data-cites="VanBelle2011b">Van Belle et al. (<a href="P5C26_references.html#ref-VanBelle2011b" role="doc-biblioref">2011</a>)</span> found an efficient reduction that sorts observations in order of outcome time and then compares each data point <span class="math inline">\(i\)</span> with the observation that has the next smallest <em>survival</em> time, skipping over censored observations, in maths: <span class="math inline">\(j(i) := \mathop{\mathrm{arg\,max}}_{j \in 1,...,n} \{t_j : t_j &lt; t_i\}\)</span>. This is visualized in <a href="#fig-svm-surv-redux" class="quarto-xref">Figure&nbsp;<span>13.4</span></a> where six observations are sorted by outcome time from smallest (left) to largest (right). Starting from right to left, the first pair is made by matching the observation to the first uncensored outcome to the left, this continues to the end. In order for all observations to be used in the optimisation, the algorithm sets the first outcome to be uncensored hence observation <span class="math inline">\(2\)</span> being compared to observation <span class="math inline">\(1\)</span>.</p>
<div id="fig-svm-surv-redux" class="quarto-float quarto-figure quarto-figure-center anchored" alt="x-axis says 'observation', y-axis says 'outcome time'. There are six observations that increase linearly from bottom-left to top-right. The order is: uncensored, censored, uncensored, censored, uncensored, uncensored. Arrows show observation 6 matched with 4, 5 matched with 5, 4 matched with 2, 3 matched with 2, 2 matched with 1.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-svm-surv-redux-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Figures/svm/comparable.png" class="img-fluid figure-img" alt="x-axis says 'observation', y-axis says 'outcome time'. There are six observations that increase linearly from bottom-left to top-right. The order is: uncensored, censored, uncensored, censored, uncensored, uncensored. Arrows show observation 6 matched with 4, 5 matched with 5, 4 matched with 2, 3 matched with 2, 2 matched with 1." width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-svm-surv-redux-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.4: Van Belle SVM nearest neighbors reduction. Sorted observations are paired with the nearest uncensored outcome ‘to the left’. Red squares are uncensored observations and blue circles are censored. The observation with the smallest outcome time is always treated as uncensored.
</figcaption>
</figure>
</div>
<p>Using this reduction, the algorithm becomes</p>
<p><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta}, \alpha, \boldsymbol{\xi}} \frac{1}{2}\|\boldsymbol{\beta}\|^2 + \gamma\sum_{i =1}^n \xi_i \\
&amp; \textrm{subject to}
\begin{dcases}
g(\mathbf{x}_i) - g(\mathbf{x}_{j(i)}) &amp; \geq t_i - t_{j(i)} - \xi_i \\
\xi_i &amp; \geq 0
\end{dcases}
\end{aligned}
\]</span></p>
<p><span class="math inline">\(\forall i = 1,...,n\)</span>. Note the updated right hand side of the constraint, which plays a similar role to the <span class="math inline">\(\epsilon\)</span> parameter by allowing ‘mistakes’ in predictions without penalty.</p>
<p>Predictions for a new observation <span class="math inline">\(\mathbf{x}^*\)</span> are calculated as,</p>
<p><span class="math display">\[
\hat{g}(\mathbf{x}^*) = \sum^n_{i=1} \mu_i(K(\mathbf{x}_i, \mathbf{x}^*) - K(\mathbf{x}_{j(i)}, \mathbf{x}^*)) + \alpha
\]</span></p>
<p>Where <span class="math inline">\(\mu_i\)</span> are again Lagrange multipliers.</p>
</section>
<section id="hybrid-ssvms" class="level3" data-number="13.2.3">
<h3 data-number="13.2.3" class="anchored" data-anchor-id="hybrid-ssvms"><span class="header-section-number">13.2.3</span> Hybrid SSVMs</h3>
<p>Finally, <span class="citation" data-cites="VanBelle2011b">Van Belle et al. (<a href="P5C26_references.html#ref-VanBelle2011b" role="doc-biblioref">2011</a>)</span> noted that the ranking algorithm could be updated to add the constraints of the regression model, thus providing a model that simultaneously optimizes for ranking whilst providing continuous values that can be interpreted as survival time predictions. This results in the hybrid SSVM with constraints <span class="math inline">\(\forall i = 1,...,n\)</span>:</p>
<p><span class="math display">\[
\begin{aligned}
&amp; \min_{\boldsymbol{\beta}, \alpha, \boldsymbol{\xi}, \boldsymbol{\xi}', \boldsymbol{\xi}^*} \frac{1}{2}\|\boldsymbol{\beta}\|^2 + \textcolor{CornflowerBlue}{\gamma\sum_{i =1}^n \xi_i} + \textcolor{Rhodamine}{C \sum^n_{i=1}(\xi_i' + \xi_i^*)} \\
&amp; \textrm{subject to}
\begin{dcases}
\textcolor{CornflowerBlue}{g(\mathbf{x}_i) - g(\mathbf{x}_{j(i)})} &amp; \textcolor{CornflowerBlue}{\geq t_i - t_{j(i)} - \xi_i} \\
\textcolor{Rhodamine}{g(\mathbf{x}_i)} &amp; \textcolor{Rhodamine}{\leq t_i + \xi^*_i, i:\delta_i=1} \\
\textcolor{Rhodamine}{g(\mathbf{x}_i)} &amp; \textcolor{Rhodamine}{\geq t_i - \xi'_i} \\
\textcolor{CornflowerBlue}{\xi_i}, \textcolor{Rhodamine}{\xi_i', \xi_i^*} &amp; \geq 0 \\
\end{dcases}
\end{aligned}
\label{eq:surv_ssvmvb2}
\]</span></p>
<p>The blue parts of the equation make up the ranking model and the red parts are the regression model. <span class="math inline">\(\gamma\)</span> is the penalty associated with the regression method and <span class="math inline">\(C\)</span> is the penalty associated with the ranking method. Setting <span class="math inline">\(\gamma = 0\)</span> results in the regression SVM and <span class="math inline">\(C = 0\)</span> results in the ranking SSVM. Hence, fitting the hybrid model and tuning these parameters is an efficient way to automatically detect which SSVM is best suited to a given task.</p>
<p>Once the model is fit, a prediction from given features <span class="math inline">\(\mathbf{x}^* \in \mathbb{R}^p\)</span>, can be made using the equation below, again with the ranking and regression contributions highlighted in blue and red respectively.</p>
<p><span class="math display">\[
\hat{g}(\mathbf{x}^*) = \sum^n_{i=1} \textcolor{CornflowerBlue}{\mu_i(K(\mathbf{x}_i, \mathbf{x}^*) - K(\mathbf{x}_{j(i)}, \mathbf{x}^*))} + \textcolor{Rhodamine}{\mu^*_i K(\mathbf{x}_i, \mathbf{x}^*) - \delta_i\mu_i'K(\mathbf{x}_i, \mathbf{x}^*)} + \alpha
\]</span></p>
<p>where <span class="math inline">\(\mu_i, \mu_i^*, \mu_i'\)</span> are Lagrange multipliers and <span class="math inline">\(K\)</span> is a chosen kernel function, which may have further hyper-parameters to select or tune.</p>
</section>
</section>
<section id="conclusion" class="level2" data-number="13.3">
<h2 data-number="13.3" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">13.3</span> Conclusion</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Key takeaways
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Support vector machines (SVMs) are a highly flexible machine learning method that can use the ‘kernel trick’ to represent infinite dimensional spaces in finite domains;</li>
<li>Survival SVMs (SSVMs) extend regression SVMs by either making survival time predictions, ranking predictions, or a combination of the two;</li>
<li>The hybrid SSVM provides an efficient method that encapsulates all the elements of regression and ranking SSVMs and is therefore a good model to include in benchmark experiments to test the potential of SSVMs.</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-important no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Limitations
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>SSVMs can only perform well with extensive tuning of hyper-parameters over a wide parameter space. To-date, no papers have experimented with the tuning range for the <span class="math inline">\(\gamma\)</span> and <span class="math inline">\(C\)</span> parameters, we note <span class="citation" data-cites="pkgsurvivalsvm">(<a href="P5C26_references.html#ref-pkgsurvivalsvm" role="doc-biblioref">Fouodo et al. 2018</a>)</span> tune over <span class="math inline">\((2^{-5}, 2^5)\)</span>.</li>
<li>Even using the regression or hybrid model, the authors’ experiments with the SSVM have consistently shown ‘survival time’ estimates tend to be unrealistically large.</li>
<li>Due to the above limitation, regression estimates cannot be meaningful interpreted and as a consequence there is no sensible composition to create a distribution prediction from an SSVM. Hence, we are hesitant to suggest usage of SSVMs outside of ranking-based problems.</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Further reading
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><span class="citation" data-cites="Shivaswamy2007">Shivaswamy, Chu, and Jansche (<a href="P5C26_references.html#ref-Shivaswamy2007" role="doc-biblioref">2007</a>)</span>, <span class="citation" data-cites="Khan2008">Khan and Bayer Zubek (<a href="P5C26_references.html#ref-Khan2008" role="doc-biblioref">2008</a>)</span>, <span class="citation" data-cites="Land2011">Land et al. (<a href="P5C26_references.html#ref-Land2011" role="doc-biblioref">2011</a>)</span>, and <span class="citation" data-cites="VanBelle2011b">Van Belle et al. (<a href="P5C26_references.html#ref-VanBelle2011b" role="doc-biblioref">2011</a>)</span> to learn more about regression SSVMs.</li>
<li><span class="citation" data-cites="Evers2008">Evers and Messow (<a href="P5C26_references.html#ref-Evers2008" role="doc-biblioref">2008</a>)</span>, <span class="citation" data-cites="VanBelle2007">Van Belle et al. (<a href="P5C26_references.html#ref-VanBelle2007" role="doc-biblioref">2007</a>)</span>, <span class="citation" data-cites="VanBelle2008">Van Belle et al. (<a href="P5C26_references.html#ref-VanBelle2008" role="doc-biblioref">2008</a>)</span>, and <span class="citation" data-cites="VanBelle2011b">Van Belle et al. (<a href="P5C26_references.html#ref-VanBelle2011b" role="doc-biblioref">2011</a>)</span> for more information about ranking SSVMs.</li>
<li><span class="citation" data-cites="Goli2016a">Goli, Mahjub, Faradmal, and Soltanian (<a href="P5C26_references.html#ref-Goli2016a" role="doc-biblioref">2016</a>)</span> and <span class="citation" data-cites="Goli2016b">Goli, Mahjub, Faradmal, Mashayekhi, et al. (<a href="P5C26_references.html#ref-Goli2016b" role="doc-biblioref">2016</a>)</span> introduce mean residual lifetime optimization SSVMs.</li>
<li><span class="citation" data-cites="pkgsurvivalsvm">Fouodo et al. (<a href="P5C26_references.html#ref-pkgsurvivalsvm" role="doc-biblioref">2018</a>)</span> surveys and benchmarks SSVMs.</li>
</ul>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-CortesVapnik1995" class="csl-entry" role="listitem">
Cortes, Corinna, and Vladimir Vapnik. 1995. <span>“<span>Support-Vector Networks</span>.”</span> <em>Machine Learning</em> 20: 273–97. <a href="https://doi.org/10.1007/BF00994018">https://doi.org/10.1007/BF00994018</a>.
</div>
<div id="ref-Evers2008" class="csl-entry" role="listitem">
Evers, Ludger, and Claudia-Martina Messow. 2008. <span>“<span class="nocase">Sparse kernel methods for high-dimensional survival data</span>.”</span> <em>Bioinformatics</em> 24 (14): 1632–38.
</div>
<div id="ref-pkgsurvivalsvm" class="csl-entry" role="listitem">
Fouodo, Cesaire J K, I Konig, C Weihs, A Ziegler, and M Wright. 2018. <span>“<span class="nocase">Support vector machines for survival analysis with R</span>.”</span> <em>The R Journal</em> 10 (July): 412–23.
</div>
<div id="ref-Goli2016b" class="csl-entry" role="listitem">
Goli, Shahrbanoo, Hossein Mahjub, Javad Faradmal, Hoda Mashayekhi, and Ali-Reza Soltanian. 2016. <span>“<span class="nocase">Survival Prediction and Feature Selection in Patients with Breast Cancer Using Support Vector Regression</span>.”</span> Edited by Francesco Pappalardo. <em>Computational and Mathematical Methods in Medicine</em> 2016: 2157984. <a href="https://doi.org/10.1155/2016/2157984">https://doi.org/10.1155/2016/2157984</a>.
</div>
<div id="ref-Goli2016a" class="csl-entry" role="listitem">
Goli, Shahrbanoo, Hossein Mahjub, Javad Faradmal, and Ali-Reza Soltanian. 2016. <span>“<span class="nocase">Performance Evaluation of Support Vector Regression Models for Survival Analysis: A Simulation Study</span>.”</span> <em>International Journal of Advanced Computer Science and Applications</em> 7 (June). <a href="https://doi.org/10.14569/IJACSA.2016.070650">https://doi.org/10.14569/IJACSA.2016.070650</a>.
</div>
<div id="ref-Hastie2001" class="csl-entry" role="listitem">
Hastie, Trevor, Robert Tibshirani, and Jerome Friedman. 2001. <em><span class="nocase">The Elements of Statistical Learning</span></em>. Springer New York Inc.
</div>
<div id="ref-Khan2008" class="csl-entry" role="listitem">
Khan, Faisal M., and Valentina Bayer Zubek. 2008. <span>“<span class="nocase">Support vector regression for censored data (SVRc): A novel tool for survival analysis</span>.”</span> <em>Proceedings - IEEE International Conference on Data Mining, ICDM</em>, 863–68. <a href="https://doi.org/10.1109/ICDM.2008.50">https://doi.org/10.1109/ICDM.2008.50</a>.
</div>
<div id="ref-Land2011" class="csl-entry" role="listitem">
Land, Walker H, Xingye Qiao, Dan Margolis, and Ron Gottlieb. 2011. <span>“<span class="nocase">A new tool for survival analysis: evolutionary programming/evolutionary strategies (EP/ES) support vector regression hybrid using both censored / non-censored (event) data</span>.”</span> <em>Procedia Computer Science</em> 6: 267–72. https://doi.org/<a href="https://doi.org/10.1016/j.procs.2011.08.050">https://doi.org/10.1016/j.procs.2011.08.050</a>.
</div>
<div id="ref-Probst2019" class="csl-entry" role="listitem">
Probst, Philipp, Anne-Laure Boulesteix, and Bernd Bischl. 2019. <span>“Tunability: Importance of Hyperparameters of Machine Learning Algorithms.”</span> <em>Journal of Machine Learning Research</em> 20 (53): 1–32. <a href="http://jmlr.org/papers/v20/18-444.html">http://jmlr.org/papers/v20/18-444.html</a>.
</div>
<div id="ref-Shivaswamy2007" class="csl-entry" role="listitem">
Shivaswamy, Pannagadatta K., Wei Chu, and Martin Jansche. 2007. <span>“<span class="nocase">A support vector approach to censored targets</span>.”</span> In <em>Proceedings - IEEE International Conference on Data Mining, ICDM</em>, 655–60. <a href="https://doi.org/10.1109/ICDM.2007.93">https://doi.org/10.1109/ICDM.2007.93</a>.
</div>
<div id="ref-VanBelle2008" class="csl-entry" role="listitem">
Van Belle, Vanya, Kristiaan Pelckmans, Johan A K Suykens, and Sabine Van Huffel. 2008. <span>“<span class="nocase">Survival SVM: a practical scalable algorithm</span>.”</span> In <em>Proceedings of the 16th European Symposium on Artificial Neural Networks (ESANN)</em>, 89–94.
</div>
<div id="ref-VanBelle2007" class="csl-entry" role="listitem">
Van Belle, Vanya, Kristiaan Pelckmans, Johan A. K. Suykens, and Sabine Van Huffel. 2007. <span>“<span class="nocase">Support Vector Machines for Survival Analysis</span>.”</span> In <em>In Proceedings of the Third International Conference on Computational Intelligence in Medicine and Healthcare</em>. 1.
</div>
<div id="ref-VanBelle2011b" class="csl-entry" role="listitem">
Van Belle, Vanya, Kristiaan Pelckmans, Sabine Van Huffel, and Johan A. K. Suykens. 2011. <span>“<span class="nocase">Support vector methods for survival analysis: A comparison between ranking and regression approaches</span>.”</span> <em>Artificial Intelligence in Medicine</em> 53 (2): 107–18. <a href="https://doi.org/10.1016/j.artmed.2011.06.006">https://doi.org/10.1016/j.artmed.2011.06.006</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./P3C14_forests.html" class="pagination-link" aria-label="Random Forests">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Random Forests</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./P3C16_boosting.html" class="pagination-link" aria-label="Boosting Methods">
        <span class="nav-page-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Boosting Methods</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>All content licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> <br> © Raphael Sonabend, Andreas Bender.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://www.mlsabook.com">Website</a> | <a href="https://github.com/mlsa-book/MLSA">GitHub</a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/mlsa-book/MLSA/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/mlsa-book/MLSA/edit/main/book/P3C15_svm.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/mlsa-book/MLSA/blob/main/book/P3C15_svm.qmd" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>